<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Using R for proteomics data analysis • RforProteomics</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Using R for proteomics data analysis">
<meta property="og:description" content="RforProteomics">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RforProteomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.37.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/RforProteomics.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/RProtVis.html">Visualisation of proteomics data using R and Bioconductor</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="RforProteomics_files/htmlwidgets-1.6.1/htmlwidgets.js"></script><link href="RforProteomics_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="RforProteomics_files/datatables-binding-0.27/datatables.js"></script><link href="RforProteomics_files/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="RforProteomics_files/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="RforProteomics_files/dt-core-1.12.1/js/jquery.dataTables.min.js"></script><link href="RforProteomics_files/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="RforProteomics_files/crosstalk-1.2.0/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Using R for proteomics data analysis</h1>
                        <h4 data-toc-skip class="author">Laurent
Gatto</h4>
                                    <h4 data-toc-skip class="author">Lisa
Breckels</h4>
                                    <h4 data-toc-skip class="author">Vlad
Petyuk</h4>
                                    <h4 data-toc-skip class="author">Thomas
Lin Pedersen</h4>
                                    <h4 data-toc-skip class="author">Sebastian
Gibb</h4>
                        
      
      
      <div class="hidden name"><code>RforProteomics.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This is the companion vignette to the ‘Visualisation of
      proteomics data using R and Bioconductor’ manuscript that presents
      an overview of R and Bioconductor software for mass spectrometry
      and proteomics data. It provides the code to reproduce the figures
      in the paper.</p>
    </div>
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div class="section level2">
<h2 id="sec:intro">Introduction<a class="anchor" aria-label="anchor" href="#sec:intro"></a>
</h2>
<p>This document illustrates some existing R infrastructure for the
analysis of proteomics data. It presents the code for the use cases
taken from <span class="citation">(Laurent Gatto and Christoforou 2013,
Gatto:2015)</span>. A pre-print of <span class="citation">(Laurent Gatto
and Christoforou 2013)</span> available on <a href="http://arxiv.org/abs/1305.6559" class="external-link">arXiv</a> and <span class="citation">(L. Gatto et al. 2015)</span> is <a href="http://onlinelibrary.wiley.com/doi/10.1002/pmic.201400392/full" class="external-link">open
access</a>.</p>
<p>There are however numerous additional R resources distributed by the
<a href="http://www.bioconductor.org" class="external-link">Bioconductor</a> and <a href="http://cran.r-project.org/web/packages/" class="external-link">CRAN</a> repositories, as
well as packages hosted on personal websites. Section @ref(sec:packages)
tries to provide a wider picture of available packages, without going
into details.</p>
<blockquote>
<p><strong>NB</strong>: I you are interested in R packages for mass
spectrometry-based proteomics and metabolomics, see also the <a href="https://www.rformassspectrometry.org/" class="external-link">R for Mass Spectrometry
initiative</a> packages and the <a href="https://rformassspectrometry.github.io/docs/" class="external-link">tutorial book</a>.
It provides more up-to-date packages and solutions for several of the
tasks described below.</p>
</blockquote>
<div class="section level3">
<h3 id="sec:rres">General R resources<a class="anchor" aria-label="anchor" href="#sec:rres"></a>
</h3>
<p>The reader is expected to have basic R knowledge to find the document
helpful. There are numerous R introductions freely available, some of
which are listed below.</p>
<p>From the R project web-page:</p>
<ul>
<li>
<em>An Introduction to R</em> is based on the former <em>Notes on
R</em>, gives an introduction to the language and how to use R for doing
statistical analysis and graphics (<a href="http://cran.r-project.org/doc/manuals/R-intro.html" class="external-link">html</a> and
<a href="http://cran.r-project.org/doc/manuals/R-intro.pdf" class="external-link">pdf</a>.</li>
<li>Several introductory tutorials in the <a href="http://cran.r-project.org/other-docs.html" class="external-link">contributed
documentation</a> section.</li>
<li>The <a href="https://github.com/lgatto/TeachingMaterial" class="external-link">TeachingMaterial
repository</a> contains several sets of slides and vignettes about R
programming.</li>
</ul>
<p>Relevant background on the R software and its application to
computational biology in general and proteomics in particular can also
be found in <span class="citation">(Laurent Gatto and Christoforou
2013)</span>. For details about the Bioconductor project, the reader is
referred to <span class="citation">(Gentleman et al. 2004)</span>.</p>
</div>
<div class="section level3">
<h3 id="sec:biocres">Bioconductor resources<a class="anchor" aria-label="anchor" href="#sec:biocres"></a>
</h3>
<p>The Bioconductor offers many educational resources on its <a href="http://bioconductor.org/help/" class="external-link">help page</a>, in addition the
package’s vignettes (vignettes are a requirement for Bioconductor
packages). We want to draw the attention to the Bioconductor work flows
that offer a cross-package overview about a specific topic. In
particular, there is now a <a href="http://bioconductor.org/help/workflows/proteomics/" class="external-link"><em>Mass
spectrometry and proteomics data analysis</em></a> work flow.</p>
</div>
<div class="section level3">
<h3 id="getting-help">Getting help<a class="anchor" aria-label="anchor" href="#getting-help"></a>
</h3>
<p>All R packages come with ample documentation. Every command
(function, class or method) a user is susceptible to use is documented.
The documentation can be accessed by preceding the command by a
<code>?</code> in the R console. For example, to obtain help about the
<code>library</code> function, that will be used in the next section,
one would type <code><a href="https://rdrr.io/r/base/library.html" class="external-link">?library</a></code>. In addition, all Bioconductor
packages come with at least one vignette (this document is the vignette
that comes with the <em><a href="https://bioconductor.org/packages/3.17/RforProteomics" class="external-link">RforProteomics</a></em>
package), a document that combines text and R code that is executed
before the pdf is assembled. To look up all vignettes that come with a
package, say <em><a href="https://bioconductor.org/packages/3.17/RforProteomics" class="external-link">RforProteomics</a></em>
and then open the vignette of interest, one uses the
<code>vignette</code> function as illustrated below. More details can be
found in <code><a href="https://rdrr.io/r/utils/vignette.html" class="external-link">?vignette</a></code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## list all the vignettes in the RforProteomics package</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/vignette.html" class="external-link">vignette</a></span><span class="op">(</span>package <span class="op">=</span> <span class="st">"RforProteomics"</span><span class="op">)</span></span>
<span><span class="co">## Open the vignette called RforProteomics</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/vignette.html" class="external-link">vignette</a></span><span class="op">(</span><span class="st">"RforProteomics"</span>, package <span class="op">=</span> <span class="st">"RforProteomics"</span><span class="op">)</span></span>
<span><span class="co">## or just</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/vignette.html" class="external-link">vignette</a></span><span class="op">(</span><span class="st">"RforProteomics"</span><span class="op">)</span></span></code></pre></div>
<p>R has several <a href="http://www.r-project.org/mail.html" class="external-link">mailing
lists</a>. The most relevant here being the main <code>R-help</code>
list, <em>for discussion about problem and solutions using R</em>, ideal
for general R content and is not suitable for bioinformatics or
proteomics questions. Bioconductor also offers several resources
dedicated to bioinformatics matters and Bioconductor packages, in
particular the main <a href="https://support.bioconductor.org/" class="external-link">Bioconductor support forum</a>
for Bioconductor-related queries.</p>
<p>It is advised to read and comply to the <a href="http://www.r-project.org/posting-guide.html" class="external-link">posting guides</a>
(and <a href="http://bioconductor.org/help/mailing-list/posting-guide/" class="external-link">here</a>
to maximise the chances to obtain good responses. It is important to
specify the software versions using the <code><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo()</a></code>
functions (see an example output at the end of this document. It the
question involves some code, make sure to isolate the relevant portion
and report it with your question, trying to make your <a href="https://github.com/hadley/devtools/wiki/Reproducibility%3E" class="external-link">code/example
reproducible</a>.</p>
</div>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>The package should be installed using as described below:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## only first time you install Bioconductor packages</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="co">## else</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"RforProteomics"</span><span class="op">)</span></span></code></pre></div>
<p>To install all dependencies and reproduce the code in the vignette,
replace the last line in the code chunk above with:)</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"RforProteomics"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Finally, the package can be loaded with</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="http://lgatto.github.com/RforProteomics/" class="external-link">"RforProteomics"</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## This is the 'RforProteomics' version 1.37.2.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   To get started, visit</span></span>
<span><span class="co">##     http://lgatto.github.com/RforProteomics/</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   or, in R, open package vignettes by typing</span></span>
<span><span class="co">##     RforProteomics() # R/Bioc for proteomics overview</span></span>
<span><span class="co">##     RProtVis()       # R/Bioc for proteomics visualisation</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   For a full list of available documents:</span></span>
<span><span class="co">##     vignette(package='RforProteomics')</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'RforProteomics'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     spectrum</span></span></code></pre>
<p>See also the <em><a href="https://bioconductor.org/packages/3.17/RforProteomics" class="external-link">RforProteomics</a></em>
<a href="http://lgatto.github.io/RforProteomics/" class="external-link">web page</a> for more
information on installation.</p>
</div>
<div class="section level3">
<h3 id="external-dependencies">External dependencies<a class="anchor" aria-label="anchor" href="#external-dependencies"></a>
</h3>
<p>Some packages used in the document depend on external libraries that
need to be installed prior to the R packages:</p>
<ul>
<li>
<em><a href="https://bioconductor.org/packages/3.17/mzR" class="external-link">mzR</a></em> depends
on the <a href="http://cdf.gsfc.nasa.gov/" class="external-link">Common Data Format</a> (CDF)
to CDF-based raw mass-spectrometry data. On Linux, the
<code>libcdf</code> library is required. On Debian-based systems, for
instance, one needs to install the <code>libnetcdf-dev</code>
package.</li>
<li>several packages depend on the <em><a href="https://CRAN.R-project.org/package=XML" class="external-link">XML</a></em> package which
requires the <code>libxml2</code> infrastructure on Linux. On
Debian-based systems, one needs to install
<code>libxml2-dev</code>.</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/biomaRt" class="external-link">biomaRt</a></em>
performs on-line requests using the <a href="http://curl.haxx.se/" class="external-link"><code>curl</code></a> infrastructure. On
Debian-based systems, you one needs to install <code>libcurl-dev</code>
or <code>libcurl4-openssl-dev</code>.</li>
<li>
<em><a href="https://bioconductor.org/packages/3.17/MSGFplus" class="external-link">MSGFplus</a></em>
is based on the <code>MS-GF+</code> java program and thus requires <a href="https://java.com" class="external-link">Java 1.7</a> in order to work.</li>
</ul>
</div>
<div class="section level3">
<h3 id="obtaining-the-code">Obtaining the code<a class="anchor" aria-label="anchor" href="#obtaining-the-code"></a>
</h3>
<p>The code in this document describes all the examples presented in
<span class="citation">(Laurent Gatto and Christoforou 2013)</span> and
can be copy, pasted and executed. It is however more convenient to have
it in a separate text file for better interaction with R to easily
modify and explore it. This can be achieved with the
<code>Stangle</code> function. One needs the Sweave source of this
document (a document combining the narration and the R code) and the
<code>Stangle</code> then specifically extracts the code chunks and
produces a clean R source file. If the package is installed, the
following code chunk will direct you to the
<code>RforProteomics.R</code> file containing all the annotated source
code contained in this document.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## gets the vignette source</span></span>
<span><span class="va">rfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"doc/RforProteomics.R"</span>,</span>
<span>                     package <span class="op">=</span> <span class="st">"RforProteomics"</span><span class="op">)</span></span>
<span><span class="va">rfile</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] ""</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="prepare-the-working-environment">Prepare the working environment<a class="anchor" aria-label="anchor" href="#prepare-the-working-environment"></a>
</h3>
<p>The packages that we will depend on to execute the examples will be
loaded in the respective sections. Here, we pre-load packages that
provide general functionality used throughout the document.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"RColorBrewer"</span><span class="op">)</span> <span class="co">## Color palettes</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span>  <span class="co">## Convenient and nice plotting</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/hadley/reshape" class="external-link">"reshape2"</a></span><span class="op">)</span> <span class="co">## Flexibly reshape data</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="data-standards-and-inputoutput">Data standards and input/output<a class="anchor" aria-label="anchor" href="#data-standards-and-inputoutput"></a>
</h2>
<div class="section level3">
<h3 id="sec:mzr">The mzR package<a class="anchor" aria-label="anchor" href="#sec:mzr"></a>
</h3>
<div class="section level4">
<h4 id="raw-ms-data">Raw MS data<a class="anchor" aria-label="anchor" href="#raw-ms-data"></a>
</h4>
<p>The <em><a href="https://bioconductor.org/packages/3.17/mzR" class="external-link">mzR</a></em> package
<span class="citation">(Chambers et al. 2012)</span> provides a unified
interface to various mass spectrometry open formats. This code chunk,
taken from the <code>openMSfile</code> documentation, illustrated how to
open a connection to an raw data file. The example <code>mzML</code>
data is taken from the <em><a href="https://bioconductor.org/packages/3.17/msdata" class="external-link">msdata</a></em>
data package. The code below would also be applicable to an
<code>mzXML</code>, <code>mzData</code> or <code>netCDF</code> file.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## load the required packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/sneumann/mzR/" class="external-link">"mzR"</a></span><span class="op">)</span> <span class="co">## the software package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"msdata"</span><span class="op">)</span> <span class="co">## the data package</span></span>
<span><span class="co">## below, we extract the releavant example file</span></span>
<span><span class="co">## from the local 'msdata' installation</span></span>
<span><span class="va">filepath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"microtofq"</span>, package <span class="op">=</span> <span class="st">"msdata"</span><span class="op">)</span></span>
<span><span class="va">file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="va">filepath</span>, pattern<span class="op">=</span><span class="st">"MM14.mzML"</span>,</span>
<span>                   full.names<span class="op">=</span><span class="cn">TRUE</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">## creates a commection to the mzML file</span></span>
<span><span class="va">mz</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mzR/man/openMSfile.html" class="external-link">openMSfile</a></span><span class="op">(</span><span class="va">file</span><span class="op">)</span></span>
<span><span class="co">## demonstraction of data access</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/path.html" class="external-link">basename</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/fileName.html" class="external-link">fileName</a></span><span class="op">(</span><span class="va">mz</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "MM14.mzML"</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/mzR/man/metadata.html" class="external-link">runInfo</a></span><span class="op">(</span><span class="va">mz</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $scanCount</span></span>
<span><span class="co">## [1] 112</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $lowMz</span></span>
<span><span class="co">## [1] 0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $highMz</span></span>
<span><span class="co">## [1] 0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $dStartTime</span></span>
<span><span class="co">## [1] 270.334</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $dEndTime</span></span>
<span><span class="co">## [1] 307.678</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $msLevels</span></span>
<span><span class="co">## [1] 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $startTimeStamp</span></span>
<span><span class="co">## [1] NA</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/mzR/man/metadata.html" class="external-link">instrumentInfo</a></span><span class="op">(</span><span class="va">mz</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $manufacturer</span></span>
<span><span class="co">## [1] "Unknown"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $model</span></span>
<span><span class="co">## [1] "instrument model"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $ionisation</span></span>
<span><span class="co">## [1] "electrospray ionization"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $analyzer</span></span>
<span><span class="co">## [1] "mass analyzer type"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $detector</span></span>
<span><span class="co">## [1] "detector type"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $software</span></span>
<span><span class="co">## [1] "so_in_0 "</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $sample</span></span>
<span><span class="co">## [1] "MM14_20uMsa_0"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $source</span></span>
<span><span class="co">## [1] ""</span></span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## once finished, it is good to explicitely</span></span>
<span><span class="co">## close the connection</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/mzR/man/mzR-class.html" class="external-link">close</a></span><span class="op">(</span><span class="va">mz</span><span class="op">)</span></span></code></pre></div>
<p><em><a href="https://bioconductor.org/packages/3.17/mzR" class="external-link">mzR</a></em>
is used by other packages, like <em><a href="https://bioconductor.org/packages/3.17/MSnbase" class="external-link">MSnbase</a></em>
<span class="citation">(L. Gatto and Lilley 2012)</span>, <em><a href="https://bioconductor.org/packages/3.17/TargetSearch" class="external-link">TargetSearch</a></em>
<span class="citation">(Cuadros-Inostroza et al. 2009)</span> and <em><a href="https://bioconductor.org/packages/3.17/xcms" class="external-link">xcms</a></em> <span class="citation">(Smith et al. 2006, Benton2008, Tautenhahn2008)</span>,
that provide a higher level abstraction to the data.</p>
</div>
<div class="section level4">
<h4 id="identification-data">Identification data<a class="anchor" aria-label="anchor" href="#identification-data"></a>
</h4>
<p>The <em><a href="https://bioconductor.org/packages/3.17/mzR" class="external-link">mzR</a></em> package
also provides very fast access to <code>mzIdentML</code> data by
leveraging proteowizard’s <code>C++</code> parser.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"mzid"</span>, <span class="st">"Tandem.mzid.gz"</span>, package<span class="op">=</span><span class="st">"msdata"</span><span class="op">)</span></span>
<span><span class="va">mzid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mzR/man/openMSfile.html" class="external-link">openIDfile</a></span><span class="op">(</span><span class="va">file</span><span class="op">)</span></span>
<span><span class="va">mzid</span></span></code></pre></div>
<pre><code><span><span class="co">## Identification file handle.</span></span>
<span><span class="co">## Filename:  Tandem.mzid.gz </span></span>
<span><span class="co">## Number of psms:  171</span></span></code></pre>
<p>Once and <code>mzRident</code> identification file handle has been
established, various data and metadata can be extracted, as illustrated
below.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/mzR/man/metadata.html" class="external-link">softwareInfo</a></span><span class="op">(</span><span class="va">mzid</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "xtandem x! tandem CYCLONE (2010.06.01.5) "    </span></span>
<span><span class="co">## [2] "ProteoWizard MzIdentML 3.0.21263 ProteoWizard"</span></span></code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/mzR/man/metadata.html" class="external-link">enzymes</a></span><span class="op">(</span><span class="va">mzid</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##      name nTermGain cTermGain minDistance missedCleavages</span></span>
<span><span class="co">## 1 Trypsin         H        OH           0               1</span></span></code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">psms</a></span><span class="op">(</span><span class="va">mzid</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "spectrumID"               "chargeState"             </span></span>
<span><span class="co">##  [3] "rank"                     "passThreshold"           </span></span>
<span><span class="co">##  [5] "experimentalMassToCharge" "calculatedMassToCharge"  </span></span>
<span><span class="co">##  [7] "sequence"                 "peptideRef"              </span></span>
<span><span class="co">##  [9] "modNum"                   "isDecoy"                 </span></span>
<span><span class="co">## [11] "post"                     "pre"                     </span></span>
<span><span class="co">## [13] "start"                    "end"                     </span></span>
<span><span class="co">## [15] "DatabaseAccess"           "DBseqLength"             </span></span>
<span><span class="co">## [17] "DatabaseSeq"              "DatabaseDescription"     </span></span>
<span><span class="co">## [19] "spectrum.title"           "acquisitionNum"</span></span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">psms</a></span><span class="op">(</span><span class="va">mzid</span><span class="op">)</span><span class="op">)</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">13</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##   spectrumID chargeState rank passThreshold experimentalMassToCharge</span></span>
<span><span class="co">## 1   index=12           3    1         FALSE                 903.7209</span></span>
<span><span class="co">## 2  index=285           3    1         FALSE                 792.3792</span></span>
<span><span class="co">## 3   index=83           3    1         FALSE                 792.5295</span></span>
<span><span class="co">## 4   index=21           3    1         FALSE                 850.0782</span></span>
<span><span class="co">## 5  index=198           3    1         FALSE                 527.2592</span></span>
<span><span class="co">## 6   index=13           2    1         FALSE                 724.8816</span></span>
<span><span class="co">##   calculatedMassToCharge                 sequence</span></span>
<span><span class="co">## 1               903.4032  LCYIALDFDEEMKAAEDSSDIEK</span></span>
<span><span class="co">## 2               792.3899   KDLYGNVVLSGGTTMYEGIGER</span></span>
<span><span class="co">## 3               792.3899   KDLYGNVVLSGGTTMYEGIGER</span></span>
<span><span class="co">## 4               849.7635 VIDENFGLVEGLMTTVHAATGTQK</span></span>
<span><span class="co">## 5               527.2849          GVGGAIVLVLYDEMK</span></span>
<span><span class="co">## 6               724.3771            HAVGGRYSSLLCK</span></span>
<span><span class="co">##                                            peptideRef modNum isDecoy post pre</span></span>
<span><span class="co">## 1 LCYIALDFDEEMKAAEDSSDIEK_15.9949@M$12;_57.0215@C$2;_      2   FALSE    S   K</span></span>
<span><span class="co">## 2              KDLYGNVVLSGGTTMYEGIGER_15.9949@M$15;__      1   FALSE    L   R</span></span>
<span><span class="co">## 3              KDLYGNVVLSGGTTMYEGIGER_15.9949@M$15;__      1   FALSE    L   R</span></span>
<span><span class="co">## 4            VIDENFGLVEGLMTTVHAATGTQK_15.9949@M$13;__      1   FALSE    V   K</span></span>
<span><span class="co">## 5                     GVGGAIVLVLYDEMK_15.9949@M$14;__      1   FALSE    R   R</span></span>
<span><span class="co">## 6                       HAVGGRYSSLLCK__57.0215@C$12;_      1    TRUE    D   K</span></span>
<span><span class="co">##   start</span></span>
<span><span class="co">## 1   217</span></span>
<span><span class="co">## 2   292</span></span>
<span><span class="co">## 3   292</span></span>
<span><span class="co">## 4   842</span></span>
<span><span class="co">## 5   297</span></span>
<span><span class="co">## 6   392</span></span></code></pre>
</div>
</div>
<div class="section level3">
<h3 id="subsec:mzID">Handling MS2 identification data with <em><a href="https://bioconductor.org/packages/3.17/mzID" class="external-link">mzID</a></em><a class="anchor" aria-label="anchor" href="#subsec:mzID"></a>
</h3>
<p>The <em><a href="https://bioconductor.org/packages/3.17/mzID" class="external-link">mzID</a></em> package
allows to load and manipulate MS2 data in the <code>mzIdentML</code>
format. The main <code>mzID</code> function reads such a file and
constructs an instance of class <code>mzID</code>.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"mzID"</span><span class="op">)</span></span>
<span><span class="va">mzids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>, package <span class="op">=</span> <span class="st">'mzID'</span><span class="op">)</span>,</span>
<span>                    pattern <span class="op">=</span> <span class="st">'*.mzid'</span>, full.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">mzids</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "/usr/local/lib/R/host-site-library/mzID/extdata/55merge_omssa.mzid"                      </span></span>
<span><span class="co">## [2] "/usr/local/lib/R/host-site-library/mzID/extdata/55merge_tandem.mzid"                     </span></span>
<span><span class="co">## [3] "/usr/local/lib/R/host-site-library/mzID/extdata/Mascot_MSMS_example.mzid"                </span></span>
<span><span class="co">## [4] "/usr/local/lib/R/host-site-library/mzID/extdata/Mascot_MSMS_example1.0.mzid"             </span></span>
<span><span class="co">## [5] "/usr/local/lib/R/host-site-library/mzID/extdata/Mascot_NA_example.mzid"                  </span></span>
<span><span class="co">## [6] "/usr/local/lib/R/host-site-library/mzID/extdata/mascot_pmf_example.mzid"                 </span></span>
<span><span class="co">## [7] "/usr/local/lib/R/host-site-library/mzID/extdata/Mascot_top_down_example.mzid"            </span></span>
<span><span class="co">## [8] "/usr/local/lib/R/host-site-library/mzID/extdata/MPC_example_Multiple_search_engines.mzid"</span></span>
<span><span class="co">## [9] "/usr/local/lib/R/host-site-library/mzID/extdata/Sequest_example_ver1.1.mzid"</span></span></code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">id</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mzID/man/mzID.html" class="external-link">mzID</a></span><span class="op">(</span><span class="va">mzids</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## reading 55merge_omssa.mzid... DONE!</span></span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">id</span></span></code></pre></div>
<pre><code><span><span class="co">## An mzID object</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Software used:   OMSSA (version: NA)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Rawfile:         D:/TestSpace/NeoTestMarch2011/55merge.mgf</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Database:        D:/Software/Databases/Neospora_3rndTryp/Neo_rndTryp_3times.fasta</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Number of scans: 39</span></span>
<span><span class="co">## Number of PSM's: 99</span></span></code></pre>
<p>Multiple files can be parsed in one go, possibly in parallel if the
environment supports it. When this is done an mzIDCollection object is
returned:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mzID/man/mzID.html" class="external-link">mzID</a></span><span class="op">(</span><span class="va">mzids</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">ids</span></span></code></pre></div>
<pre><code><span><span class="co">## An mzIDCollection object containing 2 samples</span></span></code></pre>
<p>Peptides, scans, parameters, … can be extracted with the respective
<code>peptides</code>, <code>scans</code>, <code>parameters</code>, …
functions. The <code>mzID</code> object can also be converted into a
<code>data.frame</code> using the <code>flatten</code> function.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mzID/man/flatten.html" class="external-link">flatten</a></span><span class="op">(</span><span class="va">id</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">fid</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "spectrumid"               "spectrum title"          </span></span>
<span><span class="co">##  [3] "acquisitionnum"           "passthreshold"           </span></span>
<span><span class="co">##  [5] "rank"                     "calculatedmasstocharge"  </span></span>
<span><span class="co">##  [7] "experimentalmasstocharge" "chargestate"             </span></span>
<span><span class="co">##  [9] "omssa:evalue"             "omssa:pvalue"            </span></span>
<span><span class="co">## [11] "isdecoy"                  "post"                    </span></span>
<span><span class="co">## [13] "pre"                      "end"                     </span></span>
<span><span class="co">## [15] "start"                    "accession"               </span></span>
<span><span class="co">## [17] "length"                   "description"             </span></span>
<span><span class="co">## [19] "pepseq"                   "modified"                </span></span>
<span><span class="co">## [21] "modification"             "idFile"                  </span></span>
<span><span class="co">## [23] "spectrumFile"             "databaseFile"</span></span></code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">fid</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 101  24</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="raw-data-abstraction-with-msnexp-objects">Raw data abstraction with <code>MSnExp</code> objects<a class="anchor" aria-label="anchor" href="#raw-data-abstraction-with-msnexp-objects"></a>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.17/MSnbase" class="external-link">MSnbase</a></em>
<span class="citation">(L. Gatto and Lilley 2012)</span> provides base
functions and classes for MS-based proteomics that allow facile data and
meta-data processing, manipulation and plotting (see for instance figure
below).</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://lgatto.github.io/MSnbase" class="external-link">"MSnbase"</a></span><span class="op">)</span></span>
<span><span class="co">## uses a simple dummy test included in the package</span></span>
<span><span class="va">mzXML</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">dir</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"MSnbase"</span>,dir<span class="op">=</span><span class="st">"extdata"</span><span class="op">)</span>,</span>
<span>             full.name<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>             pattern<span class="op">=</span><span class="st">"mzXML$"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/path.html" class="external-link">basename</a></span><span class="op">(</span><span class="va">mzXML</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "dummyiTRAQ.mzXML"</span></span></code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## reads the raw data into and MSnExp instance</span></span>
<span><span class="va">raw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/readMSData.html" class="external-link">readMSData</a></span><span class="op">(</span><span class="va">mzXML</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span>, centroided <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">raw</span></span></code></pre></div>
<pre><code><span><span class="co">## MSn experiment data ("MSnExp")</span></span>
<span><span class="co">## Object size in memory: 0.18 Mb</span></span>
<span><span class="co">## - - - Spectra data - - -</span></span>
<span><span class="co">##  MS level(s): 2 </span></span>
<span><span class="co">##  Number of spectra: 5 </span></span>
<span><span class="co">##  MSn retention times: 25:01 - 25:02 minutes</span></span>
<span><span class="co">## - - - Processing information - - -</span></span>
<span><span class="co">## Data loaded: Fri Jan 20 07:59:42 2023 </span></span>
<span><span class="co">##  MSnbase version: 2.25.1 </span></span>
<span><span class="co">## - - - Meta data  - - -</span></span>
<span><span class="co">## phenoData</span></span>
<span><span class="co">##   rowNames: dummyiTRAQ.mzXML</span></span>
<span><span class="co">##   varLabels: sampleNames</span></span>
<span><span class="co">##   varMetadata: labelDescription</span></span>
<span><span class="co">## Loaded from:</span></span>
<span><span class="co">##   dummyiTRAQ.mzXML </span></span>
<span><span class="co">## protocolData: none</span></span>
<span><span class="co">## featureData</span></span>
<span><span class="co">##   featureNames: F1.S1 F1.S2 ... F1.S5 (5 total)</span></span>
<span><span class="co">##   fvarLabels: spectrum</span></span>
<span><span class="co">##   fvarMetadata: labelDescription</span></span>
<span><span class="co">## experimentData: use 'experimentData(object)'</span></span></code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Extract a single spectrum</span></span>
<span><span class="va">raw</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## Object of class "Spectrum2"</span></span>
<span><span class="co">##  Precursor: 645.3741 </span></span>
<span><span class="co">##  Retention time: 25:02 </span></span>
<span><span class="co">##  Charge: 2 </span></span>
<span><span class="co">##  MSn level: 2 </span></span>
<span><span class="co">##  Peaks count: 2125 </span></span>
<span><span class="co">##  Total ion count: 150838188</span></span></code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">raw</span>, full <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">raw</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span>, full <span class="op">=</span> <span class="cn">TRUE</span>, reporters <span class="op">=</span> <span class="va">iTRAQ4</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="RforProteomics_files/figure-html/msnexp-1.png" alt="The `plot` method can be used on experiments, i.e. spectrum collections (top), or individual spectra (bottom)." width="700"><p class="caption">
The <code>plot</code> method can be used on experiments, i.e. spectrum
collections (top), or individual spectra (bottom).
</p>
</div>
<div class="section level3">
<h3 id="mgf-readwrite-support">
<code>mgf</code> read/write support<a class="anchor" aria-label="anchor" href="#mgf-readwrite-support"></a>
</h3>
<p>Read and write support for data in the <a href="http://www.matrixscience.com/help/data_file_help.html#GEN" class="external-link"><code>mgf</code></a>
and <code>mzTab</code> formats are available via the
<code>readMgfData</code>/<code>writeMgfData</code> and
<code>readMzTabData</code>/<code>writeMzTabData</code> functions,
respectively. An example for the latter is shown in the next
section.</p>
</div>
</div>
<div class="section level2">
<h2 id="quantitative-proteomics">Quantitative proteomics<a class="anchor" aria-label="anchor" href="#quantitative-proteomics"></a>
</h2>
<p>As an running example throughout this document, we will use a TMT
6-plex data set, <code>PXD000001</code> to illustrate quantitative data
processing. The code chunk below first downloads this data file from the
ProteomeXchange server using the <em><a href="https://bioconductor.org/packages/3.17/rpx" class="external-link">rpx</a></em>
package.</p>
<div class="section level3">
<h3 id="the-mztab-format">The <code>mzTab</code> format<a class="anchor" aria-label="anchor" href="#the-mztab-format"></a>
</h3>
<p>The first code chunk downloads the <code>mzTab</code> data from the
ProteomeXchange repository <span class="citation">(Vizcaino et al.
2014)</span>.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Experiment information</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/lgatto/rpx" class="external-link">"rpx"</a></span><span class="op">)</span></span>
<span><span class="va">px1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/rpx/man/PXDataset2.html" class="external-link">PXDataset</a></span><span class="op">(</span><span class="st">"PXD000001"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading PXD000001 from cache.</span></span></code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">px1</span></span></code></pre></div>
<pre><code><span><span class="co">## Project PXD000001 with 10 files</span></span>
<span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">## Resource ID BFC1 in cache in /github/home/.cache/R/rpx.</span></span></code></pre>
<pre><code><span><span class="co">##  [1] 'erwinia_carotovora.fasta' ... [10] 'PRIDE_Exp_Complete_Ac_22134.pride.mgf.gz'</span></span>
<span><span class="co">##  Use 'pxfiles(.)' to see all files.</span></span></code></pre>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/rpx/man/PXDataset2.html" class="external-link">pxfiles</a></span><span class="op">(</span><span class="va">px1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Project PXD000001 files (10):</span></span>
<span><span class="co">##  [remote] erwinia_carotovora.fasta</span></span>
<span><span class="co">##  [remote] F063721.dat</span></span>
<span><span class="co">##  [local]  F063721.dat-mztab.txt</span></span>
<span><span class="co">##  [remote] PRIDE_Exp_Complete_Ac_22134.xml.gz</span></span>
<span><span class="co">##  [remote] TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01.raw</span></span>
<span><span class="co">##  [local]  TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01.mzXML</span></span>
<span><span class="co">##  [remote] TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzXML</span></span>
<span><span class="co">##  [local]  TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzML</span></span>
<span><span class="co">##  [remote] PRIDE_Exp_Complete_Ac_22134.pride.mztab.gz</span></span>
<span><span class="co">##  [remote] PRIDE_Exp_Complete_Ac_22134.pride.mgf.gz</span></span></code></pre>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Downloading the mzTab data</span></span>
<span><span class="va">mztab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/rpx/man/PXDataset2.html" class="external-link">pxget</a></span><span class="op">(</span><span class="va">px1</span>, <span class="st">"F063721.dat-mztab.txt"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading F063721.dat-mztab.txt from cache.</span></span></code></pre>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mztab</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "/github/home/.cache/R/rpx/101d63c3861e_F063721.dat-mztab.txt"</span></span></code></pre>
<p>The code below loads the <code>mzTab</code> file into R and generates
an <code>MSnSet</code> instance<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>, removes missing values and calculates
protein intensities by summing the peptide quantitation data. The figure
below illustrates the intensities for 5 proteins.</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load mzTab peptide data</span></span>
<span><span class="va">qnt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/readMzTabData.html" class="external-link">readMzTabData</a></span><span class="op">(</span><span class="va">mztab</span>, what <span class="op">=</span> <span class="st">"PEP"</span>, version <span class="op">=</span> <span class="st">"0.9"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Version 0.9 is deprecated. Please see '?readMzTabData' and '?MzTab' for</span></span>
<span><span class="co">## details.</span></span></code></pre>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html" class="external-link">sampleNames</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/ReporterIons-class.html" class="external-link">reporterNames</a></span><span class="op">(</span><span class="va">TMT6</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   TMT6.126 TMT6.127 TMT6.128 TMT6.129 TMT6.130 TMT6.131</span></span>
<span><span class="co">## 1 10630132 11238708 12424917 10997763  9928972 10398534</span></span>
<span><span class="co">## 2 11105690 12403253 13160903 12229367 11061660 10131218</span></span>
<span><span class="co">## 3  1183431  1322371  1599088  1243715  1306602  1159064</span></span>
<span><span class="co">## 4  5384958  5508454  6883086  6136023  5626680  5213771</span></span>
<span><span class="co">## 5 18033537 17926487 21052620 19810368 17381162 17268329</span></span>
<span><span class="co">## 6  9873585 10299931 11142071 10258214  9664315  9518271</span></span></code></pre>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## remove missing values</span></span>
<span><span class="va">qnt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">filterNA</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">processingData</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## - - - Processing information - - -</span></span>
<span><span class="co">## mzTab read: Fri Jan 20 07:59:44 2023 </span></span>
<span><span class="co">## Subset [1528,6][1528,6] Fri Jan 20 07:59:44 2023 </span></span>
<span><span class="co">## Removed features with more than 0 NAs: Fri Jan 20 07:59:44 2023 </span></span>
<span><span class="co">## Dropped featureData's levels Fri Jan 20 07:59:44 2023 </span></span>
<span><span class="co">##  MSnbase version: 2.25.1</span></span></code></pre>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## combine into proteins</span></span>
<span><span class="co">## - using the 'accession' feature meta data</span></span>
<span><span class="co">## - sum the peptide intensities</span></span>
<span><span class="va">protqnt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">combineFeatures</a></span><span class="op">(</span><span class="va">qnt</span>,</span>
<span>                           groupBy <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">$</span><span class="va">accession</span>,</span>
<span>                           method <span class="op">=</span> <span class="va">sum</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span><span class="fl">5</span>, <span class="st">"Set1"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/matplot.html" class="external-link">matplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">protqnt</span><span class="op">)</span>, n <span class="op">=</span> <span class="fl">5</span><span class="op">)</span><span class="op">)</span>, type <span class="op">=</span> <span class="st">"b"</span>,</span>
<span>        lty <span class="op">=</span> <span class="fl">1</span>, col <span class="op">=</span> <span class="va">cls</span>,</span>
<span>        ylab <span class="op">=</span> <span class="st">"Protein intensity (summed peptides)"</span>,</span>
<span>        xlab <span class="op">=</span> <span class="st">"TMT reporters"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></span><span class="op">(</span><span class="st">"topright"</span>, <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html" class="external-link">featureNames</a></span><span class="op">(</span><span class="va">protqnt</span><span class="op">)</span>, n<span class="op">=</span><span class="fl">5</span><span class="op">)</span>,</span>
<span>       lty <span class="op">=</span> <span class="fl">1</span>, bty <span class="op">=</span> <span class="st">"n"</span>, cex <span class="op">=</span> <span class="fl">.8</span>, col <span class="op">=</span> <span class="va">cls</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="RforProteomics_files/figure-html/matplot-1.png" alt="Protein quantitation data." width="700"><p class="caption">
Protein quantitation data.
</p>
</div>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qntS</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/normalise-methods.html" class="external-link">normalise</a></span><span class="op">(</span><span class="va">qnt</span>, <span class="st">"sum"</span><span class="op">)</span></span>
<span><span class="va">qntV</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/normalise-methods.html" class="external-link">normalise</a></span><span class="op">(</span><span class="va">qntS</span>, <span class="st">"vsn"</span><span class="op">)</span></span>
<span><span class="va">qntV2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/normalise-methods.html" class="external-link">normalise</a></span><span class="op">(</span><span class="va">qnt</span>, <span class="st">"vsn"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">acc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"P00489"</span>, <span class="st">"P00924"</span>,</span>
<span>         <span class="st">"P02769"</span>, <span class="st">"P62894"</span>,</span>
<span>         <span class="st">"ECA"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">idx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">acc</span>, <span class="va">grep</span>, <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">$</span><span class="va">accession</span><span class="op">)</span></span>
<span><span class="va">idx2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">idx</span>, <span class="va">head</span>, <span class="fl">3</span><span class="op">)</span></span>
<span><span class="va">small</span> <span class="op">&lt;-</span> <span class="va">qntS</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">idx2</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="va">idx3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">idx</span>, <span class="va">head</span>, <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">medium</span> <span class="op">&lt;-</span> <span class="va">qntV</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">idx3</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">medium</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"126"</span>, <span class="st">"127"</span>, <span class="st">"128"</span>,</span>
<span>                 <span class="st">"129"</span>, <span class="st">"130"</span>, <span class="st">"131"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">medium</span><span class="op">)</span><span class="op">$</span><span class="va">accession</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"CYC"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"CYT"</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"ENO"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"ENO"</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"ALB"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"BSA"</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"PYGM"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"PHO"</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"ECA"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Background"</span></span>
<span></span>
<span><span class="va">cls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">-</span><span class="fl">1</span>, <span class="st">"Set1"</span><span class="op">)</span>,</span>
<span>         <span class="st">"grey"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cls</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">wbcol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html" class="external-link">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"white"</span>, <span class="st">"darkblue"</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fl">256</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/heatmap.html" class="external-link">heatmap</a></span><span class="op">(</span><span class="va">m</span>, col <span class="op">=</span> <span class="va">wbcol</span>, RowSideColors<span class="op">=</span><span class="va">cls</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">m</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="RforProteomics_files/figure-html/heatmap-1.png" alt="A heatmap." width="700"><p class="caption">
A heatmap.
</p>
</div>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dfr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">small</span><span class="op">)</span>,</span>
<span>                  Protein <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">small</span><span class="op">)</span><span class="op">$</span><span class="va">accession</span><span class="op">)</span>,</span>
<span>                  Feature <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html" class="external-link">featureNames</a></span><span class="op">(</span><span class="va">small</span><span class="op">)</span>,</span>
<span>                  stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">dfr</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"126"</span>, <span class="st">"127"</span>, <span class="st">"128"</span>, <span class="st">"129"</span>, <span class="st">"130"</span>, <span class="st">"131"</span>,</span>
<span>                   <span class="st">"Protein"</span>, <span class="st">"Feature"</span><span class="op">)</span></span>
<span><span class="va">dfr</span><span class="op">$</span><span class="va">Protein</span><span class="op">[</span><span class="va">dfr</span><span class="op">$</span><span class="va">Protein</span> <span class="op">==</span> <span class="st">"sp|P00924|ENO1_YEAST"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"ENO"</span></span>
<span><span class="va">dfr</span><span class="op">$</span><span class="va">Protein</span><span class="op">[</span><span class="va">dfr</span><span class="op">$</span><span class="va">Protein</span> <span class="op">==</span> <span class="st">"sp|P62894|CYC_BOVIN"</span><span class="op">]</span>  <span class="op">&lt;-</span> <span class="st">"CYT"</span></span>
<span><span class="va">dfr</span><span class="op">$</span><span class="va">Protein</span><span class="op">[</span><span class="va">dfr</span><span class="op">$</span><span class="va">Protein</span> <span class="op">==</span> <span class="st">"sp|P02769|ALBU_BOVIN"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"BSA"</span></span>
<span><span class="va">dfr</span><span class="op">$</span><span class="va">Protein</span><span class="op">[</span><span class="va">dfr</span><span class="op">$</span><span class="va">Protein</span> <span class="op">==</span> <span class="st">"sp|P00489|PYGM_RABIT"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"PHO"</span></span>
<span><span class="va">dfr</span><span class="op">$</span><span class="va">Protein</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"ECA"</span>, <span class="va">dfr</span><span class="op">$</span><span class="va">Protein</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Background"</span></span>
<span><span class="va">dfr2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html" class="external-link">melt</a></span><span class="op">(</span><span class="va">dfr</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Using Protein, Feature as id variables</span></span></code></pre>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">variable</span>, y <span class="op">=</span> <span class="va">value</span>, colour <span class="op">=</span> <span class="va">Protein</span><span class="op">)</span>,</span>
<span>       data <span class="op">=</span> <span class="va">dfr2</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>group<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">Feature</span><span class="op">)</span><span class="op">)</span>, alpha <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span><span class="va">.</span> <span class="op">~</span> <span class="va">Protein</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position<span class="op">=</span><span class="st">"none"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"Reporters"</span>, y <span class="op">=</span> <span class="st">"Normalised intensity"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="RforProteomics_files/figure-html/spikes-1.png" alt="Spikes plot using `r CRANpkg('ggplot2')`." width="700"><p class="caption">
Spikes plot using <code>r CRANpkg('ggplot2')</code>.
</p>
</div>
</div>
<div class="section level3">
<h3 id="third-party-data">Third-party data<a class="anchor" aria-label="anchor" href="#third-party-data"></a>
</h3>
<p>It is possible to import any arbitrary text-based spreadsheet as
<em>MSnSet</em> object using either <code>readMSnSet</code> or
<code>readMSnSet2</code>. The former takes three spreadsheets as input
(for the expression data and the feature and sample meta-data). The
latter uses a single spreadsheet and a vector of expression columns to
populate the assay data and the feature meta-data. Detailed examples are
provided in the <code>MSnbase-io</code> vignette, that can be consulted
from R with <code>vignette("MSnbase-io")</code> or <a href="https://bioconductor.org/packages/release/bioc/vignettes/MSnbase/inst/doc/MSnbase-io.html" class="external-link">online</a>.</p>
</div>
<div class="section level3">
<h3 id="working-with-raw-data">Working with raw data<a class="anchor" aria-label="anchor" href="#working-with-raw-data"></a>
</h3>
<p>We reuse our dedicated <code>px1</code> ProteomeXchange data object
to download the raw data (in <code>mzXML</code> format) and load it with
the <code>readMSData</code> from the <em><a href="https://bioconductor.org/packages/3.17/MSnbase" class="external-link">MSnbase</a></em>
package that produces a raw data experiment object of class
<code>MSnExp</code> (a new <em>on-disk</em> infrastructure is now
available to access the raw data on disk on demand, rather than loading
it all in memory, enabling the management of more and larger files - see
the <code>benchmarking</code> vignette in the <em><a href="https://bioconductor.org/packages/3.17/MSnbase" class="external-link">MSnbase</a></em>
package for details). The raw data is then quantified using the
<code>quantify</code> method specifying the TMT 6-plex isobaric tags and
a 7th peak of interest corresponding to the un-dissociated reporter tag
peaks (see the <code>MSnbase-demo</code> vignette in <em><a href="https://bioconductor.org/packages/3.17/MSnbase" class="external-link">MSnbase</a></em>
for details).</p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mzxml</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/rpx/man/PXDataset2.html" class="external-link">pxget</a></span><span class="op">(</span><span class="va">px1</span>, <span class="st">"TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01.mzXML"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01.mzXML from cache.</span></span></code></pre>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rawms</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/readMSData.html" class="external-link">readMSData</a></span><span class="op">(</span><span class="va">mzxml</span>, centroided <span class="op">=</span> <span class="cn">TRUE</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">qntms</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">quantify</a></span><span class="op">(</span><span class="va">rawms</span>, reporters <span class="op">=</span> <span class="va">TMT7</span>, method <span class="op">=</span> <span class="st">"max"</span><span class="op">)</span></span>
<span><span class="va">qntms</span></span></code></pre></div>
<pre><code><span><span class="co">## MSnSet (storageMode: lockedEnvironment)</span></span>
<span><span class="co">## assayData: 6103 features, 7 samples </span></span>
<span><span class="co">##   element names: exprs </span></span>
<span><span class="co">## protocolData: none</span></span>
<span><span class="co">## phenoData</span></span>
<span><span class="co">##   sampleNames: TMT7.126 TMT7.127 ... TMT7.230 (7 total)</span></span>
<span><span class="co">##   varLabels: mz reporters</span></span>
<span><span class="co">##   varMetadata: labelDescription</span></span>
<span><span class="co">## featureData</span></span>
<span><span class="co">##   featureNames: F1.S0001 F1.S0002 ... F1.S6103 (6103 total)</span></span>
<span><span class="co">##   fvarLabels: spectrum fileIdx ... collision.energy (12 total)</span></span>
<span><span class="co">##   fvarMetadata: labelDescription</span></span>
<span><span class="co">## experimentData: use 'experimentData(object)'</span></span>
<span><span class="co">## Annotation: No annotation </span></span>
<span><span class="co">## - - - Processing information - - -</span></span>
<span><span class="co">## Data loaded: Fri Jan 20 08:00:22 2023 </span></span>
<span><span class="co">## TMT7 quantification by max: Fri Jan 20 08:01:54 2023 </span></span>
<span><span class="co">##  MSnbase version: 2.25.1</span></span></code></pre>
<p>Identification data in the <code>mzIdentML</code> format can be added
to <code>MSnExp</code> or <code>MSnSet</code> instances with the
<code>addIdentificationData</code> function. See the function
documentation for examples.</p>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Signal <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">qntms</span><span class="op">)</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                Incomplete <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">qntms</span><span class="op">)</span><span class="op">[</span>, <span class="fl">7</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span></span>
<span><span class="va">cls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"#00000050"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pch</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">cls</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"P02769"</span>, <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">$</span><span class="va">accession</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"gold4"</span> <span class="co">## BSA</span></span>
<span><span class="va">cls</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"P00924"</span>, <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">$</span><span class="va">accession</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"dodgerblue"</span> <span class="co">## ENO</span></span>
<span><span class="va">cls</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"P62894"</span>, <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">$</span><span class="va">accession</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"springgreen4"</span> <span class="co">## CYT</span></span>
<span><span class="va">cls</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"P00489"</span>, <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">$</span><span class="va">accession</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"darkorchid2"</span> <span class="co">## PHO</span></span>
<span><span class="va">pch</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"P02769"</span>, <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">$</span><span class="va">accession</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">19</span></span>
<span><span class="va">pch</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"P00924"</span>, <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">$</span><span class="va">accession</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">19</span></span>
<span><span class="va">pch</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"P62894"</span>, <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">$</span><span class="va">accession</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">19</span></span>
<span><span class="va">pch</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"P00489"</span>, <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">$</span><span class="va">accession</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">19</span></span></code></pre></div>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mzp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/plotMzDelta-methods.html" class="external-link">plotMzDelta</a></span><span class="op">(</span><span class="va">rawms</span>, reporters <span class="op">=</span> <span class="va">TMT6</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">""</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mzp</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Removed 2 rows containing missing values (`geom_bar()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 2 rows containing missing values (`geom_vline()`).</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 2 rows containing missing values (`geom_text()`).</span></span></code></pre>
<div class="figure">
<img src="RforProteomics_files/figure-html/plotmzdelta-1.png" alt="A m/z delta plot." width="700"><p class="caption">
A m/z delta plot.
</p>
</div>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">Signal</span> <span class="op">~</span> <span class="va">Incomplete</span>, data <span class="op">=</span> <span class="va">d</span>,</span>
<span>     xlab <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html" class="external-link">expression</a></span><span class="op">(</span><span class="va">Incomplete</span><span class="op">~</span><span class="va">dissociation</span><span class="op">)</span>,</span>
<span>     ylab <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/expression.html" class="external-link">expression</a></span><span class="op">(</span><span class="va">Sum</span><span class="op">~</span><span class="va">of</span><span class="op">~</span><span class="va">reporters</span><span class="op">~</span><span class="va">intensities</span><span class="op">)</span>,</span>
<span>     pch <span class="op">=</span> <span class="fl">19</span>,</span>
<span>     col <span class="op">=</span> <span class="st">"#4582B380"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/grid.html" class="external-link">grid</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, lty <span class="op">=</span> <span class="st">"dotted"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">Signal</span> <span class="op">~</span> <span class="va">Incomplete</span>, data <span class="op">=</span> <span class="va">d</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"darkblue"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="RforProteomics_files/figure-html/incompl-1.png" alt="Incomplete dissociation." width="700"><p class="caption">
Incomplete dissociation.
</p>
</div>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">MAplot</span><span class="op">(</span><span class="va">qnt</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">2</span><span class="op">)</span><span class="op">]</span>, cex <span class="op">=</span> <span class="fl">.9</span>, col <span class="op">=</span> <span class="va">cls</span>, pch <span class="op">=</span> <span class="va">pch</span>, show.statistics <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="RforProteomics_files/figure-html/maplot-1.png" alt="MAplot on an `MSnSet` instance." width="700"><p class="caption">
MAplot on an <code>MSnSet</code> instance.
</p>
</div>
</div>
<div class="section level3">
<h3 id="the-maldiquant-package">The <em><a href="https://CRAN.R-project.org/package=MALDIquant" class="external-link">MALDIquant</a></em>
package<a class="anchor" aria-label="anchor" href="#the-maldiquant-package"></a>
</h3>
<p>This section illustrates some of <em><a href="https://CRAN.R-project.org/package=MALDIquant" class="external-link">MALDIquant</a></em>’s
data processing capabilities <span class="citation">(Gibb and Strimmer
2012)</span>. The code is taken from the <code>processing-peaks.R</code>
script downloaded from the <a href="http://strimmerlab.org/software/maldiquant/" class="external-link">package
homepage</a>.</p>
<div class="section level4">
<h4 class="unnumbered" id="loading-the-data">Loading the data<a class="anchor" aria-label="anchor" href="#loading-the-data"></a>
</h4>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## load packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://strimmerlab.github.io/software/maldiquant/" class="external-link">"MALDIquant"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://strimmerlab.github.io/software/maldiquant/" class="external-link">"MALDIquantForeign"</a></span><span class="op">)</span></span>
<span><span class="co">## getting test data</span></span>
<span><span class="va">datapath</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"Examples"</span>,</span>
<span>                        package <span class="op">=</span> <span class="st">"readBrukerFlexData"</span><span class="op">)</span>,</span>
<span>            <span class="st">"2010_05_19_Gibb_C8_A1"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">dir</a></span><span class="op">(</span><span class="va">datapath</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "0_A1" "0_A2"</span></span></code></pre>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sA1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquantForeign/man/importBrukerFlex-functions.html" class="external-link">importBrukerFlex</a></span><span class="op">(</span><span class="va">datapath</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co"># in the following we use only the first spectrum</span></span>
<span><span class="va">s</span> <span class="op">&lt;-</span> <span class="va">sA1</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/AbstractMassObject-class.html" class="external-link">mass</a></span><span class="op">(</span><span class="va">s</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">##   999.9  2373.3  4331.4  4721.3  6874.2 10001.9</span></span></code></pre>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/AbstractMassObject-class.html" class="external-link">intensity</a></span><span class="op">(</span><span class="va">s</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">##       4     180    1562    2841    4656   32594</span></span></code></pre>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">s</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##           mass intensity</span></span>
<span><span class="co">## [1,]  999.9388     11278</span></span>
<span><span class="co">## [2,] 1000.1316     11350</span></span>
<span><span class="co">## [3,] 1000.3244     10879</span></span>
<span><span class="co">## [4,] 1000.5173     10684</span></span>
<span><span class="co">## [5,] 1000.7101     10740</span></span>
<span><span class="co">## [6,] 1000.9030     10947</span></span></code></pre>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">s</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="RforProteomics_files/figure-html/mqplot1-1.png" alt="Spectrum plotting in `r CRANpkg('MALDIquant')`." width="700"><p class="caption">
Spectrum plotting in <code>r CRANpkg('MALDIquant')</code>.
</p>
</div>
</div>
<div class="section level4">
<h4 class="unnumbered" id="preprocessing">{Preprocessing}<a class="anchor" aria-label="anchor" href="#preprocessing"></a>
</h4>
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## sqrt transform (for variance stabilization)</span></span>
<span><span class="va">s2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/transformIntensity-methods.html" class="external-link">transformIntensity</a></span><span class="op">(</span><span class="va">s</span>, method<span class="op">=</span><span class="st">"sqrt"</span><span class="op">)</span></span>
<span><span class="va">s2</span></span></code></pre></div>
<pre><code><span><span class="co">## S4 class type            : MassSpectrum            </span></span>
<span><span class="co">## Number of m/z values     : 22431                   </span></span>
<span><span class="co">## Range of m/z values      : 999.939 - 10001.925     </span></span>
<span><span class="co">## Range of intensity values: 2e+00 - 1.805e+02       </span></span>
<span><span class="co">## Memory usage             : 361.602 KiB             </span></span>
<span><span class="co">## Name                     : 2010_05_19_Gibb_C8_A1.A1</span></span>
<span><span class="co">## File                     : /usr/local/lib/R/host-site-library/readBrukerFlexData/Examples/2010_05_19_Gibb_C8_A1/0_A1/1/1SLin/fid</span></span></code></pre>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## smoothing - 5 point moving average</span></span>
<span><span class="va">s3</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/smoothIntensity-methods.html" class="external-link">smoothIntensity</a></span><span class="op">(</span><span class="va">s2</span>, method<span class="op">=</span><span class="st">"MovingAverage"</span>, halfWindowSize<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">s3</span></span></code></pre></div>
<pre><code><span><span class="co">## S4 class type            : MassSpectrum            </span></span>
<span><span class="co">## Number of m/z values     : 22431                   </span></span>
<span><span class="co">## Range of m/z values      : 999.939 - 10001.925     </span></span>
<span><span class="co">## Range of intensity values: 3.606e+00 - 1.792e+02   </span></span>
<span><span class="co">## Memory usage             : 361.602 KiB             </span></span>
<span><span class="co">## Name                     : 2010_05_19_Gibb_C8_A1.A1</span></span>
<span><span class="co">## File                     : /usr/local/lib/R/host-site-library/readBrukerFlexData/Examples/2010_05_19_Gibb_C8_A1/0_A1/1/1SLin/fid</span></span></code></pre>
<div class="sourceCode" id="cb97"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## baseline subtraction</span></span>
<span><span class="va">s4</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/removeBaseline-methods.html" class="external-link">removeBaseline</a></span><span class="op">(</span><span class="va">s3</span>, method<span class="op">=</span><span class="st">"SNIP"</span><span class="op">)</span></span>
<span><span class="va">s4</span></span></code></pre></div>
<pre><code><span><span class="co">## S4 class type            : MassSpectrum            </span></span>
<span><span class="co">## Number of m/z values     : 22431                   </span></span>
<span><span class="co">## Range of m/z values      : 999.939 - 10001.925     </span></span>
<span><span class="co">## Range of intensity values: 0e+00 - 1.404e+02       </span></span>
<span><span class="co">## Memory usage             : 361.602 KiB             </span></span>
<span><span class="co">## Name                     : 2010_05_19_Gibb_C8_A1.A1</span></span>
<span><span class="co">## File                     : /usr/local/lib/R/host-site-library/readBrukerFlexData/Examples/2010_05_19_Gibb_C8_A1/0_A1/1/1SLin/fid</span></span></code></pre>
</div>
<div class="section level4">
<h4 class="unnumbered" id="peak-picking">Peak picking<a class="anchor" aria-label="anchor" href="#peak-picking"></a>
</h4>
<div class="sourceCode" id="cb99"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## peak picking</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/detectPeaks-methods.html" class="external-link">detectPeaks</a></span><span class="op">(</span><span class="va">s4</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span> <span class="co"># 181</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 186</span></span></code></pre>
<div class="sourceCode" id="cb101"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peak.data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span> <span class="co"># extract peak information</span></span></code></pre></div>
<div class="sourceCode" id="cb102"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">xl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/range.html" class="external-link">range</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/AbstractMassObject-class.html" class="external-link">mass</a></span><span class="op">(</span><span class="va">s</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co"># use same xlim on all plots for better comparison</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">s</span>, sub<span class="op">=</span><span class="st">""</span>, main<span class="op">=</span><span class="st">"1: raw"</span>, xlim<span class="op">=</span><span class="va">xl</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">s2</span>, sub<span class="op">=</span><span class="st">""</span>, main<span class="op">=</span><span class="st">"2: variance stabilisation"</span>, xlim<span class="op">=</span><span class="va">xl</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">s3</span>, sub<span class="op">=</span><span class="st">""</span>, main<span class="op">=</span><span class="st">"3: smoothing"</span>, xlim<span class="op">=</span><span class="va">xl</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">s4</span>, sub<span class="op">=</span><span class="st">""</span>, main<span class="op">=</span><span class="st">"4: base line correction"</span>, xlim<span class="op">=</span><span class="va">xl</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">s4</span>, sub<span class="op">=</span><span class="st">""</span>, main<span class="op">=</span><span class="st">"5: peak detection"</span>, xlim<span class="op">=</span><span class="va">xl</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span>
<span><span class="va">top20</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/AbstractMassObject-class.html" class="external-link">intensity</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/AbstractMassObject-class.html" class="external-link">intensity</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span>, decreasing<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/labelPeaks-methods.html" class="external-link">labelPeaks</a></span><span class="op">(</span><span class="va">p</span>, index<span class="op">=</span><span class="va">top20</span>, underline<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">p</span>, sub<span class="op">=</span><span class="st">""</span>, main<span class="op">=</span><span class="st">"6: peak plot"</span>, xlim<span class="op">=</span><span class="va">xl</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/labelPeaks-methods.html" class="external-link">labelPeaks</a></span><span class="op">(</span><span class="va">p</span>, index<span class="op">=</span><span class="va">top20</span>, underline<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="RforProteomics_files/figure-html/mqplot-1.png" alt="Spectrum plotting in `r CRANpkg('MALDIquant')`." width="700"><p class="caption">
Spectrum plotting in <code>r CRANpkg('MALDIquant')</code>.
</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="working-with-peptide-sequences">Working with peptide sequences<a class="anchor" aria-label="anchor" href="#working-with-peptide-sequences"></a>
</h3>
<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">BRAIN</span><span class="op">)</span></span>
<span><span class="va">atoms</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BRAIN/man/getAtomsFromSeq.html" class="external-link">getAtomsFromSeq</a></span><span class="op">(</span><span class="st">"SIVPSGASTGVHEALEMR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">atoms</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   C   H   N   O   S </span></span>
<span><span class="co">##  77 129  23  27   1</span></span></code></pre>
<div class="sourceCode" id="cb105"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/sneumann/Rdisop" class="external-link">Rdisop</a></span><span class="op">)</span></span>
<span><span class="va">pepmol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Rdisop/man/getMolecule.html" class="external-link">getMolecule</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">atoms</span><span class="op">)</span>,</span>
<span>                             <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">atoms</span><span class="op">)</span>,</span>
<span>                             collapse <span class="op">=</span> <span class="st">""</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pepmol</span></span></code></pre></div>
<pre><code><span><span class="co">## $formula</span></span>
<span><span class="co">## [1] "C77H129N23O27S"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $score</span></span>
<span><span class="co">## [1] 1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $exactmass</span></span>
<span><span class="co">## [1] 1839.915</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $charge</span></span>
<span><span class="co">## [1] 0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $parity</span></span>
<span><span class="co">## [1] "e"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $valid</span></span>
<span><span class="co">## [1] "Valid"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $DBE</span></span>
<span><span class="co">## [1] 25</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $isotopes</span></span>
<span><span class="co">## $isotopes[[1]]</span></span>
<span><span class="co">##              [,1]         [,2]         [,3]         [,4]         [,5]</span></span>
<span><span class="co">## [1,] 1839.9148973 1840.9177412 1841.9196777 1.842921e+03 1.843923e+03</span></span>
<span><span class="co">## [2,]    0.3427348    0.3353456    0.1960976 8.474135e-02 2.952833e-02</span></span>
<span><span class="co">##              [,6]         [,7]         [,8]         [,9]        [,10]</span></span>
<span><span class="co">## [1,] 1.844925e+03 1.845927e+03 1.846928e+03 1.847930e+03 1.848932e+03</span></span>
<span><span class="co">## [2,] 8.691735e-03 2.226358e-03 5.066488e-04 1.040196e-04 1.949686e-05</span></span></code></pre>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">##</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://OrgMassSpec.github.io/" class="external-link">OrgMassSpecR</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: grid</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'grid'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:Biostrings':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     pattern</span></span></code></pre>
<div class="sourceCode" id="cb111"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">itraqdata</span><span class="op">)</span></span>
<span></span>
<span><span class="va">simplottest</span> <span class="op">&lt;-</span></span>
<span>  <span class="va">itraqdata</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html" class="external-link">featureNames</a></span><span class="op">(</span><span class="va">itraqdata</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"X"</span>, <span class="fl">46</span><span class="op">:</span><span class="fl">47</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">sim</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/OrgMassSpecR/man/SpectrumSimilarity.html" class="external-link">SpectrumSimilarity</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">simplottest</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="st">"data.frame"</span><span class="op">)</span>,</span>
<span>                          <span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">simplottest</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>, <span class="st">"data.frame"</span><span class="op">)</span>,</span>
<span>                          top.lab <span class="op">=</span> <span class="st">"itraqdata[['X46']]"</span>,</span>
<span>                          bottom.lab <span class="op">=</span> <span class="st">"itraqdata[['X47']]"</span>,</span>
<span>                          b <span class="op">=</span> <span class="fl">25</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/title.html" class="external-link">title</a></span><span class="op">(</span>main <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Spectrum similarity"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">sim</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="RforProteomics_files/figure-html/isotopes-1.png" width="700"></p>
<div class="sourceCode" id="cb112"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/OrgMassSpecR/man/MonoisotopicMass.html" class="external-link">MonoisotopicMass</a></span><span class="op">(</span>formula <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>C <span class="op">=</span> <span class="fl">2</span>, O <span class="op">=</span> <span class="fl">1</span>, H<span class="op">=</span><span class="fl">6</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 46.04186</span></span></code></pre>
<div class="sourceCode" id="cb114"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">molecule</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Rdisop/man/getMolecule.html" class="external-link">getMolecule</a></span><span class="op">(</span><span class="st">"C2H5OH"</span><span class="op">)</span></span>
<span><span class="va">molecule</span><span class="op">$</span><span class="va">exactmass</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 46.04186</span></span></code></pre>
<div class="sourceCode" id="cb116"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## x11()</span></span>
<span><span class="co">## plot(t(.pepmol$isotopes[[1]]), type = "h")</span></span>
<span></span>
<span><span class="co">## x &lt;- IsotopicDistribution(formula = list(C = 2, O = 1, H=6))</span></span>
<span><span class="co">## t(molecule$isotopes[[1]])</span></span>
<span><span class="co">## par(mfrow = c(2,1))</span></span>
<span><span class="co">## plot(t(molecule$isotopes[[1]]), type = "h")</span></span>
<span><span class="co">## plot(x[, c(1,3)], type = "h")</span></span>
<span></span>
<span><span class="co">## data(myo500)</span></span>
<span><span class="co">## masses &lt;- c(147.053, 148.056)</span></span>
<span><span class="co">## intensities &lt;- c(93, 5.8)</span></span>
<span><span class="co">## molecules &lt;- decomposeIsotopes(masses, intensities)</span></span>
<span></span>
<span><span class="co">## experimental eno peptides</span></span>
<span><span class="va">exppep</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"ENO"</span>, <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span>, <span class="op">]</span><span class="op">)</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span><span class="op">)</span> <span class="co">## 13</span></span>
<span><span class="va">minlength</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html" class="external-link">nchar</a></span><span class="op">(</span><span class="va">exppep</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="st">"P00924.fasta"</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="va">eno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"http://www.uniprot.org/uniprot/P00924.fasta"</span>,</span>
<span>                         destfile <span class="op">=</span> <span class="st">"P00924.fasta"</span><span class="op">)</span></span>
<span><span class="va">eno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/readLines.html" class="external-link">readLines</a></span><span class="op">(</span><span class="st">"P00924.fasta"</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span>, collapse <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span>
<span><span class="va">enopep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/OrgMassSpecR/man/Digest.html" class="external-link">Digest</a></span><span class="op">(</span><span class="va">eno</span>, missed <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">enopep</span><span class="op">)</span> <span class="co">## 103</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 103</span></span></code></pre>
<div class="sourceCode" id="cb118"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html" class="external-link">nchar</a></span><span class="op">(</span><span class="va">enopep</span><span class="op">$</span><span class="va">peptide</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="va">minlength</span><span class="op">)</span> <span class="co">## 68</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 68</span></span></code></pre>
<div class="sourceCode" id="cb120"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pepcnt</span> <span class="op">&lt;-</span> <span class="va">enopep</span><span class="op">[</span><span class="va">enopep</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">exppep</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">pepcnt</span><span class="op">)</span> <span class="co">## 13</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 13</span></span></code></pre>
<p>The following code chunks demonstrate how to use the <em><a href="https://bioconductor.org/packages/3.17/cleaver" class="external-link">cleaver</a></em>
package for in-silico cleavage of polypeptides, e.g. cleaving of
<em>Gastric juice peptide 1 (P01358)</em> using <em>Trypsin</em>:</p>
<div class="sourceCode" id="cb122"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/sgibb/cleaver/" class="external-link">cleaver</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/cleaver/man/cleave-methods.html" class="external-link">cleave</a></span><span class="op">(</span><span class="st">"LAAGKVEDSD"</span>, enzym <span class="op">=</span> <span class="st">"trypsin"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $LAAGKVEDSD</span></span>
<span><span class="co">## [1] "LAAGK" "VEDSD"</span></span></code></pre>
<p>Sometimes cleavage is not perfect and the enzym miss some cleavage
positions:</p>
<div class="sourceCode" id="cb124"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## miss one cleavage position</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/cleaver/man/cleave-methods.html" class="external-link">cleave</a></span><span class="op">(</span><span class="st">"LAAGKVEDSD"</span>, enzym <span class="op">=</span> <span class="st">"trypsin"</span>, missedCleavages <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $LAAGKVEDSD</span></span>
<span><span class="co">## [1] "LAAGKVEDSD"</span></span></code></pre>
<div class="sourceCode" id="cb126"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## miss zero or one cleavage positions</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/cleaver/man/cleave-methods.html" class="external-link">cleave</a></span><span class="op">(</span><span class="st">"LAAGKVEDSD"</span>, enzym <span class="op">=</span> <span class="st">"trypsin"</span>, missedCleavages <span class="op">=</span> <span class="fl">0</span><span class="op">:</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $LAAGKVEDSD</span></span>
<span><span class="co">## [1] "LAAGK"      "VEDSD"      "LAAGKVEDSD"</span></span></code></pre>
<p>Example code to generate an Texshade image to be included directly in
a Latex document or R vignette is presented below. The R code generates
a Texshade environment and the annotated sequence display code that is
written to a TeX file that can itself be included into a LaTeX or Sweave
document.</p>
<pre><code><span><span class="va">seq1file</span> <span class="op">&lt;-</span> <span class="st">"seq1.tex"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"\\begin{texshade}{Figures/P00924.fasta}</span></span>
<span><span class="st">     \\setsize{numbering}{footnotesize}</span></span>
<span><span class="st">     \\setsize{residues}{footnotesize}</span></span>
<span><span class="st">     \\residuesperline*{70}</span></span>
<span><span class="st">     \\shadingmode{functional}</span></span>
<span><span class="st">     \\hideconsensus</span></span>
<span><span class="st">     \\vsepspace{1mm}</span></span>
<span><span class="st">     \\hidenames</span></span>
<span><span class="st">     \\noblockskip\n"</span>, file <span class="op">=</span> <span class="va">seq1file</span><span class="op">)</span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">pepcnt</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">i</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="op">(</span><span class="va">i</span> <span class="op"><a href="https://rdrr.io/r/base/Arithmetic.html" class="external-link">%%</a></span> <span class="fl">2</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span>, <span class="st">"Blue"</span>, <span class="st">"RoyalBlue"</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"\\shaderegion{1}{"</span>, <span class="va">pepcnt</span><span class="op">$</span><span class="va">start</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, <span class="st">".."</span>, <span class="va">pepcnt</span><span class="op">$</span><span class="va">stop</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, <span class="st">"}{White}{"</span>, <span class="va">col</span>, <span class="st">"}\n"</span>,</span>
<span>      file <span class="op">=</span> <span class="va">seq1file</span>, append <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"\\end{texshade}</span></span>
<span><span class="st">    \\caption{Visualising observed peptides for the Yeast enolase protein. Peptides are shaded in blue and black.</span></span>
<span><span class="st">              The last peptide is a mis-cleavage and overlaps with \`IEEELGDNAVFAGENFHHGDK`.}</span></span>
<span><span class="st">    \ {#fig:seq}</span></span>
<span><span class="st">  \\end{center}</span></span>
<span><span class="st">\\end{figure}\n\n"</span>,</span>
<span>    file <span class="op">=</span> <span class="va">seq1file</span>, append <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre>
<div class="section level4">
<h4 class="unnumbered" id="n15-incorporation">N15 incorporation<a class="anchor" aria-label="anchor" href="#n15-incorporation"></a>
</h4>
<div class="sourceCode" id="cb129"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## 15N incorporation rates from 0, 0.1, ..., 0.9, 0.95, 1</span></span>
<span><span class="va">incrate</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.9</span>, <span class="fl">0.1</span><span class="op">)</span>, <span class="fl">0.95</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">inc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">incrate</span>, <span class="kw">function</span><span class="op">(</span><span class="va">inc</span><span class="op">)</span></span>
<span>              <span class="fu"><a href="https://rdrr.io/pkg/OrgMassSpecR/man/IsotopicDistributionN.html" class="external-link">IsotopicDistributionN</a></span><span class="op">(</span><span class="st">"YEVQGEVFTKPQLWP"</span>, <span class="va">inc</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">inc</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">inc</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">3</span><span class="op">)</span><span class="op">]</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1823</span>, <span class="fl">1848</span><span class="op">)</span>, type <span class="op">=</span> <span class="st">"h"</span>,</span>
<span>       main <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"15N incorporation at "</span>, <span class="va">incrate</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">*</span><span class="fl">100</span>, <span class="st">"%"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="RforProteomics_files/figure-html/n15-1.png" alt="Isotopic envelope for the `YEVQGEVFTKPQLWP` peptide at different N15 incorporation rates " width="700"><p class="caption">
Isotopic envelope for the <code>YEVQGEVFTKPQLWP</code> peptide at
different N15 incorporation rates
</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="the-isobar-package">The <em><a href="https://bioconductor.org/packages/3.17/isobar" class="external-link">isobar</a></em>
package<a class="anchor" aria-label="anchor" href="#the-isobar-package"></a>
</h3>
<p>The <em><a href="https://bioconductor.org/packages/3.17/isobar" class="external-link">isobar</a></em>
package <span class="citation">(Breitwieser et al. 2011)</span> provides
methods for the statistical analysis of isobarically tagged MS2
experiments. Please refer to the package vignette for more details.</p>
</div>
<div class="section level3">
<h3 id="the-dep-package">The <em><a href="https://bioconductor.org/packages/3.17/DEP" class="external-link">DEP</a></em>
package<a class="anchor" aria-label="anchor" href="#the-dep-package"></a>
</h3>
<p>The <em><a href="https://bioconductor.org/packages/3.17/DEP" class="external-link">DEP</a></em> package
supports analysis of label-free and TMT pipelines using, as described in
its vignette. These can be used with <code>MSnSet</code> objects by
converting them to/from <code>SummarizedExperiment</code> objects:</p>
<div class="sourceCode" id="cb130"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">msnset</span><span class="op">)</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">msnset</span>, <span class="st">"SummarizedExperiment"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required namespace: SummarizedExperiment</span></span></code></pre>
<div class="sourceCode" id="cb132"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SummarizedExperiment </span></span>
<span><span class="co">## dim: 55 4 </span></span>
<span><span class="co">## metadata(3): MSnbaseFiles MSnbaseProcessing MSnbaseVersion</span></span>
<span><span class="co">## assays(1): ''</span></span>
<span><span class="co">## rownames(55): X1 X10 ... X8 X9</span></span>
<span><span class="co">## rowData names(15): spectrum ProteinAccession ... acquisition.number</span></span>
<span><span class="co">##   collision.energy</span></span>
<span><span class="co">## colnames(4): iTRAQ4.114 iTRAQ4.115 iTRAQ4.116 iTRAQ4.117</span></span>
<span><span class="co">## colData names(2): mz reporters</span></span></code></pre>
<div class="sourceCode" id="cb134"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ms</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">"MSnSet"</span><span class="op">)</span></span>
<span><span class="va">ms</span></span></code></pre></div>
<pre><code><span><span class="co">## MSnSet (storageMode: lockedEnvironment)</span></span>
<span><span class="co">## assayData: 55 features, 4 samples </span></span>
<span><span class="co">##   element names: exprs </span></span>
<span><span class="co">## protocolData: none</span></span>
<span><span class="co">## phenoData</span></span>
<span><span class="co">##   sampleNames: iTRAQ4.114 iTRAQ4.115 iTRAQ4.116 iTRAQ4.117</span></span>
<span><span class="co">##   varLabels: mz reporters</span></span>
<span><span class="co">##   varMetadata: labelDescription</span></span>
<span><span class="co">## featureData</span></span>
<span><span class="co">##   featureNames: X1 X10 ... X9 (55 total)</span></span>
<span><span class="co">##   fvarLabels: spectrum ProteinAccession ... collision.energy (15 total)</span></span>
<span><span class="co">##   fvarMetadata: labelDescription</span></span>
<span><span class="co">## experimentData: use 'experimentData(object)'</span></span>
<span><span class="co">## Annotation:  </span></span>
<span><span class="co">## - - - Processing information - - -</span></span>
<span><span class="co">##  MSnbase version: 2.25.1</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="the-synapter-package">The <em><a href="https://bioconductor.org/packages/3.17/synapter" class="external-link">synapter</a></em>
package<a class="anchor" aria-label="anchor" href="#the-synapter-package"></a>
</h3>
<p>The <em><a href="https://bioconductor.org/packages/3.17/synapter" class="external-link">synapter</a></em>
<span class="citation">(Bond et al. 2013)</span> package comes with a
detailed vignette that describes how to prepare the MSE data and then
process it in R. Several interfaces are available provided the user with
maximum control, easy batch processing capabilities or a graphical user
interface. The conversion into <code>MSnSet</code> instances and filter
and combination thereof as well as statistical analysis are also
described.</p>
<div class="sourceCode" id="cb136"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## open the synapter vignette</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"synapter"</span><span class="op">)</span></span>
<span><span class="fu">synapterGuide</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="ms2-spectra-identification">MS2 spectra identification<a class="anchor" aria-label="anchor" href="#ms2-spectra-identification"></a>
</h2>
<div class="section level3">
<h3 id="post-search-filtering-of-msms-ids-using-msnid">Post-search Filtering of MS/MS IDs Using <em><a href="https://bioconductor.org/packages/3.17/MSnID" class="external-link">MSnID</a></em><a class="anchor" aria-label="anchor" href="#post-search-filtering-of-msms-ids-using-msnid"></a>
</h3>
<p>The main purpose of <em><a href="https://bioconductor.org/packages/3.17/MSnID" class="external-link">MSnID</a></em>
package is to make sure that the peptide and protein identifications
resulting from MS/MS searches are sufficiently confident for a given
application.} MS/MS peptide and protein identification is a process that
prone to uncertanities. A typical and currently most reliable way to
quantify uncertainty in the list of identify spectra, peptides or
proteins relies on so-called decoy database. For bottom-up
(i.e. involving protein digestion) approaches a common way to construct
a decoy database is simple inversion of protein amino-acid sequences. If
the spectrum matches to normal protein sequence it can be true or false
match. Matches to decoy part of the database are false only (excluding
the palindromes). Therefore the false discovery rate (FDR) of
identifications can be estimated as ratio of hits to decoy over normal
parts of the protein sequence database. There are multiple levels of
identification that FDR can be estimated for. First, is at the level of
peptide/protein- to-spectrum matches. Second is at the level of unique
peptide sequences. Note, true peptides tend to be identified by more
then one spectrum. False peptide tend to be sporadic. Therefore, after
collapsing the redundant peptide identifications from multiple spectra
to the level of unique peptide sequence, the FDR typically increases.
The extend of FDR increase depends on the type and complexity of the
sample. The same trend is true for estimating the identification FDR at
the protein level. True proteins tend to be identified with multiple
peptides, while false protein identifications are commonly covered only
by one peptide. Therefore FDR estimate tend to be even higher for
protein level compare to peptide level. The estimation of the FDR is
also affected by the number of LC-MS (runs) datasets in the experiment.
Again, true identifications tend to be more consistent from run to run,
while false are sporadic. After collapsing the redundancy across the
runs, the number of true identification reduces much stronger compare to
false identifications. Therefore, the peptide and protein FDR estimates
need to be re-evaluated. The main objective of the MSnID package is to
provide convenience tools for handling tasks on estimation of FDR,
defining and optimizing the filtering criteria and ensuring confidence
in MS/MS identification data. The user can specify the criteria for
filtering the data (e.g. goodness or p-value of matching of experimental
and theoretical fragmentation mass spectrum, deviation of theoretical
from experimentally measured mass, presence of missed cleavages in the
peptide sequence, etc), evaluate the performance of the filter judging
by FDRs at spectrum, peptide and protein levels, and finally optimize
the filter to achieve the maximum number of identifications while not
exceeding maximally allowed FDR upper threshold.</p>
<div class="section level4">
<h4 class="unnumbered" id="starting-project-and-importing-data">Starting Project and Importing Data<a class="anchor" aria-label="anchor" href="#starting-project-and-importing-data"></a>
</h4>
<p>To start a project one have to specify a directory. Currently the
only use of the directory is for storing cached results.</p>
<div class="sourceCode" id="cb137"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"MSnID"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'MSnID'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:ProtGenerics':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     peptides</span></span></code></pre>
<div class="sourceCode" id="cb140"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/MSnID-class.html" class="external-link">MSnID</a></span><span class="op">(</span><span class="st">"."</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Note, the anticipated/suggested columns in the</span></span>
<span><span class="co">## peptide-to-spectrum matching results are:</span></span>
<span><span class="co">## -----------------------------------------------</span></span>
<span><span class="co">## accession</span></span>
<span><span class="co">## calculatedMassToCharge</span></span>
<span><span class="co">## chargeState</span></span>
<span><span class="co">## experimentalMassToCharge</span></span>
<span><span class="co">## isDecoy</span></span>
<span><span class="co">## peptide</span></span>
<span><span class="co">## spectrumFile</span></span>
<span><span class="co">## spectrumID</span></span></code></pre>
<p>Data can imported as <code>data.frame</code> or read from mzIdentML
file.</p>
<div class="sourceCode" id="cb142"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">PSMresults</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"human_brain.txt"</span>,</span>
<span>                                     package<span class="op">=</span><span class="st">"MSnID"</span><span class="op">)</span>,</span>
<span>                         stringsAsFactors<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">psms</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">PSMresults</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## MSnID object</span></span>
<span><span class="co">## Working directory: "."</span></span>
<span><span class="co">## #Spectrum Files:  1 </span></span>
<span><span class="co">## #PSMs: 767 at 49 % FDR</span></span>
<span><span class="co">## #peptides: 687 at 57 % FDR</span></span>
<span><span class="co">## #accessions: 665 at 65 % FDR</span></span></code></pre>
<div class="sourceCode" id="cb144"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mzids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"c_elegans.mzid.gz"</span>, package<span class="op">=</span><span class="st">"MSnID"</span><span class="op">)</span></span>
<span><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/read_mzIDs.html" class="external-link">read_mzIDs</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="va">mzids</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Reading from mzIdentMLs ...</span></span></code></pre>
<pre><code><span><span class="co">## reading c_elegans.mzid.gz...</span></span></code></pre>
<pre><code><span><span class="co">## Warning in type.convert.default(...): 'as.is' should be specified by the caller;</span></span>
<span><span class="co">## using TRUE</span></span>
<span></span>
<span><span class="co">## Warning in type.convert.default(...): 'as.is' should be specified by the caller;</span></span>
<span><span class="co">## using TRUE</span></span>
<span></span>
<span><span class="co">## Warning in type.convert.default(...): 'as.is' should be specified by the caller;</span></span>
<span><span class="co">## using TRUE</span></span>
<span></span>
<span><span class="co">## Warning in type.convert.default(...): 'as.is' should be specified by the caller;</span></span>
<span><span class="co">## using TRUE</span></span>
<span></span>
<span><span class="co">## Warning in type.convert.default(...): 'as.is' should be specified by the caller;</span></span>
<span><span class="co">## using TRUE</span></span>
<span></span>
<span><span class="co">## Warning in type.convert.default(...): 'as.is' should be specified by the caller;</span></span>
<span><span class="co">## using TRUE</span></span>
<span></span>
<span><span class="co">## Warning in type.convert.default(...): 'as.is' should be specified by the caller;</span></span>
<span><span class="co">## using TRUE</span></span>
<span></span>
<span><span class="co">## Warning in type.convert.default(...): 'as.is' should be specified by the caller;</span></span>
<span><span class="co">## using TRUE</span></span>
<span></span>
<span><span class="co">## Warning in type.convert.default(...): 'as.is' should be specified by the caller;</span></span>
<span><span class="co">## using TRUE</span></span>
<span></span>
<span><span class="co">## Warning in type.convert.default(...): 'as.is' should be specified by the caller;</span></span>
<span><span class="co">## using TRUE</span></span>
<span></span>
<span><span class="co">## Warning in type.convert.default(...): 'as.is' should be specified by the caller;</span></span>
<span><span class="co">## using TRUE</span></span>
<span></span>
<span><span class="co">## Warning in type.convert.default(...): 'as.is' should be specified by the caller;</span></span>
<span><span class="co">## using TRUE</span></span></code></pre>
<pre><code><span><span class="co">##  DONE!</span></span></code></pre>
<div class="sourceCode" id="cb149"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## MSnID object</span></span>
<span><span class="co">## Working directory: "."</span></span>
<span><span class="co">## #Spectrum Files:  1 </span></span>
<span><span class="co">## #PSMs: 12263 at 36 % FDR</span></span>
<span><span class="co">## #peptides: 9489 at 44 % FDR</span></span>
<span><span class="co">## #accessions: 7414 at 76 % FDR</span></span></code></pre>
</div>
<div class="section level4">
<h4 class="unnumbered" id="analysis-of-peptide-sequences">Analysis of Peptide Sequences<a class="anchor" aria-label="anchor" href="#analysis-of-peptide-sequences"></a>
</h4>
<p>A particular properties of peptide sequences we are interested in
are</p>
<ol style="list-style-type: decimal">
<li>irregular cleavages at the termini of the peptides and</li>
<li>missing cleavage site within the peptide sequences.</li>
</ol>
<p>A particular properties of peptide sequences we are interested in are
(1) irregular cleavages at the termini of the peptides and (2) missing
cleavage site within the peptide sequences:</p>
<ul>
<li>Counting the number of irregular cleavage termimi (0, 1 or 2) in
peptides sequence creates a new column
<code>numIrregCleavages</code>.</li>
<li>Counting the number of missed cleavages in peptides sequence
correspondingly creates a <code>numMissCleavages</code> column.</li>
</ul>
<p>The default regular expressions for the
<code>validCleavagePattern</code> and <code>missedCleavagePattern</code>
correspond to trypsin specificity.</p>
<div class="sourceCode" id="cb151"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/assess_termini.html" class="external-link">assess_termini</a></span><span class="op">(</span><span class="va">msnid</span>, validCleavagePattern<span class="op">=</span><span class="st">"[KR]\\.[^P]"</span><span class="op">)</span></span>
<span><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/assess_missed_cleavages.html" class="external-link">assess_missed_cleavages</a></span><span class="op">(</span><span class="va">msnid</span>, missedCleavagePattern<span class="op">=</span><span class="st">"[KR](?=[^P$])"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/proportions.html" class="external-link">prop.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">$</span><span class="va">numIrregCleavages</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##           0           1           2 </span></span>
<span><span class="co">## 0.801574390 0.189294149 0.009131462</span></span></code></pre>
<p>Now the object has two more columns, <code>numIrregCleavages</code>
and <code>numMissCleavages</code>, evidently corresponding to the number
of termini with irregular cleavages and number of missed cleavages
within the peptide sequence. The figure below shows that peptides with 2
or more missed cleavages are likely to be false identifications.</p>
<div class="sourceCode" id="cb153"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pepCleav</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">psms</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"numMissCleavages"</span>, <span class="st">"isDecoy"</span>, <span class="st">"peptide"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">pepCleav</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">pepCleav</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"numMissCleavages"</span>, <span class="st">"isDecoy"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">pepCleav</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">numMissCleavages</span>, y<span class="op">=</span><span class="va">Freq</span>, fill<span class="op">=</span><span class="va">isDecoy</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat<span class="op">=</span><span class="st">'identity'</span>, position<span class="op">=</span><span class="st">'dodge'</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="st">"Number of Missed Cleavages"</span><span class="op">)</span></span></code></pre></div>
<p><img src="RforProteomics_files/figure-html/MSnIDmissedCleavagesPlot-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 class="unnumbered" id="defining-the-filter">Defining the Filter<a class="anchor" aria-label="anchor" href="#defining-the-filter"></a>
</h4>
<p>The criteria that will be used for filtering the MS/MS data has to be
present in the <code>MSnID</code> object. We will use -log10 transformed
MS-GF+ Spectrum E-value, reflecting the goodness of match experimental
and theoretical fragmentation patterns as one the filtering criteria.
Let’s store it under the “msmsScore” name. The score density
distribution shows that it is a good discriminant between non-decoy
(red) and decoy hits (green).</p>
<p>For alternative MS/MS search engines refer to the engine-specific
manual for the names of parameters reflecting the quality of MS/MS
spectra matching. Examples of such parameters are <code>E-Value</code>
for X!Tandem and <code>XCorr</code> and <code>$\Delta$Cn2</code> for
SEQUEST.</p>
<p>As a second criterion we will be using the absolute mass measurement
error (in ppm units) of the parent ion. The mass measurement errors tend
to be small for non-decoy (enriched with real identificaiton) hits (red
line) and is effectively uniformly distributed for decoy hits.</p>
<div class="sourceCode" id="cb154"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msnid</span><span class="op">$</span><span class="va">msmsScore</span> <span class="op">&lt;-</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">$</span><span class="va">`MS-GF:SpecEValue`</span><span class="op">)</span></span>
<span><span class="va">msnid</span><span class="op">$</span><span class="va">absParentMassErrorPPM</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/mass_measurement_error.html" class="external-link">mass_measurement_error</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>MS/MS fiters are handled by a special <em>MSnIDFilter</em> class
objects. Individual filtering criteria can be set by name (that is
present in <code>names(msnid)</code>), comparison operator (&gt;, &lt;,
= , …) defining if we should retain hits with higher or lower given the
threshold and finally the threshold value itself. The filter below set
in such a way that retains only those matches that has less then 5 ppm
of parent ion mass measurement error and more the <span class="math inline">\(10^7\)</span> MS-GF:SpecEValue.</p>
<div class="sourceCode" id="cb155"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filtObj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/MSnIDFilter-class.html" class="external-link">MSnIDFilter</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></span>
<span><span class="va">filtObj</span><span class="op">$</span><span class="va">absParentMassErrorPPM</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>comparison<span class="op">=</span><span class="st">"&lt;"</span>, threshold<span class="op">=</span><span class="fl">5.0</span><span class="op">)</span></span>
<span><span class="va">filtObj</span><span class="op">$</span><span class="va">msmsScore</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>comparison<span class="op">=</span><span class="st">"&gt;"</span>, threshold<span class="op">=</span><span class="fl">8.0</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">filtObj</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## MSnIDFilter object</span></span>
<span><span class="co">## (absParentMassErrorPPM &lt; 5) &amp; (msmsScore &gt; 8)</span></span></code></pre>
<p>The stringency of the filter can be evaluated at different
levels.</p>
<div class="sourceCode" id="cb157"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/evaluate_filter.html" class="external-link">evaluate_filter</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="va">filtObj</span>, level<span class="op">=</span><span class="st">"PSM"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##            fdr    n</span></span>
<span><span class="co">## PSM 0.00272745 5147</span></span></code></pre>
<div class="sourceCode" id="cb159"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/evaluate_filter.html" class="external-link">evaluate_filter</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="va">filtObj</span>, level<span class="op">=</span><span class="st">"peptide"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                fdr    n</span></span>
<span><span class="co">## peptide 0.00424371 3313</span></span></code></pre>
<div class="sourceCode" id="cb161"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/evaluate_filter.html" class="external-link">evaluate_filter</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="va">filtObj</span>, level<span class="op">=</span><span class="st">"accession"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                  fdr    n</span></span>
<span><span class="co">## accession 0.01770658 1207</span></span></code></pre>
</div>
<div class="section level4">
<h4 class="unnumbered" id="optimizing-the-filter">Optimizing the Filter<a class="anchor" aria-label="anchor" href="#optimizing-the-filter"></a>
</h4>
<p>The threshold values in the example above are not necessarily optimal
and set just be in the range of probable values. Filters can be
optimized to ensure maximum number of identifications
(peptide-to-spectrum matches, unique peptide sequences or proteins)
within a given FDR upper limit.</p>
<p>First, the filter can be optimized simply by stepping through
individual parameters and their combinations. The idea has been
described in <span class="citation">(Piehowski et al. 2013)</span>. The
resulting <code>MSnIDFilter</code> object can be used for final data
filtering or can be used as a good starting parameters for follow-up
refining optimizations with more advanced algorithms.</p>
<div class="sourceCode" id="cb163"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filtObj.grid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/optimize_filter.html" class="external-link">optimize_filter</a></span><span class="op">(</span><span class="va">filtObj</span>, <span class="va">msnid</span>, fdr.max<span class="op">=</span><span class="fl">0.01</span>,</span>
<span>                                method<span class="op">=</span><span class="st">"Grid"</span>, level<span class="op">=</span><span class="st">"peptide"</span>,</span>
<span>                                n.iter<span class="op">=</span><span class="fl">500</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">filtObj.grid</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## MSnIDFilter object</span></span>
<span><span class="co">## (absParentMassErrorPPM &lt; 10) &amp; (msmsScore &gt; 7.8)</span></span></code></pre>
<p>The resulting <code>filtObj.grid</code> can be further fine tuned
with such optimization routines as simulated annealing or Nelder-Mead
optimization.</p>
<div class="sourceCode" id="cb165"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filtObj.nm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/optimize_filter.html" class="external-link">optimize_filter</a></span><span class="op">(</span><span class="va">filtObj.grid</span>, <span class="va">msnid</span>, fdr.max<span class="op">=</span><span class="fl">0.01</span>,</span>
<span>                                method<span class="op">=</span><span class="st">"Nelder-Mead"</span>, level<span class="op">=</span><span class="st">"peptide"</span>,</span>
<span>                                n.iter<span class="op">=</span><span class="fl">500</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">filtObj.nm</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## MSnIDFilter object</span></span>
<span><span class="co">## (absParentMassErrorPPM &lt; 10) &amp; (msmsScore &gt; 7.8)</span></span></code></pre>
<p>Evaluate non-optimized and optimized filters.</p>
<div class="sourceCode" id="cb167"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/evaluate_filter.html" class="external-link">evaluate_filter</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="va">filtObj</span>, level<span class="op">=</span><span class="st">"peptide"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                fdr    n</span></span>
<span><span class="co">## peptide 0.00424371 3313</span></span></code></pre>
<div class="sourceCode" id="cb169"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/evaluate_filter.html" class="external-link">evaluate_filter</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="va">filtObj.grid</span>, level<span class="op">=</span><span class="st">"peptide"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                 fdr    n</span></span>
<span><span class="co">## peptide 0.009220702 3393</span></span></code></pre>
<div class="sourceCode" id="cb171"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/evaluate_filter.html" class="external-link">evaluate_filter</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="va">filtObj.nm</span>, level<span class="op">=</span><span class="st">"peptide"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                 fdr    n</span></span>
<span><span class="co">## peptide 0.009777778 3408</span></span></code></pre>
<p>Finally applying filter to remove predominantly false
identifications.</p>
<div class="sourceCode" id="cb173"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/apply_filter.html" class="external-link">apply_filter</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="va">filtObj.nm</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## MSnID object</span></span>
<span><span class="co">## Working directory: "."</span></span>
<span><span class="co">## #Spectrum Files:  1 </span></span>
<span><span class="co">## #PSMs: 5288 at 0.63 % FDR</span></span>
<span><span class="co">## #peptides: 3408 at 0.98 % FDR</span></span>
<span><span class="co">## #accessions: 1253 at 3.8 % FDR</span></span></code></pre>
<p>Removing hits to decoy and contaminant sequences using the same
<code>apply_filter</code> method.</p>
<div class="sourceCode" id="cb175"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/apply_filter.html" class="external-link">apply_filter</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="st">"isDecoy == FALSE"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## MSnID object</span></span>
<span><span class="co">## Working directory: "."</span></span>
<span><span class="co">## #Spectrum Files:  1 </span></span>
<span><span class="co">## #PSMs: 5255 at 0 % FDR</span></span>
<span><span class="co">## #peptides: 3375 at 0 % FDR</span></span>
<span><span class="co">## #accessions: 1207 at 0 % FDR</span></span></code></pre>
<div class="sourceCode" id="cb177"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msnid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/apply_filter.html" class="external-link">apply_filter</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="st">"!grepl('Contaminant',accession)"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/methods/show.html" class="external-link">show</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## MSnID object</span></span>
<span><span class="co">## Working directory: "."</span></span>
<span><span class="co">## #Spectrum Files:  1 </span></span>
<span><span class="co">## #PSMs: 5246 at 0 % FDR</span></span>
<span><span class="co">## #peptides: 3368 at 0 % FDR</span></span>
<span><span class="co">## #accessions: 1205 at 0 % FDR</span></span></code></pre>
</div>
<div class="section level4">
<h4 class="unnumbered" id="interface-with-other-bioconductor-packages">Interface with Other Bioconductor Packages<a class="anchor" aria-label="anchor" href="#interface-with-other-bioconductor-packages"></a>
</h4>
<p>One can extract the entire PSMs tables as <code>data.frame</code> or
<code>data.table</code></p>
<div class="sourceCode" id="cb179"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">psm.df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">psms</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></span>
<span><span class="va">psm.dt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="st">"data.table"</span><span class="op">)</span></span></code></pre></div>
<p>If only interested in the non-redundant list of confidently
identified peptides or proteins</p>
<div class="sourceCode" id="cb180"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">peps</span> <span class="op">&lt;-</span> <span class="fu">MSnID</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/MSnID/man/peptides.html" class="external-link">peptides</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">peps</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "K.AISQIQEYVDYYGGSGVQHIALNTSDIITAIEALR.A"      </span></span>
<span><span class="co">## [2] "K.SAGSGYLVGDSLTFVDLLVAQHTADLLAANAALLDEFPQFK.A"</span></span>
<span><span class="co">## [3] "K.NSIFTNVAETANGEYFWEGLEDEIADKNVDITTWLGEK.W"   </span></span>
<span><span class="co">## [4] "R.VFCLLGDGESAEGSVWEAAAFASIYKLDNLVAIVDVNR.L"   </span></span>
<span><span class="co">## [5] "R.TTDSDGNNTGLDLYTVDQVEHSNYVEQNFLDFIFVFR.K"    </span></span>
<span><span class="co">## [6] "R.KFDADGSGKLEFDEFCALVYTVANTVDKETLEKELR.E"</span></span></code></pre>
<div class="sourceCode" id="cb182"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">prots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">accessions</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">prots</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "CE02347" "CE07055" "CE12728" "CE36358" "CE36359" "CE36360"</span></span></code></pre>
<div class="sourceCode" id="cb184"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">prots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">proteins</a></span><span class="op">(</span><span class="va">msnid</span><span class="op">)</span> <span class="co"># may be more intuitive then accessions</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">prots</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "CE02347" "CE07055" "CE12728" "CE36358" "CE36359" "CE36360"</span></span></code></pre>
<p>The <em><a href="https://bioconductor.org/packages/3.17/MSnID" class="external-link">MSnID</a></em>
package is aimed at providing convenience functionality to handle MS/MS
identifications. Quantification <em>per se</em> is outside of the scope
of the package. The only type of quantitation that can be seamlessly
tied with MS/MS identification analysis is so-called <em>spectral
counting</em> approach. In such an approach a peptide abundance is
considered to be directly proportional to the number of matched MS/MS
spectra. In its turn protein abunance is proportional to the sum of the
number of spectra of the matching peptides. The <em>MSnID</em> object
can be converted to an <em>MSnSet</em> object defined in <em><a href="https://bioconductor.org/packages/3.17/MSnbase" class="external-link">MSnbase</a></em>
that extends generic Bioconductor <em>eSet</em> class to quantitative
proteomics data. The spectral count data can be analyzed with <em><a href="https://bioconductor.org/packages/3.17/msmsEDA" class="external-link">msmsEDA</a></em>,
<em><a href="https://bioconductor.org/packages/3.17/msmsTests" class="external-link">msmsTests</a></em>
or <em><a href="https://bioconductor.org/packages/3.17/DESeq" class="external-link">DESeq</a></em>
packages.</p>
<div class="sourceCode" id="cb186"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msnset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/methods/as.html" class="external-link">as</a></span><span class="op">(</span><span class="va">msnid</span>, <span class="st">"MSnSet"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://lgatto.github.io/MSnbase" class="external-link">"MSnbase"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">msnset</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                                                     peptide</span></span>
<span><span class="co">## -.APPSQDVLKEIFNLYDEELDGK.I                       -.APPSQDVLKEIFNLYDEELDGK.I</span></span>
<span><span class="co">## -.APPSQDVLKEIFNLYDEELDGKIDGTQVGDVAR.A -.APPSQDVLKEIFNLYDEELDGKIDGTQVGDVAR.A</span></span>
<span><span class="co">## -.APPTFADLGK.S                                               -.APPTFADLGK.S</span></span>
<span><span class="co">## -.GFQNLWFSHPR.K                                             -.GFQNLWFSHPR.K</span></span>
<span><span class="co">## -.GIDINHKHDR.V                                               -.GIDINHKHDR.V</span></span>
<span><span class="co">## -.MFSNLFIFL.V                                                 -.MFSNLFIFL.V</span></span>
<span><span class="co">##                                          accession</span></span>
<span><span class="co">## -.APPSQDVLKEIFNLYDEELDGK.I            CE01236,....</span></span>
<span><span class="co">## -.APPSQDVLKEIFNLYDEELDGKIDGTQVGDVAR.A CE01236,....</span></span>
<span><span class="co">## -.APPTFADLGK.S                             CE29443</span></span>
<span><span class="co">## -.GFQNLWFSHPR.K                            CE26849</span></span>
<span><span class="co">## -.GIDINHKHDR.V                             CE16650</span></span>
<span><span class="co">## -.MFSNLFIFL.V                              CE21589</span></span></code></pre>
<div class="sourceCode" id="cb188"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">msnset</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                       c_elegans_A_3_1_21Apr10_Draco_10-03-04_dta.txt</span></span>
<span><span class="co">## -.APPSQDVLKEIFNLYDEELDGK.I                                                         1</span></span>
<span><span class="co">## -.APPSQDVLKEIFNLYDEELDGKIDGTQVGDVAR.A                                              4</span></span>
<span><span class="co">## -.APPTFADLGK.S                                                                     2</span></span>
<span><span class="co">## -.GFQNLWFSHPR.K                                                                    1</span></span>
<span><span class="co">## -.GIDINHKHDR.V                                                                     2</span></span>
<span><span class="co">## -.MFSNLFIFL.V                                                                      1</span></span></code></pre>
<p>Note, the convertion from <code>MSnID</code> to <code>MSnSet</code>
uses peptides as features. The number of redundant peptide observations
represent so-called spectral count that can be used for rough
quantitative analysis. Summing of all of the peptide counts to a
proteins level can be done with <code>combineFeatures</code> function
from <em><a href="https://bioconductor.org/packages/3.17/MSnbase" class="external-link">MSnbase</a></em>
package.</p>
<div class="sourceCode" id="cb190"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">msnset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">combineFeatures</a></span><span class="op">(</span><span class="va">msnset</span>,</span>
<span>                            <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">msnset</span><span class="op">)</span><span class="op">$</span><span class="va">accession</span>,</span>
<span>                            redundancy.handler<span class="op">=</span><span class="st">"unique"</span>,</span>
<span>                            fun<span class="op">=</span><span class="st">"sum"</span>,</span>
<span>                            cv<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Parameter 'fun' is deprecated. Please use 'method' instead</span></span></code></pre>
<div class="sourceCode" id="cb192"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">msnset</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                     peptide accession</span></span>
<span><span class="co">## CE00078                         K.RLPVAPR.G   CE00078</span></span>
<span><span class="co">## CE00103 K.LPNDDIGVQVSYLGEPHTFTPEQVLAALLTK.L   CE00103</span></span>
<span><span class="co">## CE00134                       I.PAEVAEHLK.A   CE00134</span></span>
<span><span class="co">## CE00209            K.ALEGPGPGEDAAHSENNPPR.N   CE00209</span></span>
<span><span class="co">## CE00302                  K.LTYFDIHGLAEPIR.L   CE00302</span></span>
<span><span class="co">## CE00318        K.ALNALCAQLMTELADALEVLDTDK.S   CE00318</span></span></code></pre>
<div class="sourceCode" id="cb194"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">msnset</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##         c_elegans_A_3_1_21Apr10_Draco_10-03-04_dta.txt</span></span>
<span><span class="co">## CE00078                                            1.0</span></span>
<span><span class="co">## CE00103                                            1.0</span></span>
<span><span class="co">## CE00134                                            1.0</span></span>
<span><span class="co">## CE00209                                            2.0</span></span>
<span><span class="co">## CE00302                                            1.0</span></span>
<span><span class="co">## CE00318                                            2.2</span></span></code></pre>
</div>
</div>
</div>
<div class="section level2">
<h2 id="sec:qc">Quality control<a class="anchor" aria-label="anchor" href="#sec:qc"></a>
</h2>
<p>Quality control (QC) is an essential part of any high throughput data
driven approach. Bioconductor has a rich history of QC for various
genomics data and currently two packages support proteomics QC.</p>
<p><em><a href="https://bioconductor.org/packages/3.17/proteoQC" class="external-link">proteoQC</a></em>
provides a dedicated a dedicated pipeline that will produce a dynamic
and extensive html report. It uses the <em><a href="https://bioconductor.org/packages/3.17/rTANDEM" class="external-link">rTANDEM</a></em>
package to automate the generation of identification data and uses
information about the experimental/replication design.</p>
<p>The <em><a href="https://bioconductor.org/packages/3.17/qcmetrics" class="external-link">qcmetrics</a></em>
package is a general framework to define QC metrics and bundle them
together to generate html or pdf reports. It provides some ready made
metrics for MS data and N15 labelled data.</p>
</div>
<div class="section level2">
<h2 id="sec:annot">Annotation<a class="anchor" aria-label="anchor" href="#sec:annot"></a>
</h2>
<p>In this section, we briefly present some Bioconductor annotation
infrastructure.</p>
<p>We start with the <em><a href="https://bioconductor.org/packages/3.17/hpar" class="external-link">hpar</a></em>
package, an interface to the <em>Human Protein Atlas</em> <span class="citation">(Uhlén et al. 2005, Uhlen2010)</span>, to retrieve
subcellular localisation information for the
<code>ENSG00000002746</code> ensemble gene.</p>
<div class="sourceCode" id="cb196"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">id</span> <span class="op">&lt;-</span> <span class="st">"ENSG00000105323"</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"hpar"</span><span class="op">)</span></span>
<span><span class="va">subcell</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hpar/man/hpar-package.html" class="external-link">hpaSubcellularLoc</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## see ?hpar and browseVignettes('hpar') for documentation</span></span></code></pre>
<pre><code><span><span class="co">## loading from cache</span></span></code></pre>
<div class="sourceCode" id="cb199"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">subcell</span>, <span class="va">Gene</span> <span class="op">==</span> <span class="va">id</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                 Gene Gene.name Reliability Main.location Additional.location</span></span>
<span><span class="co">## 2357 ENSG00000105323  HNRNPUL1    Enhanced   Nucleoplasm                    </span></span>
<span><span class="co">##      Extracellular.location    Enhanced Supported Approved Uncertain</span></span>
<span><span class="co">## 2357                        Nucleoplasm                             </span></span>
<span><span class="co">##      Single.cell.variation.intensity Single.cell.variation.spatial</span></span>
<span><span class="co">## 2357                                                              </span></span>
<span><span class="co">##      Cell.cycle.dependency                    GO.id</span></span>
<span><span class="co">## 2357                       Nucleoplasm (GO:0005654)</span></span></code></pre>
<p>Below, we make use of the human annotation package <em><a href="https://bioconductor.org/packages/3.17/org.Hs.eg.db" class="external-link">org.Hs.eg.db</a></em>
and the Gene Ontology annotation package <em><a href="https://bioconductor.org/packages/3.17/GO.db" class="external-link">GO.db</a></em> to
retrieve compatible information with above.</p>
<div class="sourceCode" id="cb201"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"org.Hs.eg.db"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"GO.db"</span><span class="op">)</span></span>
<span><span class="va">ans</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>,</span>
<span>                             keys <span class="op">=</span> <span class="va">id</span>,</span>
<span>                             columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENSEMBL"</span>, <span class="st">"GO"</span>, <span class="st">"ONTOLOGY"</span><span class="op">)</span>,</span>
<span>                             keytype <span class="op">=</span> <span class="st">"ENSEMBL"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'select()' returned 1:many mapping between keys and columns</span></span></code></pre>
<div class="sourceCode" id="cb203"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ans</span> <span class="op">&lt;-</span> <span class="va">ans</span><span class="op">[</span><span class="va">ans</span><span class="op">$</span><span class="va">ONTOLOGY</span> <span class="op">==</span> <span class="st">"CC"</span>, <span class="op">]</span></span>
<span><span class="va">ans</span></span></code></pre></div>
<pre><code><span><span class="co">##            ENSEMBL         GO EVIDENCE ONTOLOGY</span></span>
<span><span class="co">## 4  ENSG00000105323 GO:0005634      IDA       CC</span></span>
<span><span class="co">## 5  ENSG00000105323 GO:0005654      IBA       CC</span></span>
<span><span class="co">## 6  ENSG00000105323 GO:0005654      IDA       CC</span></span>
<span><span class="co">## 7  ENSG00000105323 GO:0005654      TAS       CC</span></span>
<span><span class="co">## 11 ENSG00000105323 GO:1990904      IEA       CC</span></span></code></pre>
<div class="sourceCode" id="cb205"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">GOTERM</span><span class="op">[</span><span class="va">ans</span><span class="op">$</span><span class="va">GO</span><span class="op">]</span><span class="op">)</span>, <span class="va">slot</span>, <span class="st">"Term"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                  GO:0005634                  GO:0005654 </span></span>
<span><span class="co">##                   "nucleus"               "nucleoplasm" </span></span>
<span><span class="co">##                  GO:0005654                  GO:0005654 </span></span>
<span><span class="co">##               "nucleoplasm"               "nucleoplasm" </span></span>
<span><span class="co">##                  GO:1990904 </span></span>
<span><span class="co">## "ribonucleoprotein complex"</span></span></code></pre>
<p>Finally, this information can also be retrieved from on-line
databases using the <em><a href="https://bioconductor.org/packages/3.17/biomaRt" class="external-link">biomaRt</a></em>
package <span class="citation">(Durinck et al. 2005)</span>.</p>
<div class="sourceCode" id="cb207"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"biomaRt"</span><span class="op">)</span></span>
<span><span class="va">ensembl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useMart.html" class="external-link">useMart</a></span><span class="op">(</span><span class="st">"ensembl"</span>, dataset<span class="op">=</span><span class="st">"hsapiens_gene_ensembl"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb208"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">efilter</span> <span class="op">&lt;-</span> <span class="st">"ensembl_gene_id"</span></span>
<span><span class="va">eattr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"go_id"</span>, <span class="st">"name_1006"</span>, <span class="st">"namespace_1003"</span><span class="op">)</span></span>
<span><span class="va">bmres</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/getBM.html" class="external-link">getBM</a></span><span class="op">(</span>attributes<span class="op">=</span><span class="va">eattr</span>, filters <span class="op">=</span> <span class="va">efilter</span>, values <span class="op">=</span> <span class="va">id</span>, mart <span class="op">=</span> <span class="va">ensembl</span><span class="op">)</span></span>
<span><span class="va">bmres</span><span class="op">[</span><span class="va">bmres</span><span class="op">$</span><span class="va">namespace_1003</span> <span class="op">==</span> <span class="st">"cellular_component"</span>, <span class="st">"name_1006"</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "nucleoplasm"               "nucleus"                  </span></span>
<span><span class="co">## [3] "ribonucleoprotein complex"</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="sec:packages">Other packages<a class="anchor" aria-label="anchor" href="#sec:packages"></a>
</h2>
<div class="section level3">
<h3 id="bioconductor-packages">Bioconductor packages<a class="anchor" aria-label="anchor" href="#bioconductor-packages"></a>
</h3>
<p>This section provides a complete list of packages available in the
relevant Bioconductor version 3.17 <a href="http://www.bioconductor.org/packages/devel/BiocViews.html%3E" class="external-link"><em>biocView</em></a>
categories. the tables below represent the packages for the
<code>Proteomics</code> (165 packages), <code>MassSpectrometry</code>
(124 packages) and <code>MassSpectrometryData</code> (25 experiment
packages) categories.</p>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-9042001f30927f1deea1" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-9042001f30927f1deea1">{"x":{"filter":"none","vertical":false,"data":[["artMS","ASEB","AssessORF","bandle","bioassayR","biobroom","BioNetStat","BRAIN","calm","Cardinal","CausalR","CellNOptR","ChemmineOB","ChemmineR","cisPath","cleaver","clippda","CNORdt","CNORfeeder","CNORode","ComPrAn","CONSTANd","cosmosR","customProDB","cydar","CytoGLMM","cytoKernel","cytoMEM","DAPAR","deltaGseg","DEP","DEqMS","DIAlignR","diffcyt","diffuStats","DominoEffect","Doscheda","drawProteins","drugTargetInteractions","EGSEA","eiR","EWCE","fCI","fmcsR","GenProSeq","GeomxTools","GraphPAC","HPAanalyze","hpar","HPiP","idpr","IMMAN","infinityFlow","InterMineR","iPAC","isobar","KinSwingR","limma","LPEadj","MassSpecWavelet","MatrixQCvis","MBQN","mitch","mixOmics","MobilityTransformR","MQmetrics","MsBackendMgf","MsBackendMsp","MsBackendRawFileReader","MsCoreUtils","MsExperiment","msImpute","msmsEDA","msmsTests","MSnbase","MSnID","msqrob2","MsQuality","MSstats","MSstatsConvert","MSstatsLiP","MSstatsLOBD","MSstatsPTM","MSstatsQC","MSstatsQCgui","MSstatsSampleSize","MSstatsShiny","MSstatsTMT","mzID","mzR","NanoStringNCTools","NormalyzerDE","NPARC","OmicsLonDA","oppti","PAA","Path2PPI","pathview","pathwayPCA","PECA","pepXMLTab","pgca","phosphonormalizer","PhosR","plgem","PLPE","PrInCE","proBAMr","PROcess","procoil","proDA","pRoloc","pRolocGUI","Prostar","proteasy","ProteoDisco","ProteoMM","ProtGenerics","PSMatch","Pviz","qcmetrics","QFeatures","qPLEXanalyzer","QuartPAC","rain","RankProd","rawrr","RCASPAR","Rcpi","ReactomeGSA","RolDE","ROTS","rpx","rScudo","Rtpca","SBGNview","scp","SDAMS","SMAD","SpacePAC","sparsenetgls","specL","Spectra","statTarget","SubCellBarCode","SWATH2stats","synapter","TargetDecoy","timeOmics","topdownr","TPP","TPP2D","vsclust","wpm","XINA","biosigner","omicRexposome","phenomis","proBatch","ropls","proFIA","deco","pulsedSilac","censcyt","phemd"],["artMS","ASEB","AssessORF","bandle","bioassayR","biobroom","BioNetStat","BRAIN","calm","Cardinal","CausalR","CellNOptR","ChemmineOB","ChemmineR","cisPath","cleaver","clippda","CNORdt","CNORfeeder","CNORode","ComPrAn","CONSTANd","cosmosR","customProDB","cydar","CytoGLMM","cytoKernel","cytoMEM","DAPAR","deltaGseg","DEP","DEqMS","DIAlignR","diffcyt","diffuStats","DominoEffect","Doscheda","drawProteins","drugTargetInteractions","EGSEA","eiR","EWCE","fCI","fmcsR","GenProSeq","GeomxTools","GraphPAC","HPAanalyze","hpar","HPiP","idpr","IMMAN","infinityFlow","InterMineR","iPAC","isobar","KinSwingR","limma","LPEadj","MassSpecWavelet","MatrixQCvis","MBQN","mitch","mixOmics","MobilityTransformR","MQmetrics","MsBackendMgf","MsBackendMsp","MsBackendRawFileReader","MsCoreUtils","MsExperiment","msImpute","msmsEDA","msmsTests","MSnbase","MSnID","msqrob2","MsQuality","MSstats","MSstatsConvert","MSstatsLiP","MSstatsLOBD","MSstatsPTM","MSstatsQC","MSstatsQCgui","MSstatsSampleSize","MSstatsShiny","MSstatsTMT","mzID","mzR","NanoStringNCTools","NormalyzerDE","NPARC","OmicsLonDA","oppti","PAA","Path2PPI","pathview","pathwayPCA","PECA","pepXMLTab","pgca","phosphonormalizer","PhosR","plgem","PLPE","PrInCE","proBAMr","PROcess","procoil","proDA","pRoloc","pRolocGUI","Prostar","proteasy","ProteoDisco","ProteoMM","ProtGenerics","PSMatch","Pviz","qcmetrics","QFeatures","qPLEXanalyzer","QuartPAC","rain","RankProd","rawrr","RCASPAR","Rcpi","ReactomeGSA","RolDE","ROTS","rpx","rScudo","Rtpca","SBGNview","scp","SDAMS","SMAD","SpacePAC","sparsenetgls","specL","Spectra","statTarget","SubCellBarCode","SWATH2stats","synapter","TargetDecoy","timeOmics","topdownr","TPP","TPP2D","vsclust","wpm","XINA","biosigner","omicRexposome","phenomis","proBatch","ropls","proFIA","deco","pulsedSilac","censcyt","phemd"],["Analytical R tools for Mass Spectrometry","Predict Acetylated Lysine Sites","Assess Gene Predictions Using Proteomics and Evolutionary Conservation","An R package for the Bayesian analysis of differential subcellular localisation experiments","Cross-target analysis of small molecule bioactivity","Turn Bioconductor objects into tidy data frames","Biological Network Analysis","Baffling Recursive Algorithm for Isotope distributioN calculations","Covariate Assisted Large-scale Multiple testing","A mass spectrometry imaging toolbox for statistical analysis","Causal network analysis methods","Training of boolean logic models of signalling networks using prior knowledge networks and perturbation data","R interface to a subset of OpenBabel functionalities","Cheminformatics Toolkit for R","Visualization and management of the protein-protein interaction networks.","Cleavage of Polypeptide Sequences","A package for the clinical proteomic profiling data analysis","Add-on to CellNOptR: Discretized time treatments","Integration of CellNOptR to add missing links","ODE add-on to CellNOptR","Complexome Profiling Analysis package","Data normalization by matrix raking","COSMOS (Causal Oriented Search of Multi-Omic Space)","Generate customized protein database from NGS data, with a focus on RNA-Seq data, for proteomics search","Using Mass Cytometry for Differential Abundance Analyses","Conditional Differential Analysis for Flow and Mass Cytometry Experiments","Differential expression using kernel-based score test","Marker Enrichment Modeling (MEM)","Tools for the Differential Analysis of Proteins Abundance with R","deltaGseg","Differential Enrichment analysis of Proteomics data","a tool to perform statistical analysis of differential protein expression for quantitative proteomics data.","Dynamic Programming Based Alignment of MS2 Chromatograms","Differential discovery in high-dimensional cytometry via high-resolution clustering","Diffusion scores on biological networks","Identification and Annotation of Protein Hotspot Residues","A DownStream Chemo-Proteomics Analysis Pipeline","Package to Draw Protein Schematics from Uniprot API output","Drug-Target Interactions","Ensemble of Gene Set Enrichment Analyses","Accelerated similarity searching of small molecules","Expression Weighted Celltype Enrichment","f-divergence Cutoff Index for Differential Expression Analysis in Transcriptomics and Proteomics","Mismatch Tolerant Maximum Common Substructure Searching","Generating Protein Sequences with Deep Generative Models","NanoString GeoMx Tools","Identification of Mutational Clusters in Proteins via a Graph Theoretical Approach.","Retrieve and analyze data from the Human Protein Atlas","Human Protein Atlas in R","Host-Pathogen Interaction Prediction","Profiling and Analyzing Intrinsically Disordered Proteins in R","Interlog protein network reconstruction by Mapping and Mining ANalysis","Augmenting Massively Parallel Cytometry Experiments Using Multivariate Non-Linear Regressions","R Interface with InterMine-Powered Databases","Identification of Protein Amino acid Clustering","Analysis and quantitation of isobarically tagged MSMS proteomics data","KinSwingR: network-based kinase activity prediction","Linear Models for Microarray Data","A correction of the local pooled error (LPE) method to replace the asymptotic variance adjustment with an unbiased adjustment\nbased on sample size.","Peak Detection for Mass Spectrometry data using wavelet-based algorithms","Shiny-based interactive data-quality exploration for omics data","Mean/Median-balanced quantile normalization","Multi-Contrast Gene Set Enrichment Analysis","Omics Data Integration Project","Effective mobility scale transformation of CE-MS(/MS) data","Quality Control of Protemics Data","Mass Spectrometry Data Backend for Mascot Generic Format (mgf) Files","Mass Spectrometry Data Backend for NIST msp Files","Mass Spectrometry Backend for Reading Thermo Fisher Scientific raw Files","Core Utils for Mass Spectrometry Data","Infrastructure for Mass Spectrometry Experiments","Imputation of label-free mass spectrometry peptides","Exploratory Data Analysis of LC-MS/MS data by spectral counts","LC-MS/MS Differential Expression Tests","Base Functions and Classes for Mass Spectrometry and Proteomics","Utilities for Exploration and Assessment of Confidence of LC-MSn Proteomics Identifications","Robust statistical inference for quantitative LC-MS proteomics","MsQuality - Quality metric calculation from Spectra objects","Protein Significance Analysis in DDA, SRM and DIA for Label-free or Label-based Proteomics Experiments","Import Data from Various Mass Spectrometry Signal Processing Tools to MSstats Format","LiP Significance Analysis in shotgun mass spectrometry-based proteomic experiments","Assay characterization: estimation of limit of blanc(LoB) and limit of detection(LOD)","Statistical Characterization of Post-translational Modifications","Longitudinal system suitability monitoring and quality control for proteomic experiments","A graphical user interface for MSstatsQC package","Simulation tool for optimal design of high-dimensional MS-based proteomics experiment","MSstats GUI for Statistical Anaylsis of Proteomics Experiments","Protein Significance Analysis in shotgun mass spectrometry-based proteomic experiments with tandem mass tag (TMT) labeling","An mzIdentML parser for R","parser for netCDF, mzXML, mzData and mzML and mzIdentML files (mass spectrometry data)","NanoString nCounter Tools","Evaluation of normalization methods and calculation of differential expression analysis statistics","Non-parametric analysis of response curves for thermal proteome profiling experiments","Omics Longitudinal Differential Analysis","Outlier Protein and Phosphosite Target Identifier","PAA (Protein Array Analyzer)","Prediction of pathway-related protein-protein interaction networks","a tool set for pathway based data integration and visualization","Integrative Pathway Analysis with Modern PCA Methodology and Gene Selection","Probe-level Expression Change Averaging","Parsing pepXML files and filter based on peptide FDR.","PGCA: An Algorithm to Link Protein Groups Created from MS/MS Data","Compensates for the bias introduced by median normalization in","A set of methods and tools for comprehensive analysis of phosphoproteomics data","Detect differential expression in microarray and proteomics datasets with the Power Law Global Error Model (PLGEM)","Local Pooled Error Test for Differential Expression with Paired High-throughput Data","Predicting Interactomes from Co-Elution","Generating SAM file for PSMs in shotgun proteomics data","Ciphergen SELDI-TOF Processing","Prediction of Oligomerization of Coiled Coil Proteins","Differential Abundance Analysis of Label-Free Mass Spectrometry Data","A unifying bioinformatics framework for spatial proteomics","Interactive visualisation of spatial proteomics data","Provides a GUI for DAPAR","Protease Mapping","Generation of customized protein variant databases from genomic variants, splice-junctions and manual sequences","Multi-Dataset Model-based Differential Expression Proteomics Analysis Platform","Generic infrastructure for Bioconductor mass spectrometry packages","Handling and Managing Peptide Spectrum Matches","Peptide Annotation and Data Visualization using Gviz","A Framework for Quality Control","Quantitative features for mass spectrometry data","Tools for quantitative proteomics data analysis","Identification of mutational clusters in protein quaternary structures.","Rhythmicity Analysis Incorporating Non-parametric Methods","Rank Product method for identifying differentially expressed genes with application in meta-analysis","Direct Access to Orbitrap Data and Beyond","A package for survival time prediction based on a piecewise baseline hazard Cox regression model.","Molecular Informatics Toolkit for Compound-Protein Interaction in Drug Discovery","Client for the Reactome Analysis Service for comparative multi-omics gene set analysis","RolDE: Robust longitudinal Differential Expression","Reproducibility-Optimized Test Statistic","R Interface to the ProteomeXchange Repository","Signature-based Clustering for Diagnostic Purposes","Thermal proximity co-aggregation with R","\"SBGNview: Data Analysis, Integration and Visualization on SBGN Pathways\"","Mass Spectrometry-Based Single-Cell Proteomics Data Analysis","Differential Abundant/Expression Analysis for Metabolomics, Proteomics and single-cell RNA sequencing Data","Statistical Modelling of AP-MS Data (SMAD)","Identification of Mutational Clusters in 3D Protein Space via Simulation.","Using Gaussian graphical structue learning estimation in generalized least squared regression for multivariate normal\nregression","specL - Prepare Peptide Spectrum Matches for Use in Targeted Proteomics","Spectra Infrastructure for Mass Spectrometry Data","Statistical Analysis of Molecular Profiles","SubCellBarCode: Integrated workflow for robust mapping and visualizing whole human spatial proteome","Transform and Filter SWATH Data for Statistical Packages","Label-free data analysis pipeline for optimal identification and quantitation","Diagnostic Plots to Evaluate the Target Decoy Approach","Time-Course Multi-Omics data integration","Investigation of Fragmentation Conditions in Top-Down Proteomics","Analyze thermal proteome profiling (TPP) experiments","Detection of ligand-protein interactions from 2D thermal profiles (DLPTP)","Feature-based variance-sensitive quantitative clustering","Well Plate Maker","Multiplexes Isobaric Mass Tagged-based Kinetics Data for Network Analysis","Signature discovery from omics data","Exposome and omic data associatin and integration analysis","Postprocessing and univariate analysis of omics data","Tools for Diagnostics and Corrections of Batch Effects in Proteomics","PCA, PLS(-DA) and OPLS(-DA) for multivariate analysis and feature selection of omics data","Preprocessing of FIA-HRMS data","Decomposing Heterogeneous Cohorts using Omic Data Profiling","Analysis of pulsed-SILAC quantitative proteomics data","Differential abundance analysis with a right censored covariate in high-dimensional cytometry","Phenotypic EMD for comparison of single-cell samples"],["1.17.0","1.43.0","1.17.0","1.3.0","1.37.0","1.31.0","1.19.0","1.45.0","1.13.0","3.1.0","1.31.0","1.45.0","1.37.0","3.51.0","1.39.0","1.37.0","1.49.0","1.41.0","1.39.0","1.41.0","1.7.0","1.7.0","1.7.0","1.39.0","1.23.0","1.7.0","1.5.0","1.3.0","1.31.0","1.39.0","1.21.0","1.17.3","2.7.0","1.19.0","1.19.0","1.19.0","1.21.0","1.19.0","1.7.0","1.27.0","1.39.0","1.7.0","1.29.0","1.41.0","1.3.0","3.3.0","1.41.0","1.17.0","1.41.1","1.5.0","1.9.15","1.19.0","1.9.0","1.21.0","1.43.0","1.45.0","1.17.0","3.55.2","1.59.0","1.65.0","1.7.2","2.11.0","1.11.4","6.23.3","1.3.0","1.7.0","1.7.2","1.3.1","1.5.0","1.11.3","1.1.1","1.9.0","1.37.0","1.37.0","2.25.1","1.33.0","1.7.0","0.99.6","4.7.1","1.9.2","1.5.1","1.7.0","2.1.1","2.17.0","1.19.0","1.13.0","1.1.7","2.7.0","1.37.0","2.33.0","1.7.0","1.17.0","1.11.0","1.15.0","1.13.0","1.33.0","1.29.0","1.39.0","1.15.0","1.35.0","1.33.0","1.23.0","1.23.0","1.9.1","1.71.0","1.59.0","1.15.0","1.33.0","1.75.0","2.27.0","1.13.0","1.39.0","2.9.0","1.31.0","1.1.0","1.5.0","1.17.0","1.31.0","1.3.0","1.33.0","1.37.0","1.9.2","1.17.0","1.31.0","1.33.0","3.25.0","1.7.0","1.45.0","1.35.0","1.13.0","1.3.0","1.27.0","2.7.0","1.15.0","1.9.0","1.13.0","1.9.0","1.19.0","1.15.1","1.37.0","1.17.0","1.33.3","1.9.6","1.29.0","1.15.0","1.29.0","2.23.0","1.5.0","1.11.0","1.21.0","3.27.0","1.15.2","1.1.0","1.9.0","1.17.0","1.27.0","1.21.0","1.1.2","1.15.0","1.31.0","1.25.0","1.15.0","1.13.0","1.7.0","1.15.0"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Package<\/th>\n      <th>Title<\/th>\n      <th>Version<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script><div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-5b3de2eb153354a2e086" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-5b3de2eb153354a2e086">{"x":{"filter":"none","vertical":false,"data":[["adductomicsR","apComplex","artMS","autonomics","bandle","BRAIN","CAMERA","Cardinal","cliqueMS","CompoundDb","ComPrAn","CONSTANd","cosmiq","customProDB","DAPAR","DEP","DEqMS","DIAlignR","Doscheda","flagme","fobitools","gaga","INDEED","IPO","isobar","IsoCorrectoR","IsoCorrectoRGUI","LOBSTAHS","MAIT","marr","MassArray","MassSpecWavelet","Metab","metabCombiner","MetaboAnnotation","MetaboCoreUtils","metabomxtr","metaMS","metapone","MetCirc","MetNet","MobilityTransformR","MQmetrics","MsBackendMassbank","MsBackendMgf","MsBackendMsp","MsBackendRawFileReader","MsCoreUtils","MsExperiment","MsFeatures","msImpute","msmsEDA","msmsTests","MSnbase","MSnID","MSPrep","msPurity","msqrob2","MsQuality","MSstats","MSstatsConvert","MSstatsLiP","MSstatsLOBD","MSstatsPTM","MSstatsQC","MSstatsQCgui","MSstatsSampleSize","MSstatsShiny","MSstatsTMT","mzID","mzR","ncGTW","peakPantheR","pepXMLTab","pgca","plgem","pmp","proBAMr","PROcess","proDA","pRoloc","Prostar","ProteoMM","ProtGenerics","PSMatch","ptairMS","qcmetrics","QFeatures","qmtools","qPLEXanalyzer","rawrr","Rdisop","rgoslin","Risa","RMassBank","rols","rpx","scDataviz","scp","SDAMS","SIMAT","SMAD","sparsenetgls","specL","Spectra","statTarget","SubCellBarCode","SWATH2stats","synapter","TargetDecoy","TargetSearch","topdownr","TPP","treekoR","wpm","xcms","yamss","biosigner","lipidr","phenomis","proBatch","ropls","proFIA","CATALYST"],["adductomicsR","apComplex","artMS","autonomics","bandle","BRAIN","CAMERA","Cardinal","cliqueMS","CompoundDb","ComPrAn","CONSTANd","cosmiq","customProDB","DAPAR","DEP","DEqMS","DIAlignR","Doscheda","flagme","fobitools","gaga","INDEED","IPO","isobar","IsoCorrectoR","IsoCorrectoRGUI","LOBSTAHS","MAIT","marr","MassArray","MassSpecWavelet","Metab","metabCombiner","MetaboAnnotation","MetaboCoreUtils","metabomxtr","metaMS","metapone","MetCirc","MetNet","MobilityTransformR","MQmetrics","MsBackendMassbank","MsBackendMgf","MsBackendMsp","MsBackendRawFileReader","MsCoreUtils","MsExperiment","MsFeatures","msImpute","msmsEDA","msmsTests","MSnbase","MSnID","MSPrep","msPurity","msqrob2","MsQuality","MSstats","MSstatsConvert","MSstatsLiP","MSstatsLOBD","MSstatsPTM","MSstatsQC","MSstatsQCgui","MSstatsSampleSize","MSstatsShiny","MSstatsTMT","mzID","mzR","ncGTW","peakPantheR","pepXMLTab","pgca","plgem","pmp","proBAMr","PROcess","proDA","pRoloc","Prostar","ProteoMM","ProtGenerics","PSMatch","ptairMS","qcmetrics","QFeatures","qmtools","qPLEXanalyzer","rawrr","Rdisop","rgoslin","Risa","RMassBank","rols","rpx","scDataviz","scp","SDAMS","SIMAT","SMAD","sparsenetgls","specL","Spectra","statTarget","SubCellBarCode","SWATH2stats","synapter","TargetDecoy","TargetSearch","topdownr","TPP","treekoR","wpm","xcms","yamss","biosigner","lipidr","phenomis","proBatch","ropls","proFIA","CATALYST"],["Processing of adductomic mass spectral datasets","Estimate protein complex membership using AP-MS protein data","Analytical R tools for Mass Spectrometry","Generifying and intuifying cross-platform omics analysis","An R package for the Bayesian analysis of differential subcellular localisation experiments","Baffling Recursive Algorithm for Isotope distributioN calculations","Collection of annotation related methods for mass spectrometry data","A mass spectrometry imaging toolbox for statistical analysis","Annotation of Isotopes, Adducts and Fragmentation Adducts for in-Source LC/MS Metabolomics Data","Creating and Using (Chemical) Compound Annotation Databases","Complexome Profiling Analysis package","Data normalization by matrix raking","cosmiq - COmbining Single Masses Into Quantities","Generate customized protein database from NGS data, with a focus on RNA-Seq data, for proteomics search","Tools for the Differential Analysis of Proteins Abundance with R","Differential Enrichment analysis of Proteomics data","a tool to perform statistical analysis of differential protein expression for quantitative proteomics data.","Dynamic Programming Based Alignment of MS2 Chromatograms","A DownStream Chemo-Proteomics Analysis Pipeline","Analysis of Metabolomics GC/MS Data","Tools For Manipulating FOBI Ontology","GaGa hierarchical model for high-throughput data analysis","Interactive Visualization of Integrated Differential Expression and Differential Network Analysis for Biomarker Candidate\nSelection Package","Automated Optimization of XCMS Data Processing parameters","Analysis and quantitation of isobarically tagged MSMS proteomics data","Correction for natural isotope abundance and tracer purity in MS and MS/MS data from stable isotope labeling experiments","Graphical User Interface for IsoCorrectoR","Lipid and Oxylipin Biomarker Screening through Adduct Hierarchy Sequences","Statistical Analysis of Metabolomic Data","Maximum rank reproducibility","Analytical Tools for MassArray Data","Peak Detection for Mass Spectrometry data using wavelet-based algorithms","Metab: An R Package for a High-Throughput Analysis of Metabolomics Data Generated by GC-MS.","Method for Combining LC-MS Metabolomics Feature Measurements","Utilities for Annotation of Metabolomics Data","Core Utils for Metabolomics Data","A package to run mixture models for truncated metabolomics data with normal or lognormal distributions","MS-based metabolomics annotation pipeline","Conducts pathway test of metabolomics data using a weighted permutation test","Navigating mass spectral similarity in high-resolution MS/MS metabolomics data metabolomics data","Inferring metabolic networks from untargeted high-resolution mass spectrometry data","Effective mobility scale transformation of CE-MS(/MS) data","Quality Control of Protemics Data","Mass Spectrometry Data Backend for MassBank record Files","Mass Spectrometry Data Backend for Mascot Generic Format (mgf) Files","Mass Spectrometry Data Backend for NIST msp Files","Mass Spectrometry Backend for Reading Thermo Fisher Scientific raw Files","Core Utils for Mass Spectrometry Data","Infrastructure for Mass Spectrometry Experiments","Functionality for Mass Spectrometry Features","Imputation of label-free mass spectrometry peptides","Exploratory Data Analysis of LC-MS/MS data by spectral counts","LC-MS/MS Differential Expression Tests","Base Functions and Classes for Mass Spectrometry and Proteomics","Utilities for Exploration and Assessment of Confidence of LC-MSn Proteomics Identifications","Package for Summarizing, Filtering, Imputing, and Normalizing Metabolomics Data","Automated Evaluation of Precursor Ion Purity for Mass Spectrometry Based Fragmentation in Metabolomics","Robust statistical inference for quantitative LC-MS proteomics","MsQuality - Quality metric calculation from Spectra objects","Protein Significance Analysis in DDA, SRM and DIA for Label-free or Label-based Proteomics Experiments","Import Data from Various Mass Spectrometry Signal Processing Tools to MSstats Format","LiP Significance Analysis in shotgun mass spectrometry-based proteomic experiments","Assay characterization: estimation of limit of blanc(LoB) and limit of detection(LOD)","Statistical Characterization of Post-translational Modifications","Longitudinal system suitability monitoring and quality control for proteomic experiments","A graphical user interface for MSstatsQC package","Simulation tool for optimal design of high-dimensional MS-based proteomics experiment","MSstats GUI for Statistical Anaylsis of Proteomics Experiments","Protein Significance Analysis in shotgun mass spectrometry-based proteomic experiments with tandem mass tag (TMT) labeling","An mzIdentML parser for R","parser for netCDF, mzXML, mzData and mzML and mzIdentML files (mass spectrometry data)","Alignment of LC-MS Profiles by Neighbor-wise Compound-specific Graphical Time Warping with Misalignment Detection","Peak Picking and Annotation of High Resolution Experiments","Parsing pepXML files and filter based on peptide FDR.","PGCA: An Algorithm to Link Protein Groups Created from MS/MS Data","Detect differential expression in microarray and proteomics datasets with the Power Law Global Error Model (PLGEM)","Peak Matrix Processing and signal batch correction for metabolomics datasets","Generating SAM file for PSMs in shotgun proteomics data","Ciphergen SELDI-TOF Processing","Differential Abundance Analysis of Label-Free Mass Spectrometry Data","A unifying bioinformatics framework for spatial proteomics","Provides a GUI for DAPAR","Multi-Dataset Model-based Differential Expression Proteomics Analysis Platform","Generic infrastructure for Bioconductor mass spectrometry packages","Handling and Managing Peptide Spectrum Matches","Pre-processing PTR-TOF-MS Data","A Framework for Quality Control","Quantitative features for mass spectrometry data","Quantitative Metabolomics Data Processing Tools","Tools for quantitative proteomics data analysis","Direct Access to Orbitrap Data and Beyond","Decomposition of Isotopic Patterns","Lipid Shorthand Name Parsing and Normalization","Converting experimental metadata from ISA-tab into Bioconductor data structures","Workflow to process tandem MS files and build MassBank records","An R interface to the Ontology Lookup Service","R Interface to the ProteomeXchange Repository","scDataviz: single cell dataviz and downstream analyses","Mass Spectrometry-Based Single-Cell Proteomics Data Analysis","Differential Abundant/Expression Analysis for Metabolomics, Proteomics and single-cell RNA sequencing Data","GC-SIM-MS data processing and alaysis tool","Statistical Modelling of AP-MS Data (SMAD)","Using Gaussian graphical structue learning estimation in generalized least squared regression for multivariate normal\nregression","specL - Prepare Peptide Spectrum Matches for Use in Targeted Proteomics","Spectra Infrastructure for Mass Spectrometry Data","Statistical Analysis of Molecular Profiles","SubCellBarCode: Integrated workflow for robust mapping and visualizing whole human spatial proteome","Transform and Filter SWATH Data for Statistical Packages","Label-free data analysis pipeline for optimal identification and quantitation","Diagnostic Plots to Evaluate the Target Decoy Approach","A package for the analysis of GC-MS metabolite profiling data","Investigation of Fragmentation Conditions in Top-Down Proteomics","Analyze thermal proteome profiling (TPP) experiments","Cytometry Cluster Hierarchy and Cellular-to-phenotype Associations","Well Plate Maker","LC-MS and GC-MS Data Analysis","Tools for high-throughput metabolomics","Signature discovery from omics data","Data Mining and Analysis of Lipidomics Datasets","Postprocessing and univariate analysis of omics data","Tools for Diagnostics and Corrections of Batch Effects in Proteomics","PCA, PLS(-DA) and OPLS(-DA) for multivariate analysis and feature selection of omics data","Preprocessing of FIA-HRMS data","Cytometry dATa anALYSis Tools"],["1.15.0","2.65.0","1.17.0","1.7.0","1.3.0","1.45.0","1.55.0","3.1.0","1.13.0","1.3.2","1.7.0","1.7.0","1.33.0","1.39.0","1.31.0","1.21.0","1.17.3","2.7.0","1.21.0","1.55.0","1.7.0","2.45.0","2.13.0","1.25.0","1.45.0","1.17.0","1.15.0","1.25.0","1.33.0","1.9.0","1.51.0","1.65.0","1.33.0","1.9.0","1.3.0","1.7.0","1.33.0","1.35.0","1.5.0","1.29.1","1.17.1","1.3.0","1.7.0","1.7.2","1.7.2","1.3.1","1.5.0","1.11.3","1.1.1","1.7.0","1.9.0","1.37.0","1.37.0","2.25.1","1.33.0","1.9.0","1.25.0","1.7.0","0.99.6","4.7.1","1.9.2","1.5.1","1.7.0","2.1.1","2.17.0","1.19.0","1.13.0","1.1.7","2.7.0","1.37.0","2.33.0","1.13.0","1.13.1","1.33.0","1.23.0","1.71.0","1.11.0","1.33.0","1.75.0","1.13.0","1.39.0","1.31.0","1.17.0","1.31.0","1.3.0","1.7.0","1.37.0","1.9.2","1.3.0","1.17.0","1.7.0","1.59.0","1.3.0","1.41.0","3.9.0","2.27.0","2.7.0","1.9.0","1.9.0","1.19.0","1.31.0","1.15.1","1.17.0","1.33.3","1.9.6","1.29.0","1.15.0","1.29.0","2.23.0","1.5.0","2.1.0","1.21.0","3.27.0","1.7.0","1.9.0","3.21.0","1.25.0","1.27.0","2.13.0","1.1.2","1.15.0","1.31.0","1.25.0","1.23.0"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Package<\/th>\n      <th>Title<\/th>\n      <th>Version<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script><div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-9348124dec52196a3362" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-9348124dec52196a3362">{"x":{"filter":"none","vertical":false,"data":[["adductData","BioPlex","CardinalWorkflows","CluMSIDdata","DAPARdata","faahKO","gcspikelite","metaMSdata","msdata","msPurityData","msqc1","mtbls2","NestLink","ProData","pRolocdata","ptairData","PtH2O2lipids","qPLEXdata","RMassBankData","scpdata","synapterdata","tartare","topdownrdata","RforProteomics","plasFIA"],["adductData","BioPlex","CardinalWorkflows","CluMSIDdata","DAPARdata","faahKO","gcspikelite","metaMSdata","msdata","msPurityData","msqc1","mtbls2","NestLink","ProData","pRolocdata","ptairData","PtH2O2lipids","qPLEXdata","RMassBankData","scpdata","synapterdata","tartare","topdownrdata","RforProteomics","plasFIA"],["Data from untargeted MS of modifications to Cys34 of serum albumin","R-side access to BioPlex protein-protein interaction data","Datasets and workflows for the Cardinal mass spectrometry imaging package","Data for the CluMSID package","Data accompanying the DAPAR and Prostar packages","Saghatelian et al. (2004) FAAH knockout LC/MS data","Spike-in data for GC/MS data and methods within flagme","Example CDF data for the metaMS package","Various Mass Spectrometry raw data example files","Fragmentation spectral libraries and data to test the msPurity package","Sigma mix MSQC1 data","MetaboLights MTBLS2: Comparative LC/MS-based profiling of silver nitrate-treated Arabidopsis thaliana leaves of wild-type and\ncyp79B2 cyp79B3 double knockout plants. Böttcher et al. (2004)","NestLink an R data package to guide through Engineered Peptide Barcodes for In-Depth Analyzes of Binding Protein Ensembles","SELDI-TOF data of Breast cancer samples","Data accompanying the pRoloc package","PTR-TOF-MS volatolomics raw datasets from exhaled air and cell culture headspace","P. tricornutum HPLC-ESI-MS Lipid Data from van Creveld et al. (2015)","Data accompanying qPLEXanalyzer package","Test dataset for RMassBank","Single-Cell Proteomics Data Package","Data accompanying the synapter package","Raw ground spectra recorded on Thermo Fisher Scientific mass spectrometers","Example Files for the topdownr R Package","Companion package to the 'Using R and Bioconductor for proteomics data analysis' publication","FIA-HRMS plasma dataset"],["1.15.0","1.5.3","1.31.0","1.15.0","1.29.0","1.39.0","1.37.0","1.35.0","0.39.0","1.27.0","1.27.1","1.29.0","1.15.0","1.37.0","1.37.0","1.7.0","1.25.0","1.17.0","1.37.0","1.7.0","1.37.0","1.13.0","1.21.0","1.37.1","1.27.0"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Package<\/th>\n      <th>Title<\/th>\n      <th>Version<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script><p>The tables can easily be generated with the
<code>proteomicsPackages</code>, <code>massSpectrometryPackages</code>
and <code>massSpectrometryDataPackages</code> functions. The respective
package tables can then be interactively explored using the
<code>display</code> function.</p>
<div class="sourceCode" id="cb210"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/proteomicsPackages.html">proteomicsPackages</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu">display</span><span class="op">(</span><span class="va">pp</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="other-cran-packages">Other CRAN packages<a class="anchor" aria-label="anchor" href="#other-cran-packages"></a>
</h3>
<p>The CRAN task view on <a href="http://cran.r-project.org/web/views/ChemPhys.html" class="external-link">Chemometrics
and Computational Physics</a> is another useful ressource listing
additional packages, including a set of packages for mass spectrometry
and proteomics, some of which are illustrated in this document.</p>
<ul>
<li>
<em><a href="https://CRAN.R-project.org/package=MALDIquant" class="external-link">MALDIquant</a></em>
provides tools for quantitative analysis of MALDI-TOF mass spectrometry
data, with support for baseline correction, peak detection and plotting
of mass spectra.</li>
<li>
<em><a href="https://CRAN.R-project.org/package=OrgMassSpecR" class="external-link">OrgMassSpecR</a></em>
is for organic/biological mass spectrometry, with a focus on graphical
display, quantification using stable isotope dilution, and protein
hydrogen/deuterium exchange experiments.</li>
<li>
<em><a href="https://CRAN.R-project.org/package=FTICRMS" class="external-link">FTICRMS</a></em>
provides functions for Analyzing Fourier Transform-Ion Cyclotron
Resonance Mass Spectrometry Data.</li>
<li>
<em><a href="https://CRAN.R-project.org/package=titan" class="external-link">titan</a></em> provides
a GUI to analyze mass spectrometric data on the relative abundance of
two substances from a titration series.</li>
<li>
<em><a href="https://CRAN.R-project.org/package=digeR" class="external-link">digeR</a></em> provides
a GUI interface for analysing 2D DIGE data. It allows to perform
correlation analysis, score plot, classification, feature selection and
power analysis for 2D DIGE experiment data.</li>
<li>
<em><a href="https://CRAN.R-project.org/package=protViz" class="external-link">protViz</a></em> helps
with quality checks, visualizations and analysis of mass spectrometry
data, coming from proteomics experiments. The package is developed,
tested and used at the Functional Genomics Center Zurich.</li>
</ul>
<p>Suggestions for additional R packages are welcome and will be added
to the vignette. Please send suggestions and possibly a short
description and/or a example utilisation with code to the RforProteomics
package maintainer. The only requirement is that the package must be
available on an official package channel (CRAN, Bioconductor, R-forge,
Omegahat), i.e. not only available through a personal web page.</p>
</div>
</div>
<div class="section level2">
<h2 id="sec:sessionInfo">Session information<a class="anchor" aria-label="anchor" href="#sec:sessionInfo"></a>
</h2>
<p>All software and respective versions used in this document, as
returned by <code><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo()</a></code> are detailed below.</p>
<pre><code><span><span class="co">## R Under development (unstable) (2023-01-11 r83598)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.1 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] grid      stats4    stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">## [8] methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] MSnID_1.33.0           cleaver_1.37.0         OrgMassSpecR_0.5-3    </span></span>
<span><span class="co">##  [4] msdata_0.39.0          reshape2_1.4.4         ggplot2_3.4.0         </span></span>
<span><span class="co">##  [7] RColorBrewer_1.1-3     RforProteomics_1.37.2  biomaRt_2.55.0        </span></span>
<span><span class="co">## [10] Rdisop_1.59.0          GO.db_3.16.0           org.Hs.eg.db_3.16.0   </span></span>
<span><span class="co">## [13] AnnotationDbi_1.61.0   BRAIN_1.45.0           lattice_0.20-45       </span></span>
<span><span class="co">## [16] Biostrings_2.67.0      GenomeInfoDb_1.35.12   XVector_0.39.0        </span></span>
<span><span class="co">## [19] IRanges_2.33.0         PolynomF_2.0-5         hpar_1.41.1           </span></span>
<span><span class="co">## [22] rols_2.27.0            MALDIquantForeign_0.13 MALDIquant_1.22       </span></span>
<span><span class="co">## [25] rpx_2.5.1              mzID_1.37.0            MSnbase_2.25.1        </span></span>
<span><span class="co">## [28] ProtGenerics_1.31.0    S4Vectors_0.37.3       Biobase_2.59.0        </span></span>
<span><span class="co">## [31] BiocGenerics_0.45.0    mzR_2.33.0             Rcpp_1.0.9            </span></span>
<span><span class="co">## [34] BiocManager_1.30.19    DT_0.27                BiocStyle_2.27.0      </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] later_1.3.0                   bitops_1.0-7                 </span></span>
<span><span class="co">##   [3] filelock_1.0.2                tibble_3.1.8                 </span></span>
<span><span class="co">##   [5] R.oo_1.25.0                   preprocessCore_1.61.0        </span></span>
<span><span class="co">##   [7] graph_1.77.1                  XML_3.99-0.13                </span></span>
<span><span class="co">##   [9] lifecycle_1.0.3               doParallel_1.0.17            </span></span>
<span><span class="co">##  [11] rprojroot_2.0.3               MASS_7.3-58.1                </span></span>
<span><span class="co">##  [13] crosstalk_1.2.0               magrittr_2.0.3               </span></span>
<span><span class="co">##  [15] limma_3.55.2                  sass_0.4.4                   </span></span>
<span><span class="co">##  [17] rmarkdown_2.20                jquerylib_0.1.4              </span></span>
<span><span class="co">##  [19] yaml_2.3.6                    httpuv_1.6.8                 </span></span>
<span><span class="co">##  [21] RUnit_0.4.32                  MsCoreUtils_1.11.3           </span></span>
<span><span class="co">##  [23] DBI_1.1.3                     zlibbioc_1.45.0              </span></span>
<span><span class="co">##  [25] R.cache_0.16.0                GenomicRanges_1.51.4         </span></span>
<span><span class="co">##  [27] purrr_1.0.1                   R.utils_2.12.2               </span></span>
<span><span class="co">##  [29] RCurl_1.98-1.9                rappdirs_0.3.3               </span></span>
<span><span class="co">##  [31] GenomeInfoDbData_1.2.9        readMzXmlData_2.8.2          </span></span>
<span><span class="co">##  [33] pkgdown_2.0.7.9000            ncdf4_1.21                   </span></span>
<span><span class="co">##  [35] codetools_0.2-18              DelayedArray_0.25.0          </span></span>
<span><span class="co">##  [37] xml2_1.3.3                    tidyselect_1.2.0             </span></span>
<span><span class="co">##  [39] farver_2.1.1                  matrixStats_0.63.0           </span></span>
<span><span class="co">##  [41] BiocFileCache_2.7.1           base64enc_0.1-3              </span></span>
<span><span class="co">##  [43] jsonlite_1.8.4                ellipsis_0.3.2               </span></span>
<span><span class="co">##  [45] iterators_1.0.14              systemfonts_1.0.4            </span></span>
<span><span class="co">##  [47] foreach_1.5.2                 tools_4.3.0                  </span></span>
<span><span class="co">##  [49] progress_1.2.2                ragg_1.2.5                   </span></span>
<span><span class="co">##  [51] readBrukerFlexData_1.9.0      glue_1.6.2                   </span></span>
<span><span class="co">##  [53] xfun_0.36                     MatrixGenerics_1.11.0        </span></span>
<span><span class="co">##  [55] dplyr_1.0.10                  withr_2.5.0                  </span></span>
<span><span class="co">##  [57] fastmap_1.1.0                 fansi_1.0.3                  </span></span>
<span><span class="co">##  [59] digest_0.6.31                 R6_2.5.1                     </span></span>
<span><span class="co">##  [61] mime_0.12                     textshaping_0.3.6            </span></span>
<span><span class="co">##  [63] colorspace_2.0-3              RSQLite_2.2.20               </span></span>
<span><span class="co">##  [65] R.methodsS3_1.8.2             utf8_1.2.2                   </span></span>
<span><span class="co">##  [67] generics_0.1.3                data.table_1.14.6            </span></span>
<span><span class="co">##  [69] prettyunits_1.1.1             httr_1.4.4                   </span></span>
<span><span class="co">##  [71] htmlwidgets_1.6.1             pkgconfig_2.0.3              </span></span>
<span><span class="co">##  [73] gtable_0.3.1                  blob_1.2.3                   </span></span>
<span><span class="co">##  [75] impute_1.73.0                 htmltools_0.5.4              </span></span>
<span><span class="co">##  [77] bookdown_0.32                 RBGL_1.75.0                  </span></span>
<span><span class="co">##  [79] clue_0.3-63                   scales_1.2.1                 </span></span>
<span><span class="co">##  [81] png_0.1-8                     knitr_1.41                   </span></span>
<span><span class="co">##  [83] curl_5.0.0                    biocViews_1.67.2             </span></span>
<span><span class="co">##  [85] cachem_1.0.6                  stringr_1.5.0                </span></span>
<span><span class="co">##  [87] BiocVersion_3.17.1            parallel_4.3.0               </span></span>
<span><span class="co">##  [89] vsn_3.67.0                    desc_1.4.2                   </span></span>
<span><span class="co">##  [91] pillar_1.8.1                  vctrs_0.5.1                  </span></span>
<span><span class="co">##  [93] pcaMethods_1.91.0             promises_1.2.0.1             </span></span>
<span><span class="co">##  [95] dbplyr_2.3.0                  xtable_1.8-4                 </span></span>
<span><span class="co">##  [97] cluster_2.1.4                 evaluate_0.20                </span></span>
<span><span class="co">##  [99] cli_3.6.0                     compiler_4.3.0               </span></span>
<span><span class="co">## [101] rlang_1.0.6                   crayon_1.5.2                 </span></span>
<span><span class="co">## [103] labeling_0.4.2                affy_1.77.0                  </span></span>
<span><span class="co">## [105] plyr_1.8.8                    fs_1.5.2                     </span></span>
<span><span class="co">## [107] stringi_1.7.12                BiocParallel_1.33.9          </span></span>
<span><span class="co">## [109] assertthat_0.2.1              munsell_0.5.0                </span></span>
<span><span class="co">## [111] Matrix_1.5-3                  ExperimentHub_2.7.0          </span></span>
<span><span class="co">## [113] hms_1.1.2                     bit64_4.0.5                  </span></span>
<span><span class="co">## [115] KEGGREST_1.39.0               shiny_1.7.4                  </span></span>
<span><span class="co">## [117] SummarizedExperiment_1.29.1   interactiveDisplayBase_1.37.0</span></span>
<span><span class="co">## [119] highr_0.10                    AnnotationHub_3.7.0          </span></span>
<span><span class="co">## [121] memoise_2.0.1                 affyio_1.69.0                </span></span>
<span><span class="co">## [123] bslib_0.4.2                   bit_4.0.5</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-synapter" class="csl-entry">
Bond, Nicholas J., Pavel V. Shliaha, Kathryn S. Lilley, and Laurent
Gatto. 2013. <span>“Improving Qualitative and Quantitative Performance
for Label Free Proteomics.”</span> <em>J. Proteome Res.</em> <a href="https://doi.org/10.1021/pr300776t" class="external-link">https://doi.org/10.1021/pr300776t</a>.
</div>
<div id="ref-Breitwieser2011" class="csl-entry">
Breitwieser, F P, A Müller, L Dayon, T Köcher, A Hainard, P Pichler, U
Schmidt-Erfurth, et al. 2011. <span>“General Statistical Modeling of
Data from Protein Relative Expression Isobaric Tags.”</span> <em>J
Proteome Res</em> 10 (6): 2758–66. <a href="https://doi.org/10.1021/pr1012784" class="external-link">https://doi.org/10.1021/pr1012784</a>.
</div>
<div id="ref-Chambers2012" class="csl-entry">
Chambers, M C, B Maclean, R Burke, D Amodei, D L Ruderman, S Neumann, L
Gatto, et al. 2012. <span>“A Cross-Platform Toolkit for Mass
Spectrometry and Proteomics.”</span> <em>Nat Biotechnol</em> 30 (10):
918–20. <a href="https://doi.org/10.1038/nbt.2377" class="external-link">https://doi.org/10.1038/nbt.2377</a>.
</div>
<div id="ref-TargetSearch2009" class="csl-entry">
Cuadros-Inostroza, Alvaro, Camila Caldana, Henning Redestig, Jan Lisec,
Hugo Pena-Cortes, Lothar Willmitzer, and Matthew A Hannah. 2009.
<span>“<span>TargetSearch</span> - a <span>Bioconductor</span> Package
for the Efficient Pre-Processing of <span>GC-MS</span> Metabolite
Profiling Data.”</span> <em>BMC Bioinformatics</em> 10: 428. <a href="https://doi.org/10.1186/1471-2105-10-428" class="external-link">https://doi.org/10.1186/1471-2105-10-428</a>.
</div>
<div id="ref-Durinck2005" class="csl-entry">
Durinck, S, Y Moreau, A Kasprzyk, S Davis, B De Moor, A Brazma, and W
Huber. 2005. <span>“BioMart and Bioconductor: A Powerful Link Between
Biological Databases and Microarray Data Analysis.”</span>
<em>Bioinformatics</em> 21 (16): 3439–40. <a href="https://doi.org/10.1093/bioinformatics/bti525" class="external-link">https://doi.org/10.1093/bioinformatics/bti525</a>.
</div>
<div id="ref-R4Prot2013" class="csl-entry">
Gatto, Laurent, and Andy Christoforou. 2013. <span>“Using <span>R</span>
and Bioconductor for Proteomics Data Analysis.”</span> <em>BBA -
Proteins and Proteomics</em>. <a href="https://doi.org/10.1016/j.bbapap.2013.04.032" class="external-link">https://doi.org/10.1016/j.bbapap.2013.04.032</a>.
</div>
<div id="ref-Gatto:2015" class="csl-entry">
Gatto, L, L M Breckels, T Naake, and S Gibb. 2015. <span>“<span class="nocase">Visualisation of proteomics data using R and
Bioconductor</span>.”</span> <em>Proteomics</em>, February. <a href="https://doi.org/10.1002/pmic.201400392" class="external-link">https://doi.org/10.1002/pmic.201400392</a>.
</div>
<div id="ref-Gatto2012" class="csl-entry">
Gatto, L, and K S Lilley. 2012. <span>“<span>MSnbase</span> – an
<span>R/Bioconductor</span> Package for Isobaric Tagged Mass
Spectrometry Data Visualization, Processing and Quantitation.”</span>
<em>Bioinformatics</em> 28 (2): 288–89. <a href="https://doi.org/10.1093/bioinformatics/btr645" class="external-link">https://doi.org/10.1093/bioinformatics/btr645</a>.
</div>
<div id="ref-Gentleman2004" class="csl-entry">
Gentleman, Robert C., Vincent J. Carey, Douglas M. Bates, Ben Bolstad,
Marcel Dettling, Sandrine Dudoit, Byron Ellis, et al. 2004.
<span>“<span>Bioconductor</span>: Open Software Development for
Computational Biology and Bioinformatics.”</span> <em>Genome Biol</em> 5
(10): –80. <a href="https://doi.org/10.1186/gb-2004-5-10-r80" class="external-link">https://doi.org/10.1186/gb-2004-5-10-r80</a>.
</div>
<div id="ref-Gibb2012" class="csl-entry">
Gibb, S, and K Strimmer. 2012. <span>“<span>MALDIquant</span>: A
Versatile <span>R</span> Package for the Analysis of Mass Spectrometry
Data.”</span> <em>Bioinformatics</em> 28 (17): 2270–71. <a href="https://doi.org/10.1093/bioinformatics/bts447" class="external-link">https://doi.org/10.1093/bioinformatics/bts447</a>.
</div>
<div id="ref-Piehowski2013a" class="csl-entry">
Piehowski, Paul D., Vladislav A. Petyuk, John D. Sandoval, Kristin E.
Burnum, Gary R. Kiebel, Matthew E. Monroe, Gordon A. Anderson, David G
Camp 2nd, and Richard D. Smith. 2013. <span>“STEPS: A Grid Search
Methodology for Optimized Peptide Identification Filtering of MS/MS
Database Search Results.”</span> <em>Proteomics</em> 13 (5): 766–70. <a href="https://doi.org/10.1002/pmic.201200096" class="external-link">https://doi.org/10.1002/pmic.201200096</a>.
</div>
<div id="ref-Smith2006" class="csl-entry">
Smith, C A, E J Want, G O’Maille, R Abagyan, and G Siuzdak. 2006.
<span>“<span>XCMS</span>: Processing Mass Spectrometry Data for
Metabolite Profiling Using Nonlinear Peak Alignment, Matching, and
Identification.”</span> <em>Anal Chem</em> 78 (3): 779–87. <a href="https://doi.org/10.1021/ac051437y" class="external-link">https://doi.org/10.1021/ac051437y</a>.
</div>
<div id="ref-Uhlen2005" class="csl-entry">
Uhlén, Mathias, Erik Björling, Charlotta Agaton, Cristina Al-Khalili A.
Szigyarto, Bahram Amini, Elisabet Andersen, Ann-Catrin C. Andersson, et
al. 2005. <span>“<span class="nocase">A human protein atlas for normal
and cancer tissues based on antibody proteomics.</span>”</span>
<em>Molecular &amp; Cellular Proteomics : MCP</em> 4 (12): 1920–32. <a href="https://doi.org/10.1074/mcp.M500279-MCP200" class="external-link">https://doi.org/10.1074/mcp.M500279-MCP200</a>.
</div>
<div id="ref-Vizcaino2014" class="csl-entry">
Vizcaino, Juan A, Eric W Deutsch, Rui Wang, Attila Csordas, Florian
Reisinger, Daniel Rios, Jose A Dianes, et al. 2014.
<span>“<span>ProteomeXchange</span> Provides Globally Coordinated
Proteomics Data Submission and Dissemination.”</span> <em>Nat
Biotech</em> 32: 223–26. <a href="https://doi.org/10.1038/nbt.2839" class="external-link">https://doi.org/10.1038/nbt.2839</a>.
</div>
</div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr>
<ol>
<li id="fn1"><p>Here, we specify <code>mzTab</code> format version 0.9.
Recent files have been generated according to the latest specifications,
version 1.0, and the <code>version</code> does not need to be specified
explicitly.<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Laurent Gatto.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
