<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Visualisation of proteomics data using R and Bioconductor • RforProteomics</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Visualisation of proteomics data using R and Bioconductor">
<meta property="og:description" content="RforProteomics">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RforProteomics</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.37.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/RforProteomics.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/RProtVis.html">Visualisation of proteomics data using R and Bioconductor</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="RProtVis_files/htmlwidgets-1.6.1/htmlwidgets.js"></script><link href="RProtVis_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="RProtVis_files/datatables-binding-0.26/datatables.js"></script><link href="RProtVis_files/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="RProtVis_files/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="RProtVis_files/dt-core-1.12.1/js/jquery.dataTables.min.js"></script><link href="RProtVis_files/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="RProtVis_files/crosstalk-1.2.0/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Visualisation of proteomics data using R and
Bioconductor</h1>
                        <h4 data-toc-skip class="author">Laurent
Gatto</h4>
                                    <h4 data-toc-skip class="author">Lisa
Breckels</h4>
                                    <h4 data-toc-skip class="author">Sebastian
Gibb</h4>
                        
      
      
      <div class="hidden name"><code>RProtVis.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This is the companion vignette to the ‘Using R and Bioconductor
      for proteomics data analysis’ manuscript that presents an overview
      of R and Bioconductor software for mass spectrometry and
      proteomics data. It provides the code to reproduce the figures in
      the paper.</p>
    </div>
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette illustrates existing and Bioconductor infrastructure
for the visualisation of mass spectrometry and proteomics data. The code
details the visualisations presented in</p>
<blockquote>
<p>Gatto L, Breckels LM, Naake T, Gibb S. Visualisation of proteomics
data using R and Bioconductor. Proteomics. 2015 Feb 18. doi:
10.1002/pmic.201400392. PubMed PMID: <a href="http://www.ncbi.nlm.nih.gov/pubmed/25690415" class="external-link">25690415</a>.</p>
</blockquote>
<blockquote>
<p><strong>NB</strong>: I you are interested in R packages for mass
spectrometry-based proteomics and metabolomics, see also the <a href="https://www.rformassspectrometry.org/" class="external-link">R for Mass Spectrometry
initiative</a> packages and the <a href="https://rformassspectrometry.github.io/docs/" class="external-link">tutorial
book</a></p>
</blockquote>
<div class="section level3">
<h3 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<ul>
<li>CRAN Task View: Graphic Displays &amp; Dynamic Graphics &amp;
Graphic Devices &amp; Visualization: <a href="http://cran.r-project.org/web/views/Graphics.html" class="external-link uri">http://cran.r-project.org/web/views/Graphics.html</a>
</li>
<li>CRAN Task View: Web Technologies and Services: <a href="http://cran.r-project.org/web/views/WebTechnologies.html" class="external-link uri">http://cran.r-project.org/web/views/WebTechnologies.html</a>
</li>
<li>ggplot2 <a href="http://link.springer.com/book/10.1007%2F978-0-387-98141-3" class="external-link">book</a>
(syntax is slightly outdated) (<a href="http://ggplot2.org/book/" class="external-link">code</a>), <a href="http://ggplot2.org/" class="external-link">web page</a> and <a href="http://docs.ggplot2.org/current/" class="external-link">on-line docs</a>
</li>
<li>lattice <a href="http://lmdvr.r-forge.r-project.org/figures/figures.html" class="external-link">book</a>
and <a href="http://lattice.r-forge.r-project.org/" class="external-link">web page</a>
</li>
<li>
<em>R Graphics</em> <a href="https://www.stat.auckland.ac.nz/~paul/RG2e/" class="external-link">book</a>
</li>
<li>
<a href="http://www.cookbook-r.com/Graphs/" class="external-link">R Cookbook</a> and <a href="http://shop.oreilly.com/product/0636920023135.do" class="external-link">R Graphics
Cookbook</a>
</li>
</ul>
</div>
<div class="section level3">
<h3 id="relevant-packages">Relevant packages<a class="anchor" aria-label="anchor" href="#relevant-packages"></a>
</h3>
<p>There are currently 165 <a href="http://bioconductor.org/packages/devel/BiocViews.html#___Proteomics" class="external-link">Proteomics</a>
and 124 <a href="http://bioconductor.org/packages/devel/BiocViews.html#___MassSpectrometry" class="external-link">MassSpectrometry</a>
packages in Bioconductor version 3.17. Other non-Bioconductor packages
are described in the <code>r Biocexptpkg("RforProteomics")</code>
vignette (open it in R with <code><a href="../reference/RforProteomics.html">RforProteomics()</a></code> or read it <a href="file:///home/lg390/dev/00_github/RforProteomics/docs/articles/RforProteomics.html">online</a>.)</p>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-91af88f530b5048b4824" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-91af88f530b5048b4824">{"x":{"filter":"none","vertical":false,"data":[["artMS","ASEB","AssessORF","bandle","bioassayR","biobroom","BioNetStat","BRAIN","calm","Cardinal","CausalR","CellNOptR","ChemmineOB","ChemmineR","cisPath","cleaver","clippda","CNORdt","CNORfeeder","CNORode","ComPrAn","CONSTANd","cosmosR","customProDB","cydar","CytoGLMM","cytoKernel","cytoMEM","DAPAR","deltaGseg","DEP","DEqMS","DIAlignR","diffcyt","diffuStats","DominoEffect","Doscheda","drawProteins","drugTargetInteractions","EGSEA","eiR","EWCE","fCI","fmcsR","GenProSeq","GeomxTools","GraphPAC","HPAanalyze","hpar","HPiP","idpr","IMMAN","infinityFlow","InterMineR","iPAC","isobar","KinSwingR","limma","LPEadj","MassSpecWavelet","MatrixQCvis","MBQN","mitch","mixOmics","MobilityTransformR","MQmetrics","MsBackendMgf","MsBackendMsp","MsBackendRawFileReader","MsCoreUtils","MsExperiment","msImpute","msmsEDA","msmsTests","MSnbase","MSnID","msqrob2","MsQuality","MSstats","MSstatsConvert","MSstatsLiP","MSstatsLOBD","MSstatsPTM","MSstatsQC","MSstatsQCgui","MSstatsSampleSize","MSstatsShiny","MSstatsTMT","mzID","mzR","NanoStringNCTools","NormalyzerDE","NPARC","OmicsLonDA","oppti","PAA","Path2PPI","pathview","pathwayPCA","PECA","pepXMLTab","pgca","phosphonormalizer","PhosR","plgem","PLPE","PrInCE","proBAMr","PROcess","procoil","proDA","pRoloc","pRolocGUI","Prostar","proteasy","ProteoDisco","ProteoMM","ProtGenerics","PSMatch","Pviz","qcmetrics","QFeatures","qPLEXanalyzer","QuartPAC","rain","RankProd","rawrr","RCASPAR","Rcpi","ReactomeGSA","RolDE","ROTS","rpx","rScudo","Rtpca","SBGNview","scp","SDAMS","SMAD","SpacePAC","sparsenetgls","specL","Spectra","statTarget","SubCellBarCode","SWATH2stats","synapter","TargetDecoy","timeOmics","topdownr","TPP","TPP2D","vsclust","wpm","XINA","biosigner","omicRexposome","phenomis","proBatch","ropls","proFIA","deco","pulsedSilac","censcyt","phemd"],["artMS","ASEB","AssessORF","bandle","bioassayR","biobroom","BioNetStat","BRAIN","calm","Cardinal","CausalR","CellNOptR","ChemmineOB","ChemmineR","cisPath","cleaver","clippda","CNORdt","CNORfeeder","CNORode","ComPrAn","CONSTANd","cosmosR","customProDB","cydar","CytoGLMM","cytoKernel","cytoMEM","DAPAR","deltaGseg","DEP","DEqMS","DIAlignR","diffcyt","diffuStats","DominoEffect","Doscheda","drawProteins","drugTargetInteractions","EGSEA","eiR","EWCE","fCI","fmcsR","GenProSeq","GeomxTools","GraphPAC","HPAanalyze","hpar","HPiP","idpr","IMMAN","infinityFlow","InterMineR","iPAC","isobar","KinSwingR","limma","LPEadj","MassSpecWavelet","MatrixQCvis","MBQN","mitch","mixOmics","MobilityTransformR","MQmetrics","MsBackendMgf","MsBackendMsp","MsBackendRawFileReader","MsCoreUtils","MsExperiment","msImpute","msmsEDA","msmsTests","MSnbase","MSnID","msqrob2","MsQuality","MSstats","MSstatsConvert","MSstatsLiP","MSstatsLOBD","MSstatsPTM","MSstatsQC","MSstatsQCgui","MSstatsSampleSize","MSstatsShiny","MSstatsTMT","mzID","mzR","NanoStringNCTools","NormalyzerDE","NPARC","OmicsLonDA","oppti","PAA","Path2PPI","pathview","pathwayPCA","PECA","pepXMLTab","pgca","phosphonormalizer","PhosR","plgem","PLPE","PrInCE","proBAMr","PROcess","procoil","proDA","pRoloc","pRolocGUI","Prostar","proteasy","ProteoDisco","ProteoMM","ProtGenerics","PSMatch","Pviz","qcmetrics","QFeatures","qPLEXanalyzer","QuartPAC","rain","RankProd","rawrr","RCASPAR","Rcpi","ReactomeGSA","RolDE","ROTS","rpx","rScudo","Rtpca","SBGNview","scp","SDAMS","SMAD","SpacePAC","sparsenetgls","specL","Spectra","statTarget","SubCellBarCode","SWATH2stats","synapter","TargetDecoy","timeOmics","topdownr","TPP","TPP2D","vsclust","wpm","XINA","biosigner","omicRexposome","phenomis","proBatch","ropls","proFIA","deco","pulsedSilac","censcyt","phemd"],["Analytical R tools for Mass Spectrometry","Predict Acetylated Lysine Sites","Assess Gene Predictions Using Proteomics and Evolutionary Conservation","An R package for the Bayesian analysis of differential subcellular localisation experiments","Cross-target analysis of small molecule bioactivity","Turn Bioconductor objects into tidy data frames","Biological Network Analysis","Baffling Recursive Algorithm for Isotope distributioN calculations","Covariate Assisted Large-scale Multiple testing","A mass spectrometry imaging toolbox for statistical analysis","Causal network analysis methods","Training of boolean logic models of signalling networks using prior knowledge networks and perturbation data","R interface to a subset of OpenBabel functionalities","Cheminformatics Toolkit for R","Visualization and management of the protein-protein interaction networks.","Cleavage of Polypeptide Sequences","A package for the clinical proteomic profiling data analysis","Add-on to CellNOptR: Discretized time treatments","Integration of CellNOptR to add missing links","ODE add-on to CellNOptR","Complexome Profiling Analysis package","Data normalization by matrix raking","COSMOS (Causal Oriented Search of Multi-Omic Space)","Generate customized protein database from NGS data, with a focus on RNA-Seq data, for proteomics search","Using Mass Cytometry for Differential Abundance Analyses","Conditional Differential Analysis for Flow and Mass Cytometry Experiments","Differential expression using kernel-based score test","Marker Enrichment Modeling (MEM)","Tools for the Differential Analysis of Proteins Abundance with R","deltaGseg","Differential Enrichment analysis of Proteomics data","a tool to perform statistical analysis of differential protein expression for quantitative proteomics data.","Dynamic Programming Based Alignment of MS2 Chromatograms","Differential discovery in high-dimensional cytometry via high-resolution clustering","Diffusion scores on biological networks","Identification and Annotation of Protein Hotspot Residues","A DownStream Chemo-Proteomics Analysis Pipeline","Package to Draw Protein Schematics from Uniprot API output","Drug-Target Interactions","Ensemble of Gene Set Enrichment Analyses","Accelerated similarity searching of small molecules","Expression Weighted Celltype Enrichment","f-divergence Cutoff Index for Differential Expression Analysis in Transcriptomics and Proteomics","Mismatch Tolerant Maximum Common Substructure Searching","Generating Protein Sequences with Deep Generative Models","NanoString GeoMx Tools","Identification of Mutational Clusters in Proteins via a Graph Theoretical Approach.","Retrieve and analyze data from the Human Protein Atlas","Human Protein Atlas in R","Host-Pathogen Interaction Prediction","Profiling and Analyzing Intrinsically Disordered Proteins in R","Interlog protein network reconstruction by Mapping and Mining ANalysis","Augmenting Massively Parallel Cytometry Experiments Using Multivariate Non-Linear Regressions","R Interface with InterMine-Powered Databases","Identification of Protein Amino acid Clustering","Analysis and quantitation of isobarically tagged MSMS proteomics data","KinSwingR: network-based kinase activity prediction","Linear Models for Microarray Data","A correction of the local pooled error (LPE) method to replace the asymptotic variance adjustment with an unbiased adjustment\nbased on sample size.","Peak Detection for Mass Spectrometry data using wavelet-based algorithms","Shiny-based interactive data-quality exploration for omics data","Mean/Median-balanced quantile normalization","Multi-Contrast Gene Set Enrichment Analysis","Omics Data Integration Project","Effective mobility scale transformation of CE-MS(/MS) data","Quality Control of Protemics Data","Mass Spectrometry Data Backend for Mascot Generic Format (mgf) Files","Mass Spectrometry Data Backend for NIST msp Files","Mass Spectrometry Backend for Reading Thermo Fisher Scientific raw Files","Core Utils for Mass Spectrometry Data","Infrastructure for Mass Spectrometry Experiments","Imputation of label-free mass spectrometry peptides","Exploratory Data Analysis of LC-MS/MS data by spectral counts","LC-MS/MS Differential Expression Tests","Base Functions and Classes for Mass Spectrometry and Proteomics","Utilities for Exploration and Assessment of Confidence of LC-MSn Proteomics Identifications","Robust statistical inference for quantitative LC-MS proteomics","MsQuality - Quality metric calculation from Spectra objects","Protein Significance Analysis in DDA, SRM and DIA for Label-free or Label-based Proteomics Experiments","Import Data from Various Mass Spectrometry Signal Processing Tools to MSstats Format","LiP Significance Analysis in shotgun mass spectrometry-based proteomic experiments","Assay characterization: estimation of limit of blanc(LoB) and limit of detection(LOD)","Statistical Characterization of Post-translational Modifications","Longitudinal system suitability monitoring and quality control for proteomic experiments","A graphical user interface for MSstatsQC package","Simulation tool for optimal design of high-dimensional MS-based proteomics experiment","MSstats GUI for Statistical Anaylsis of Proteomics Experiments","Protein Significance Analysis in shotgun mass spectrometry-based proteomic experiments with tandem mass tag (TMT) labeling","An mzIdentML parser for R","parser for netCDF, mzXML, mzData and mzML and mzIdentML files (mass spectrometry data)","NanoString nCounter Tools","Evaluation of normalization methods and calculation of differential expression analysis statistics","Non-parametric analysis of response curves for thermal proteome profiling experiments","Omics Longitudinal Differential Analysis","Outlier Protein and Phosphosite Target Identifier","PAA (Protein Array Analyzer)","Prediction of pathway-related protein-protein interaction networks","a tool set for pathway based data integration and visualization","Integrative Pathway Analysis with Modern PCA Methodology and Gene Selection","Probe-level Expression Change Averaging","Parsing pepXML files and filter based on peptide FDR.","PGCA: An Algorithm to Link Protein Groups Created from MS/MS Data","Compensates for the bias introduced by median normalization in","A set of methods and tools for comprehensive analysis of phosphoproteomics data","Detect differential expression in microarray and proteomics datasets with the Power Law Global Error Model (PLGEM)","Local Pooled Error Test for Differential Expression with Paired High-throughput Data","Predicting Interactomes from Co-Elution","Generating SAM file for PSMs in shotgun proteomics data","Ciphergen SELDI-TOF Processing","Prediction of Oligomerization of Coiled Coil Proteins","Differential Abundance Analysis of Label-Free Mass Spectrometry Data","A unifying bioinformatics framework for spatial proteomics","Interactive visualisation of spatial proteomics data","Provides a GUI for DAPAR","Protease Mapping","Generation of customized protein variant databases from genomic variants, splice-junctions and manual sequences","Multi-Dataset Model-based Differential Expression Proteomics Analysis Platform","Generic infrastructure for Bioconductor mass spectrometry packages","Handling and Managing Peptide Spectrum Matches","Peptide Annotation and Data Visualization using Gviz","A Framework for Quality Control","Quantitative features for mass spectrometry data","Tools for quantitative proteomics data analysis","Identification of mutational clusters in protein quaternary structures.","Rhythmicity Analysis Incorporating Non-parametric Methods","Rank Product method for identifying differentially expressed genes with application in meta-analysis","Direct Access to Orbitrap Data and Beyond","A package for survival time prediction based on a piecewise baseline hazard Cox regression model.","Molecular Informatics Toolkit for Compound-Protein Interaction in Drug Discovery","Client for the Reactome Analysis Service for comparative multi-omics gene set analysis","RolDE: Robust longitudinal Differential Expression","Reproducibility-Optimized Test Statistic","R Interface to the ProteomeXchange Repository","Signature-based Clustering for Diagnostic Purposes","Thermal proximity co-aggregation with R","\"SBGNview: Data Analysis, Integration and Visualization on SBGN Pathways\"","Mass Spectrometry-Based Single-Cell Proteomics Data Analysis","Differential Abundant/Expression Analysis for Metabolomics, Proteomics and single-cell RNA sequencing Data","Statistical Modelling of AP-MS Data (SMAD)","Identification of Mutational Clusters in 3D Protein Space via Simulation.","Using Gaussian graphical structue learning estimation in generalized least squared regression for multivariate normal\nregression","specL - Prepare Peptide Spectrum Matches for Use in Targeted Proteomics","Spectra Infrastructure for Mass Spectrometry Data","Statistical Analysis of Molecular Profiles","SubCellBarCode: Integrated workflow for robust mapping and visualizing whole human spatial proteome","Transform and Filter SWATH Data for Statistical Packages","Label-free data analysis pipeline for optimal identification and quantitation","Diagnostic Plots to Evaluate the Target Decoy Approach","Time-Course Multi-Omics data integration","Investigation of Fragmentation Conditions in Top-Down Proteomics","Analyze thermal proteome profiling (TPP) experiments","Detection of ligand-protein interactions from 2D thermal profiles (DLPTP)","Feature-based variance-sensitive quantitative clustering","Well Plate Maker","Multiplexes Isobaric Mass Tagged-based Kinetics Data for Network Analysis","Signature discovery from omics data","Exposome and omic data associatin and integration analysis","Postprocessing and univariate analysis of omics data","Tools for Diagnostics and Corrections of Batch Effects in Proteomics","PCA, PLS(-DA) and OPLS(-DA) for multivariate analysis and feature selection of omics data","Preprocessing of FIA-HRMS data","Decomposing Heterogeneous Cohorts using Omic Data Profiling","Analysis of pulsed-SILAC quantitative proteomics data","Differential abundance analysis with a right censored covariate in high-dimensional cytometry","Phenotypic EMD for comparison of single-cell samples"],["1.17.0","1.43.0","1.17.0","1.3.0","1.37.0","1.31.0","1.19.0","1.45.0","1.13.0","3.1.0","1.31.0","1.45.0","1.37.0","3.51.0","1.39.0","1.37.0","1.49.0","1.41.0","1.39.0","1.41.0","1.7.0","1.7.0","1.7.0","1.39.0","1.23.0","1.7.0","1.5.0","1.3.0","1.31.0","1.39.0","1.21.0","1.17.3","2.7.0","1.19.0","1.19.0","1.19.0","1.21.0","1.19.0","1.7.0","1.27.0","1.39.0","1.7.0","1.29.0","1.41.0","1.3.0","3.3.0","1.41.0","1.17.0","1.41.1","1.5.0","1.9.15","1.19.0","1.9.0","1.21.0","1.43.0","1.45.0","1.17.0","3.55.2","1.59.0","1.65.0","1.7.2","2.11.0","1.11.4","6.23.3","1.3.0","1.7.0","1.7.1","1.3.0","1.5.0","1.11.2","1.1.1","1.9.0","1.37.0","1.37.0","2.25.1","1.33.0","1.7.0","0.99.6","4.7.1","1.9.2","1.5.1","1.7.0","2.1.1","2.17.0","1.19.0","1.13.0","1.1.5","2.7.0","1.37.0","2.33.0","1.7.0","1.17.0","1.11.0","1.15.0","1.13.0","1.33.0","1.29.0","1.39.0","1.15.0","1.35.0","1.33.0","1.23.0","1.23.0","1.9.1","1.71.0","1.59.0","1.15.0","1.33.0","1.75.0","2.27.0","1.13.0","1.39.0","2.9.0","1.31.0","1.1.0","1.5.0","1.17.0","1.31.0","1.3.0","1.33.0","1.37.0","1.9.2","1.17.0","1.31.0","1.33.0","3.25.0","1.7.0","1.45.0","1.35.0","1.13.0","1.3.0","1.27.0","2.7.0","1.15.0","1.9.0","1.13.0","1.9.0","1.19.0","1.15.1","1.37.0","1.17.0","1.33.3","1.9.5","1.29.0","1.15.0","1.29.0","2.23.0","1.5.0","1.11.0","1.21.0","3.27.0","1.15.2","1.1.0","1.9.0","1.17.0","1.27.0","1.21.0","1.1.2","1.15.0","1.31.0","1.25.0","1.15.0","1.13.0","1.7.0","1.15.0"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Package<\/th>\n      <th>Title<\/th>\n      <th>Version<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-4232c106d899ec029ed7" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-4232c106d899ec029ed7">{"x":{"filter":"none","vertical":false,"data":[["adductomicsR","apComplex","artMS","autonomics","bandle","BRAIN","CAMERA","Cardinal","cliqueMS","CompoundDb","ComPrAn","CONSTANd","cosmiq","customProDB","DAPAR","DEP","DEqMS","DIAlignR","Doscheda","flagme","fobitools","gaga","INDEED","IPO","isobar","IsoCorrectoR","IsoCorrectoRGUI","LOBSTAHS","MAIT","marr","MassArray","MassSpecWavelet","Metab","metabCombiner","MetaboAnnotation","MetaboCoreUtils","metabomxtr","metaMS","metapone","MetCirc","MetNet","MobilityTransformR","MQmetrics","MsBackendMassbank","MsBackendMgf","MsBackendMsp","MsBackendRawFileReader","MsCoreUtils","MsExperiment","MsFeatures","msImpute","msmsEDA","msmsTests","MSnbase","MSnID","MSPrep","msPurity","msqrob2","MsQuality","MSstats","MSstatsConvert","MSstatsLiP","MSstatsLOBD","MSstatsPTM","MSstatsQC","MSstatsQCgui","MSstatsSampleSize","MSstatsShiny","MSstatsTMT","mzID","mzR","ncGTW","peakPantheR","pepXMLTab","pgca","plgem","pmp","proBAMr","PROcess","proDA","pRoloc","Prostar","ProteoMM","ProtGenerics","PSMatch","ptairMS","qcmetrics","QFeatures","qmtools","qPLEXanalyzer","rawrr","Rdisop","rgoslin","Risa","RMassBank","rols","rpx","scDataviz","scp","SDAMS","SIMAT","SMAD","sparsenetgls","specL","Spectra","statTarget","SubCellBarCode","SWATH2stats","synapter","TargetDecoy","TargetSearch","topdownr","TPP","treekoR","wpm","xcms","yamss","biosigner","lipidr","phenomis","proBatch","ropls","proFIA","CATALYST"],["adductomicsR","apComplex","artMS","autonomics","bandle","BRAIN","CAMERA","Cardinal","cliqueMS","CompoundDb","ComPrAn","CONSTANd","cosmiq","customProDB","DAPAR","DEP","DEqMS","DIAlignR","Doscheda","flagme","fobitools","gaga","INDEED","IPO","isobar","IsoCorrectoR","IsoCorrectoRGUI","LOBSTAHS","MAIT","marr","MassArray","MassSpecWavelet","Metab","metabCombiner","MetaboAnnotation","MetaboCoreUtils","metabomxtr","metaMS","metapone","MetCirc","MetNet","MobilityTransformR","MQmetrics","MsBackendMassbank","MsBackendMgf","MsBackendMsp","MsBackendRawFileReader","MsCoreUtils","MsExperiment","MsFeatures","msImpute","msmsEDA","msmsTests","MSnbase","MSnID","MSPrep","msPurity","msqrob2","MsQuality","MSstats","MSstatsConvert","MSstatsLiP","MSstatsLOBD","MSstatsPTM","MSstatsQC","MSstatsQCgui","MSstatsSampleSize","MSstatsShiny","MSstatsTMT","mzID","mzR","ncGTW","peakPantheR","pepXMLTab","pgca","plgem","pmp","proBAMr","PROcess","proDA","pRoloc","Prostar","ProteoMM","ProtGenerics","PSMatch","ptairMS","qcmetrics","QFeatures","qmtools","qPLEXanalyzer","rawrr","Rdisop","rgoslin","Risa","RMassBank","rols","rpx","scDataviz","scp","SDAMS","SIMAT","SMAD","sparsenetgls","specL","Spectra","statTarget","SubCellBarCode","SWATH2stats","synapter","TargetDecoy","TargetSearch","topdownr","TPP","treekoR","wpm","xcms","yamss","biosigner","lipidr","phenomis","proBatch","ropls","proFIA","CATALYST"],["Processing of adductomic mass spectral datasets","Estimate protein complex membership using AP-MS protein data","Analytical R tools for Mass Spectrometry","Generifying and intuifying cross-platform omics analysis","An R package for the Bayesian analysis of differential subcellular localisation experiments","Baffling Recursive Algorithm for Isotope distributioN calculations","Collection of annotation related methods for mass spectrometry data","A mass spectrometry imaging toolbox for statistical analysis","Annotation of Isotopes, Adducts and Fragmentation Adducts for in-Source LC/MS Metabolomics Data","Creating and Using (Chemical) Compound Annotation Databases","Complexome Profiling Analysis package","Data normalization by matrix raking","cosmiq - COmbining Single Masses Into Quantities","Generate customized protein database from NGS data, with a focus on RNA-Seq data, for proteomics search","Tools for the Differential Analysis of Proteins Abundance with R","Differential Enrichment analysis of Proteomics data","a tool to perform statistical analysis of differential protein expression for quantitative proteomics data.","Dynamic Programming Based Alignment of MS2 Chromatograms","A DownStream Chemo-Proteomics Analysis Pipeline","Analysis of Metabolomics GC/MS Data","Tools For Manipulating FOBI Ontology","GaGa hierarchical model for high-throughput data analysis","Interactive Visualization of Integrated Differential Expression and Differential Network Analysis for Biomarker Candidate\nSelection Package","Automated Optimization of XCMS Data Processing parameters","Analysis and quantitation of isobarically tagged MSMS proteomics data","Correction for natural isotope abundance and tracer purity in MS and MS/MS data from stable isotope labeling experiments","Graphical User Interface for IsoCorrectoR","Lipid and Oxylipin Biomarker Screening through Adduct Hierarchy Sequences","Statistical Analysis of Metabolomic Data","Maximum rank reproducibility","Analytical Tools for MassArray Data","Peak Detection for Mass Spectrometry data using wavelet-based algorithms","Metab: An R Package for a High-Throughput Analysis of Metabolomics Data Generated by GC-MS.","Method for Combining LC-MS Metabolomics Feature Measurements","Utilities for Annotation of Metabolomics Data","Core Utils for Metabolomics Data","A package to run mixture models for truncated metabolomics data with normal or lognormal distributions","MS-based metabolomics annotation pipeline","Conducts pathway test of metabolomics data using a weighted permutation test","Navigating mass spectral similarity in high-resolution MS/MS metabolomics data metabolomics data","Inferring metabolic networks from untargeted high-resolution mass spectrometry data","Effective mobility scale transformation of CE-MS(/MS) data","Quality Control of Protemics Data","Mass Spectrometry Data Backend for MassBank record Files","Mass Spectrometry Data Backend for Mascot Generic Format (mgf) Files","Mass Spectrometry Data Backend for NIST msp Files","Mass Spectrometry Backend for Reading Thermo Fisher Scientific raw Files","Core Utils for Mass Spectrometry Data","Infrastructure for Mass Spectrometry Experiments","Functionality for Mass Spectrometry Features","Imputation of label-free mass spectrometry peptides","Exploratory Data Analysis of LC-MS/MS data by spectral counts","LC-MS/MS Differential Expression Tests","Base Functions and Classes for Mass Spectrometry and Proteomics","Utilities for Exploration and Assessment of Confidence of LC-MSn Proteomics Identifications","Package for Summarizing, Filtering, Imputing, and Normalizing Metabolomics Data","Automated Evaluation of Precursor Ion Purity for Mass Spectrometry Based Fragmentation in Metabolomics","Robust statistical inference for quantitative LC-MS proteomics","MsQuality - Quality metric calculation from Spectra objects","Protein Significance Analysis in DDA, SRM and DIA for Label-free or Label-based Proteomics Experiments","Import Data from Various Mass Spectrometry Signal Processing Tools to MSstats Format","LiP Significance Analysis in shotgun mass spectrometry-based proteomic experiments","Assay characterization: estimation of limit of blanc(LoB) and limit of detection(LOD)","Statistical Characterization of Post-translational Modifications","Longitudinal system suitability monitoring and quality control for proteomic experiments","A graphical user interface for MSstatsQC package","Simulation tool for optimal design of high-dimensional MS-based proteomics experiment","MSstats GUI for Statistical Anaylsis of Proteomics Experiments","Protein Significance Analysis in shotgun mass spectrometry-based proteomic experiments with tandem mass tag (TMT) labeling","An mzIdentML parser for R","parser for netCDF, mzXML, mzData and mzML and mzIdentML files (mass spectrometry data)","Alignment of LC-MS Profiles by Neighbor-wise Compound-specific Graphical Time Warping with Misalignment Detection","Peak Picking and Annotation of High Resolution Experiments","Parsing pepXML files and filter based on peptide FDR.","PGCA: An Algorithm to Link Protein Groups Created from MS/MS Data","Detect differential expression in microarray and proteomics datasets with the Power Law Global Error Model (PLGEM)","Peak Matrix Processing and signal batch correction for metabolomics datasets","Generating SAM file for PSMs in shotgun proteomics data","Ciphergen SELDI-TOF Processing","Differential Abundance Analysis of Label-Free Mass Spectrometry Data","A unifying bioinformatics framework for spatial proteomics","Provides a GUI for DAPAR","Multi-Dataset Model-based Differential Expression Proteomics Analysis Platform","Generic infrastructure for Bioconductor mass spectrometry packages","Handling and Managing Peptide Spectrum Matches","Pre-processing PTR-TOF-MS Data","A Framework for Quality Control","Quantitative features for mass spectrometry data","Quantitative Metabolomics Data Processing Tools","Tools for quantitative proteomics data analysis","Direct Access to Orbitrap Data and Beyond","Decomposition of Isotopic Patterns","Lipid Shorthand Name Parsing and Normalization","Converting experimental metadata from ISA-tab into Bioconductor data structures","Workflow to process tandem MS files and build MassBank records","An R interface to the Ontology Lookup Service","R Interface to the ProteomeXchange Repository","scDataviz: single cell dataviz and downstream analyses","Mass Spectrometry-Based Single-Cell Proteomics Data Analysis","Differential Abundant/Expression Analysis for Metabolomics, Proteomics and single-cell RNA sequencing Data","GC-SIM-MS data processing and alaysis tool","Statistical Modelling of AP-MS Data (SMAD)","Using Gaussian graphical structue learning estimation in generalized least squared regression for multivariate normal\nregression","specL - Prepare Peptide Spectrum Matches for Use in Targeted Proteomics","Spectra Infrastructure for Mass Spectrometry Data","Statistical Analysis of Molecular Profiles","SubCellBarCode: Integrated workflow for robust mapping and visualizing whole human spatial proteome","Transform and Filter SWATH Data for Statistical Packages","Label-free data analysis pipeline for optimal identification and quantitation","Diagnostic Plots to Evaluate the Target Decoy Approach","A package for the analysis of GC-MS metabolite profiling data","Investigation of Fragmentation Conditions in Top-Down Proteomics","Analyze thermal proteome profiling (TPP) experiments","Cytometry Cluster Hierarchy and Cellular-to-phenotype Associations","Well Plate Maker","LC-MS and GC-MS Data Analysis","Tools for high-throughput metabolomics","Signature discovery from omics data","Data Mining and Analysis of Lipidomics Datasets","Postprocessing and univariate analysis of omics data","Tools for Diagnostics and Corrections of Batch Effects in Proteomics","PCA, PLS(-DA) and OPLS(-DA) for multivariate analysis and feature selection of omics data","Preprocessing of FIA-HRMS data","Cytometry dATa anALYSis Tools"],["1.15.0","2.65.0","1.17.0","1.7.0","1.3.0","1.45.0","1.55.0","3.1.0","1.13.0","1.3.1","1.7.0","1.7.0","1.33.0","1.39.0","1.31.0","1.21.0","1.17.3","2.7.0","1.21.0","1.55.0","1.7.0","2.45.0","2.13.0","1.25.0","1.45.0","1.17.0","1.15.0","1.25.0","1.33.0","1.9.0","1.51.0","1.65.0","1.33.0","1.9.0","1.3.0","1.7.0","1.33.0","1.35.0","1.5.0","1.29.1","1.17.1","1.3.0","1.7.0","1.7.0","1.7.1","1.3.0","1.5.0","1.11.2","1.1.1","1.7.0","1.9.0","1.37.0","1.37.0","2.25.1","1.33.0","1.9.0","1.25.0","1.7.0","0.99.6","4.7.1","1.9.2","1.5.1","1.7.0","2.1.1","2.17.0","1.19.0","1.13.0","1.1.5","2.7.0","1.37.0","2.33.0","1.13.0","1.13.1","1.33.0","1.23.0","1.71.0","1.11.0","1.33.0","1.75.0","1.13.0","1.39.0","1.31.0","1.17.0","1.31.0","1.3.0","1.7.0","1.37.0","1.9.2","1.3.0","1.17.0","1.7.0","1.59.0","1.3.0","1.41.0","3.9.0","2.27.0","2.7.0","1.9.0","1.9.0","1.19.0","1.31.0","1.15.1","1.17.0","1.33.3","1.9.5","1.29.0","1.15.0","1.29.0","2.23.0","1.5.0","2.1.0","1.21.0","3.27.0","1.7.0","1.9.0","3.21.0","1.25.0","1.27.0","2.13.0","1.1.2","1.15.0","1.31.0","1.25.0","1.23.0"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Package<\/th>\n      <th>Title<\/th>\n      <th>Version<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="section level2">
<h2 id="ascombes-quartet">Ascombe’s quartet<a class="anchor" aria-label="anchor" href="#ascombes-quartet"></a>
</h2>
<table class="table">
<thead><tr>
<th style="text-align:right;">
x1
</th>
<th style="text-align:right;">
x2
</th>
<th style="text-align:right;">
x3
</th>
<th style="text-align:right;">
x4
</th>
<th style="text-align:right;">
y1
</th>
<th style="text-align:right;">
y2
</th>
<th style="text-align:right;">
y3
</th>
<th style="text-align:right;">
y4
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
8.04
</td>
<td style="text-align:right;">
9.14
</td>
<td style="text-align:right;">
7.46
</td>
<td style="text-align:right;">
6.58
</td>
</tr>
<tr>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
6.95
</td>
<td style="text-align:right;">
8.14
</td>
<td style="text-align:right;">
6.77
</td>
<td style="text-align:right;">
5.76
</td>
</tr>
<tr>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
7.58
</td>
<td style="text-align:right;">
8.74
</td>
<td style="text-align:right;">
12.74
</td>
<td style="text-align:right;">
7.71
</td>
</tr>
<tr>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
8.81
</td>
<td style="text-align:right;">
8.77
</td>
<td style="text-align:right;">
7.11
</td>
<td style="text-align:right;">
8.84
</td>
</tr>
<tr>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
8.33
</td>
<td style="text-align:right;">
9.26
</td>
<td style="text-align:right;">
7.81
</td>
<td style="text-align:right;">
8.47
</td>
</tr>
<tr>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
9.96
</td>
<td style="text-align:right;">
8.10
</td>
<td style="text-align:right;">
8.84
</td>
<td style="text-align:right;">
7.04
</td>
</tr>
<tr>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
7.24
</td>
<td style="text-align:right;">
6.13
</td>
<td style="text-align:right;">
6.08
</td>
<td style="text-align:right;">
5.25
</td>
</tr>
<tr>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
4.26
</td>
<td style="text-align:right;">
3.10
</td>
<td style="text-align:right;">
5.39
</td>
<td style="text-align:right;">
12.50
</td>
</tr>
<tr>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
10.84
</td>
<td style="text-align:right;">
9.13
</td>
<td style="text-align:right;">
8.15
</td>
<td style="text-align:right;">
5.56
</td>
</tr>
<tr>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
7
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
4.82
</td>
<td style="text-align:right;">
7.26
</td>
<td style="text-align:right;">
6.42
</td>
<td style="text-align:right;">
7.91
</td>
</tr>
<tr>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
5
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:right;">
5.68
</td>
<td style="text-align:right;">
4.74
</td>
<td style="text-align:right;">
5.73
</td>
<td style="text-align:right;">
6.89
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="cn">NA</span>, <span class="fl">5</span>, <span class="fl">4</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">tab</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">tab</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"var(x)"</span>, <span class="st">"mean(x)"</span>,</span>
<span>                   <span class="st">"var(y)"</span>, <span class="st">"mean(y)"</span>,</span>
<span>                   <span class="st">"cor(x,y)"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span></span>
<span>    <span class="va">tab</span><span class="op">[</span>, <span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu">var</span><span class="op">(</span><span class="va">anscombe</span><span class="op">[</span>, <span class="va">i</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">anscombe</span><span class="op">[</span>, <span class="va">i</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                  <span class="fu">var</span><span class="op">(</span><span class="va">anscombe</span><span class="op">[</span>, <span class="va">i</span><span class="op">+</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                  <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">anscombe</span><span class="op">[</span>, <span class="va">i</span><span class="op">+</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                  <span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="va">anscombe</span><span class="op">[</span>, <span class="va">i</span><span class="op">]</span>, <span class="va">anscombe</span><span class="op">[</span>, <span class="va">i</span><span class="op">+</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">1</th>
<th align="right">2</th>
<th align="right">3</th>
<th align="right">4</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">var(x)</td>
<td align="right">11.0000000</td>
<td align="right">11.0000000</td>
<td align="right">11.0000000</td>
<td align="right">11.0000000</td>
</tr>
<tr class="even">
<td align="left">mean(x)</td>
<td align="right">9.0000000</td>
<td align="right">9.0000000</td>
<td align="right">9.0000000</td>
<td align="right">9.0000000</td>
</tr>
<tr class="odd">
<td align="left">var(y)</td>
<td align="right">4.1272691</td>
<td align="right">4.1276291</td>
<td align="right">4.1226200</td>
<td align="right">4.1232491</td>
</tr>
<tr class="even">
<td align="left">mean(y)</td>
<td align="right">7.5009091</td>
<td align="right">7.5009091</td>
<td align="right">7.5000000</td>
<td align="right">7.5009091</td>
</tr>
<tr class="odd">
<td align="left">cor(x,y)</td>
<td align="right">0.8164205</td>
<td align="right">0.8162365</td>
<td align="right">0.8162867</td>
<td align="right">0.8165214</td>
</tr>
</tbody>
</table>
<p>While the residuals of the linear regression clearly indicate
fundamental differences in these data, the most simple and
straightforward approach is visualisation to highlight the fundamental
differences in the datasets.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ff</span> <span class="op">&lt;-</span> <span class="va">y</span> <span class="op">~</span> <span class="va">x</span></span>
<span></span>
<span><span class="va">mods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html" class="external-link">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"lm"</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span>, mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">4</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">ff</span><span class="op">[</span><span class="fl">2</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"y"</span>,<span class="st">"x"</span><span class="op">)</span>, <span class="va">i</span><span class="op">)</span>, <span class="va">as.name</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">ff</span>, data <span class="op">=</span> <span class="va">anscombe</span>, pch <span class="op">=</span> <span class="fl">19</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">19</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">13</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="va">mods</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/lm.html" class="external-link">lm</a></span><span class="op">(</span><span class="va">ff</span>, data <span class="op">=</span> <span class="va">anscombe</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span><span class="va">mods</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/anscombefig-1.png" width="700"></p>
<table class="table">
<thead><tr class="header">
<th align="right">lm1</th>
<th align="right">lm2</th>
<th align="right">lm3</th>
<th align="right">lm4</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">0.0390000</td>
<td align="right">1.1390909</td>
<td align="right">-0.5397273</td>
<td align="right">-0.421</td>
</tr>
<tr class="even">
<td align="right">-0.0508182</td>
<td align="right">1.1390909</td>
<td align="right">-0.2302727</td>
<td align="right">-1.241</td>
</tr>
<tr class="odd">
<td align="right">-1.9212727</td>
<td align="right">-0.7609091</td>
<td align="right">3.2410909</td>
<td align="right">0.709</td>
</tr>
<tr class="even">
<td align="right">1.3090909</td>
<td align="right">1.2690909</td>
<td align="right">-0.3900000</td>
<td align="right">1.839</td>
</tr>
<tr class="odd">
<td align="right">-0.1710909</td>
<td align="right">0.7590909</td>
<td align="right">-0.6894545</td>
<td align="right">1.469</td>
</tr>
<tr class="even">
<td align="right">-0.0413636</td>
<td align="right">-1.9009091</td>
<td align="right">-1.1586364</td>
<td align="right">0.039</td>
</tr>
<tr class="odd">
<td align="right">1.2393636</td>
<td align="right">0.1290909</td>
<td align="right">0.0791818</td>
<td align="right">-1.751</td>
</tr>
<tr class="even">
<td align="right">-0.7404545</td>
<td align="right">-1.9009091</td>
<td align="right">0.3886364</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="right">1.8388182</td>
<td align="right">0.1290909</td>
<td align="right">-0.8491818</td>
<td align="right">-1.441</td>
</tr>
<tr class="even">
<td align="right">-1.6807273</td>
<td align="right">0.7590909</td>
<td align="right">-0.0805455</td>
<td align="right">0.909</td>
</tr>
<tr class="odd">
<td align="right">0.1794545</td>
<td align="right">-0.7609091</td>
<td align="right">0.2289091</td>
<td align="right">-0.111</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="the-ma-plot-example">The MA plot example<a class="anchor" aria-label="anchor" href="#the-ma-plot-example"></a>
</h2>
<p>The following code chunk connects to the <code>PXD000001</code> data
set on the ProteomeXchange repository and fetches the <code>mzTab</code>
file. After missing values filtering, we extract relevant data (log2
fold-changes and log10 mean expression intensities) into
<code>data.frames</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/lgatto/rpx" class="external-link">"rpx"</a></span><span class="op">)</span></span>
<span><span class="va">px1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/rpx/man/PXDataset2.html" class="external-link">PXDataset</a></span><span class="op">(</span><span class="st">"PXD000001"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading PXD000001 from cache.</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mztab</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/rpx/man/PXDataset2.html" class="external-link">pxget</a></span><span class="op">(</span><span class="va">px1</span>, <span class="st">"F063721.dat-mztab.txt"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading F063721.dat-mztab.txt from cache.</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://lgatto.github.io/MSnbase" class="external-link">"MSnbase"</a></span><span class="op">)</span></span>
<span><span class="co">## here, we need to specify the (old) mzTab version 0.9</span></span>
<span><span class="va">qnt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/readMzTabData.html" class="external-link">readMzTabData</a></span><span class="op">(</span><span class="va">mztab</span>, what <span class="op">=</span> <span class="st">"PEP"</span>, version <span class="op">=</span> <span class="st">"0.9"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html" class="external-link">sampleNames</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/ReporterIons-class.html" class="external-link">reporterNames</a></span><span class="op">(</span><span class="va">TMT6</span><span class="op">)</span></span>
<span><span class="va">qnt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">filterNA</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span></span>
<span><span class="co">## may be combineFeatuers</span></span>
<span></span>
<span><span class="va">spikes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"P02769"</span>, <span class="st">"P00924"</span>, <span class="st">"P62894"</span>, <span class="st">"P00489"</span><span class="op">)</span></span>
<span><span class="va">protclasses</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">$</span><span class="va">accession</span><span class="op">)</span></span>
<span><span class="va">protclasses</span><span class="op">[</span><span class="op">!</span><span class="va">protclasses</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">spikes</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Background"</span></span>
<span></span>
<span></span>
<span><span class="va">madata42</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>A <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">2</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                       M <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">[</span>, <span class="fl">4</span><span class="op">]</span>, <span class="fl">2</span><span class="op">)</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span>                       data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"4vs2"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                       protein <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">$</span><span class="va">accession</span>,</span>
<span>                       class <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">protclasses</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">madata62</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>A <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">6</span>, <span class="fl">2</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                       M <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">[</span>, <span class="fl">6</span><span class="op">]</span>, <span class="fl">2</span><span class="op">)</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span>                       data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"6vs2"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                       protein <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">qnt</span><span class="op">)</span><span class="op">$</span><span class="va">accession</span>,</span>
<span>                       class <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">protclasses</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">madata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">madata42</span>, <span class="va">madata62</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="the-traditional-plotting-system">The traditional plotting system<a class="anchor" aria-label="anchor" href="#the-traditional-plotting-system"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">M</span> <span class="op">~</span> <span class="va">A</span>, data <span class="op">=</span> <span class="va">madata42</span>, main <span class="op">=</span> <span class="st">"4vs2"</span>,</span>
<span>     xlab <span class="op">=</span> <span class="st">"A"</span>, ylab <span class="op">=</span> <span class="st">"M"</span>, col <span class="op">=</span> <span class="va">madata62</span><span class="op">$</span><span class="va">class</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">M</span> <span class="op">~</span> <span class="va">A</span>, data <span class="op">=</span> <span class="va">madata62</span>, main <span class="op">=</span> <span class="st">"6vs2"</span>,</span>
<span>     xlab <span class="op">=</span> <span class="st">"A"</span>, ylab <span class="op">=</span> <span class="st">"M"</span>, col <span class="op">=</span> <span class="va">madata62</span><span class="op">$</span><span class="va">class</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/mafig1-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="lattice">lattice<a class="anchor" aria-label="anchor" href="#lattice"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="http://lattice.r-forge.r-project.org/" class="external-link">"lattice"</a></span><span class="op">)</span></span>
<span><span class="va">latma</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/lattice/man/xyplot.html" class="external-link">xyplot</a></span><span class="op">(</span><span class="va">M</span> <span class="op">~</span> <span class="va">A</span> <span class="op">|</span> <span class="va">data</span>, data <span class="op">=</span> <span class="va">madata</span>,</span>
<span>                groups <span class="op">=</span> <span class="va">madata</span><span class="op">$</span><span class="va">class</span>,</span>
<span>                auto.key <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">latma</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/mafig2-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="ggplot2">ggplot2<a class="anchor" aria-label="anchor" href="#ggplot2"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span></span>
<span><span class="va">ggma</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">A</span>, y <span class="op">=</span> <span class="va">M</span>, colour <span class="op">=</span> <span class="va">class</span><span class="op">)</span>, data <span class="op">=</span> <span class="va">madata</span>,</span>
<span>               colour <span class="op">=</span> <span class="va">class</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span><span class="va">.</span> <span class="op">~</span> <span class="va">data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">ggma</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/mafig3-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="customization">Customization<a class="anchor" aria-label="anchor" href="#customization"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"RColorBrewer"</span><span class="op">)</span></span>
<span><span class="va">bcols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RColorBrewer/man/ColorBrewer.html" class="external-link">brewer.pal</a></span><span class="op">(</span><span class="fl">4</span>, <span class="st">"Set1"</span><span class="op">)</span></span>
<span><span class="va">cls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Background"</span> <span class="op">=</span> <span class="st">"#12121230"</span>,</span>
<span>         <span class="st">"P02769"</span> <span class="op">=</span> <span class="va">bcols</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>         <span class="st">"P00924"</span> <span class="op">=</span> <span class="va">bcols</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,</span>
<span>         <span class="st">"P62894"</span> <span class="op">=</span> <span class="va">bcols</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span>,</span>
<span>         <span class="st">"P00489"</span> <span class="op">=</span> <span class="va">bcols</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ggma2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">A</span>, y <span class="op">=</span> <span class="va">M</span>, colour <span class="op">=</span> <span class="va">class</span><span class="op">)</span>,</span>
<span>                data <span class="op">=</span> <span class="va">madata</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>shape <span class="op">=</span> <span class="fl">19</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span><span class="va">.</span> <span class="op">~</span> <span class="va">data</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_colour_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">cls</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span>override.aes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">ggma2</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/macust-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="the-maplot-method-for-msnset-instances">The <code>MAplot</code> method for <code>MSnSet</code>
instances<a class="anchor" aria-label="anchor" href="#the-maplot-method-for-msnset-instances"></a>
</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">MAplot</span><span class="op">(</span><span class="va">qnt</span>, cex <span class="op">=</span> <span class="fl">.8</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/mafigmsnset-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="an-interactive-shiny-app-for-ma-plots">An interactive <em><a href="https://CRAN.R-project.org/package=shiny" class="external-link">shiny</a></em> app for
MA plots<a class="anchor" aria-label="anchor" href="#an-interactive-shiny-app-for-ma-plots"></a>
</h3>
<p>This (now outdated and deprecated) app is based on Mike Love’s <a href="https://github.com/mikelove/shinyMA" class="external-link">shinyMA</a> application,
adapted for a proteomics data. A screen shot is displayed below.</p>
<p><img src="figures/shinyMA.png" width="80%" style="display: block; margin: auto;"></p>
<p>See the excellent <em><a href="https://CRAN.R-project.org/package=shiny" class="external-link">shiny</a></em> <a href="http://shiny.rstudio.com/" class="external-link">web page</a> for tutorials and the <a href="https://mastering-shiny.org/" class="external-link">Mastering Shiny</a> book for details
on <code>shiny</code>.</p>
</div>
<div class="section level3">
<h3 id="volcano-plots">Volcano plots<a class="anchor" aria-label="anchor" href="#volcano-plots"></a>
</h3>
<p>Below, using the <em><a href="https://bioconductor.org/packages/3.17/msmsTest" class="external-link">msmsTest</a></em>
package, we load a example <code>MSnSet</code> data with spectral
couting data (from the <code>r Biocpkg("msmsEDA")</code> package) and
run a statistical test to obtain (adjusted) p-values and
fold-changes.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"msmsEDA"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"msmsTests"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">msms.dataset</span><span class="op">)</span></span>
<span><span class="co">## Pre-process expression matrix</span></span>
<span><span class="va">e</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/msmsEDA/man/pp.msms.data.html" class="external-link">pp.msms.data</a></span><span class="op">(</span><span class="va">msms.dataset</span><span class="op">)</span></span>
<span><span class="co">## Models and normalizing condition</span></span>
<span><span class="va">null.f</span> <span class="op">&lt;-</span> <span class="st">"y~batch"</span></span>
<span><span class="va">alt.f</span> <span class="op">&lt;-</span> <span class="st">"y~treat+batch"</span></span>
<span><span class="va">div</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">e</span><span class="op">)</span>, <span class="fl">2</span>, <span class="va">sum</span><span class="op">)</span></span>
<span><span class="co">## Test</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/msmsTests/man/msms.glm.qlll.html" class="external-link">msms.glm.qlll</a></span><span class="op">(</span><span class="va">e</span>, <span class="va">alt.f</span>, <span class="va">null.f</span>, div <span class="op">=</span> <span class="va">div</span><span class="op">)</span></span>
<span><span class="va">lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/msmsTests/man/test.results.html" class="external-link">test.results</a></span><span class="op">(</span><span class="va">res</span>, <span class="va">e</span>, <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">e</span><span class="op">)</span><span class="op">$</span><span class="va">treat</span>, <span class="st">"U600"</span>, <span class="st">"U200 "</span>, <span class="va">div</span>,</span>
<span>                    alpha <span class="op">=</span> <span class="fl">0.05</span>, minSpC <span class="op">=</span> <span class="fl">2</span>, minLFC <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fl">1.8</span><span class="op">)</span>,</span>
<span>                    method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span></span></code></pre></div>
<p>Here, we produce the volcano plot by hand, with the <code>plot</code>
function. In the second plot, we limit the x axis limits and add grid
lines.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">lst</span><span class="op">$</span><span class="va">tres</span><span class="op">$</span><span class="va">LogFC</span>, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">lst</span><span class="op">$</span><span class="va">tres</span><span class="op">$</span><span class="va">p.value</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/volc1-1.png" width="700"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">lst</span><span class="op">$</span><span class="va">tres</span><span class="op">$</span><span class="va">LogFC</span>, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">lst</span><span class="op">$</span><span class="va">tres</span><span class="op">$</span><span class="va">p.value</span><span class="op">)</span>,</span>
<span>     xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">3</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/grid.html" class="external-link">grid</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/volc1-2.png" width="700"></p>
<p>Below, we use the <code>res.volcanoplot</code> function from the
<code>r Biocpkg("msmsTests")</code> package. This functions uses the
sample annotation stored with the quantitative data in the
<code>MSnSet</code> object to colour the samples according to their
phenotypes.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Plot</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/msmsTests/man/res.volcanoplot.html" class="external-link">res.volcanoplot</a></span><span class="op">(</span><span class="va">lst</span><span class="op">$</span><span class="va">tres</span>,</span>
<span>                max.pval <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                min.LFC <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                maxx <span class="op">=</span> <span class="fl">3</span>,</span>
<span>                maxy <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                ylbls <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/volc2-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="a-pca-plot">A PCA plot<a class="anchor" aria-label="anchor" href="#a-pca-plot"></a>
</h3>
<p>Using the <code>counts.pca</code> function from the <em><a href="https://bioconductor.org/packages/3.17/msmsEDA" class="external-link">msmsEDA</a></em>
package:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"msmsEDA"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">msms.dataset</span><span class="op">)</span></span>
<span><span class="va">msnset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/msmsEDA/man/pp.msms.data.html" class="external-link">pp.msms.data</a></span><span class="op">(</span><span class="va">msms.dataset</span><span class="op">)</span></span>
<span><span class="va">lst</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/msmsEDA/man/counts.pca.html" class="external-link">counts.pca</a></span><span class="op">(</span><span class="va">msnset</span>, wait <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/msmsedapca-1.png" width="700"><img src="RProtVis_files/figure-html/msmsedapca-2.png" width="700"></p>
<p>It is also possible to generate the PCA data using the
<code>prcomp</code>. Below, we extract the coordinates of PC1 and PC2
from the <code>counts.pca</code> result and plot them using the
<code>plot</code> function.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pcadata</span> <span class="op">&lt;-</span> <span class="va">lst</span><span class="op">$</span><span class="va">pca</span><span class="op">$</span><span class="va">x</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pcadata</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                  PC1       PC2</span></span>
<span><span class="co">## U2.2502.1 -120.26080 -53.55270</span></span>
<span><span class="co">## U2.2502.2  -99.90618 -53.89979</span></span>
<span><span class="co">## U2.2502.3 -127.35928 -49.29906</span></span>
<span><span class="co">## U2.2502.4 -166.04611 -39.27557</span></span>
<span><span class="co">## U6.2502.1 -127.18423  37.11614</span></span>
<span><span class="co">## U6.2502.2 -117.97016  47.03702</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">pcadata</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>, <span class="va">pcadata</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span>,</span>
<span>     xlab <span class="op">=</span> <span class="st">"PCA1"</span>, ylab <span class="op">=</span> <span class="st">"PCA2"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/grid.html" class="external-link">grid</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/pca-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="plotting-with-r">Plotting with R<a class="anchor" aria-label="anchor" href="#plotting-with-r"></a>
</h2>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">kable</span><span class="op">(</span><span class="va">plotfuns</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="14%">
<col width="17%">
<col width="35%">
<col width="32%">
</colgroup>
<thead><tr class="header">
<th align="left">plot type</th>
<th align="left">traditional</th>
<th align="left">lattice</th>
<th align="left">ggplot2</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">scatterplots</td>
<td align="left">plot</td>
<td align="left">xyplot</td>
<td align="left">geom_point</td>
</tr>
<tr class="even">
<td align="left">histograms</td>
<td align="left">hist</td>
<td align="left">histgram</td>
<td align="left">geom_histogram</td>
</tr>
<tr class="odd">
<td align="left">density plots</td>
<td align="left">plot(density())</td>
<td align="left">densityplot</td>
<td align="left">geom_density</td>
</tr>
<tr class="even">
<td align="left">boxplots</td>
<td align="left">boxplot</td>
<td align="left">bwplot</td>
<td align="left">geom_boxplot</td>
</tr>
<tr class="odd">
<td align="left">violin plots</td>
<td align="left">vioplot::vioplot</td>
<td align="left">bwplot(…, panel = panel.violin)</td>
<td align="left">geom_violin</td>
</tr>
<tr class="even">
<td align="left">line plots</td>
<td align="left">plot, matplot</td>
<td align="left">xyploy, parallelplot</td>
<td align="left">geom_line</td>
</tr>
<tr class="odd">
<td align="left">bar plots</td>
<td align="left">barplot</td>
<td align="left">barchart</td>
<td align="left">geom_bar</td>
</tr>
<tr class="even">
<td align="left">pie charts</td>
<td align="left">pie</td>
<td align="left"></td>
<td align="left">geom_bar with polar coordinates</td>
</tr>
<tr class="odd">
<td align="left">dot plots</td>
<td align="left">dotchart</td>
<td align="left">dotplot</td>
<td align="left">geom_point</td>
</tr>
<tr class="even">
<td align="left">stip plots</td>
<td align="left">stripchart</td>
<td align="left">stripplot</td>
<td align="left">goem_point</td>
</tr>
<tr class="odd">
<td align="left">dendrogramms</td>
<td align="left">plot(hclust())</td>
<td align="left">latticeExtra package</td>
<td align="left">ggdendro package</td>
</tr>
<tr class="even">
<td align="left">heatmaps</td>
<td align="left">image, heatmap</td>
<td align="left">levelplot</td>
<td align="left">geom_tile</td>
</tr>
</tbody>
</table>
<p>Below, we are going to use a data from the <em><a href="https://bioconductor.org/packages/3.17/pRolocdata" class="external-link">pRolocdata</a></em>
to illustrate the plotting functions.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/lgatto/pRolocdata" class="external-link">"pRolocdata"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="scatter-plots">Scatter plots<a class="anchor" aria-label="anchor" href="#scatter-plots"></a>
</h3>
<p>See the MA and volcano plot examples.</p>
<p>The default plot <code>type</code> is <code>p</code>, for points.
Other important types are <code>l</code> for lines and <code>h</code>
for <em>histogram</em> (see below).</p>
</div>
<div class="section level3">
<h3 id="historams-and-density-plots">Historams and density plots<a class="anchor" aria-label="anchor" href="#historams-and-density-plots"></a>
</h3>
<p>We extract the (normalised) intensities of the first sample</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span></span></code></pre></div>
<p>and plot the distribution with a histogram and a density plot next to
each other on the same figure (using the <code>mfrow</code>
<code>par</code> plotting paramter)</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/density.html" class="external-link">density</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/histplot-1.png" width="1344"></p>
</div>
<div class="section level3">
<h3 id="box-plots-and-violin-plots">Box plots and violin plots<a class="anchor" aria-label="anchor" href="#box-plots-and-violin-plots"></a>
</h3>
<p>we first extract the 888 proteins by <code>r ncol(tan2009r1)</code>
samples data matrix and plot the sample distributions next to each other
using <code>boxplot</code> and <code>beanplot</code> (from the <em><a href="https://CRAN.R-project.org/package=beanplot" class="external-link">beanplot</a></em>
package).</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"beanplot"</span><span class="op">)</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html" class="external-link">boxplot</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/beanplot/man/beanplot.html" class="external-link">beanplot</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>, <span class="va">x</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span>, <span class="va">x</span><span class="op">[</span>, <span class="fl">3</span><span class="op">]</span>, <span class="va">x</span><span class="op">[</span>, <span class="fl">4</span><span class="op">]</span>, log <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/bxplot-1.png" width="672"></p>
</div>
<div class="section level3">
<h3 id="line-plots">Line plots<a class="anchor" aria-label="anchor" href="#line-plots"></a>
</h3>
<p>below, we produce line plots that describe the protein quantitative
profiles for two sets of proteins, namely er and mitochondrial proteins
using <code>matplot</code>.</p>
<p>we need to transpose the matrix (with <code>t</code>) and set the
type to both (<code>b</code>), to display points and lines, the colours
to red and steel blue, the point characters to 1 (an empty point) and
the line type to 1 (a solid line).</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">er</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span><span class="op">$</span><span class="va">markers</span> <span class="op">==</span> <span class="st">"ER"</span></span>
<span><span class="va">mt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span><span class="op">$</span><span class="va">markers</span> <span class="op">==</span> <span class="st">"mitochondrion"</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/matplot.html" class="external-link">matplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="va">er</span>, <span class="op">]</span><span class="op">)</span>, type <span class="op">=</span> <span class="st">"b"</span>, col <span class="op">=</span> <span class="st">"red"</span>, pch <span class="op">=</span> <span class="fl">1</span>, lty <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/matplot.html" class="external-link">matplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="va">mt</span>, <span class="op">]</span><span class="op">)</span>, type <span class="op">=</span> <span class="st">"b"</span>, col <span class="op">=</span> <span class="st">"steelblue"</span>, pch <span class="op">=</span> <span class="fl">1</span>, lty <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/matplotex-1.png" width="672"></p>
<p>In the last section, about spatial proteomics, we use the specialised
<code>plotDist</code> function from the <em><a href="https://bioconductor.org/packages/3.17/pRoloc" class="external-link">pRoloc</a></em>
package to generate such figures.</p>
</div>
<div class="section level3">
<h3 id="bar-and-dot-charts">Bar and dot charts<a class="anchor" aria-label="anchor" href="#bar-and-dot-charts"></a>
</h3>
<p>To illustrate bar and dot charts, we cound the number of proteins in
the respective class using table.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span><span class="op">$</span><span class="va">markers</span><span class="op">)</span></span>
<span><span class="va">x</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##  Cytoskeleton            ER         Golgi      Lysosome mitochondrion </span></span>
<span><span class="co">##             7            28            13             8            29 </span></span>
<span><span class="co">##       Nucleus    Peroxisome            PM    Proteasome  Ribosome 40S </span></span>
<span><span class="co">##            21             4            34            15            20 </span></span>
<span><span class="co">##  Ribosome 60S       unknown </span></span>
<span><span class="co">##            32           677</span></span></code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html" class="external-link">barplot</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/dotchart.html" class="external-link">dotchart</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/mrkplot-1.png" width="1152"></p>
<!-- ### Strip plots -->
</div>
<div class="section level3">
<h3 id="heatmaps">Heatmaps<a class="anchor" aria-label="anchor" href="#heatmaps"></a>
</h3>
<p>The easiest to produce a complete heatmap is with the
<code>heatmap</code> function:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/heatmap.html" class="external-link">heatmap</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/hmap-1.png" width="700"></p>
<p>To produce the a heatmap without the dendrograms, one can use the
image function on a matrix or the specialised version for
<code>MSnSet</code> objects from the <em><a href="https://bioconductor.org/packages/3.17/MSnbase" class="external-link">MSnbase</a></em>
package.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">9</span>, ncol <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/image.html" class="external-link">image</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/image.html" class="external-link">image</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/image-1.png" width="1344"></p>
<p>See also <em><a href="https://CRAN.R-project.org/package=gplots" class="external-link">gplots</a></em>’s
<code>heatmap.2</code> function and the <em><a href="https://bioconductor.org/packages/3.17/Heatplus" class="external-link">Heatplus</a></em>
Bioconductor package for more advanced heatmaps and the <em><a href="https://CRAN.R-project.org/package=corrplot" class="external-link">corrplot</a></em>
package for correlation matrices.</p>
</div>
<div class="section level3">
<h3 id="dendrograms">Dendrograms<a class="anchor" aria-label="anchor" href="#dendrograms"></a>
</h3>
<p>The easiest way to produce and plot a dendrogram is:</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">d</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">dist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="co">## distance between samples</span></span>
<span><span class="va">hc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html" class="external-link">hclust</a></span><span class="op">(</span><span class="va">d</span><span class="op">)</span> <span class="co">## hierarchical clustering</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">hc</span><span class="op">)</span> <span class="co">## visualisation</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/dendro-1.png" width="700"></p>
<p>See also <em><a href="https://CRAN.R-project.org/package=dendextend" class="external-link">dendextend</a></em>
and this <a href="http://stackoverflow.com/questions/6673162/reproducing-lattice-dendrogram-graph-with-ggplot2" class="external-link">post</a>
to illustrate <em><a href="https://CRAN.R-project.org/package=latticeExtra" class="external-link">latticeExtra</a></em>
and <em><a href="https://CRAN.R-project.org/package=ggdendro" class="external-link">ggdendro</a></em>.</p>
</div>
<div class="section level3">
<h3 id="venn-diagrams">Venn diagrams<a class="anchor" aria-label="anchor" href="#venn-diagrams"></a>
</h3>
<ul>
<li>The <em><a href="https://bioconductor.org/packages/3.17/limma" class="external-link">limma</a></em>
package.</li>
<li>The <em><a href="https://CRAN.R-project.org/package=VennDiagram" class="external-link">VennDiagram</a></em>
package.</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="visualising-mass-spectrometry-data">Visualising mass spectrometry data<a class="anchor" aria-label="anchor" href="#visualising-mass-spectrometry-data"></a>
</h2>
<div class="section level3">
<h3 id="direct-access-to-the-raw-data">Direct access to the raw data<a class="anchor" aria-label="anchor" href="#direct-access-to-the-raw-data"></a>
</h3>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/sneumann/mzR/" class="external-link">"mzR"</a></span><span class="op">)</span></span>
<span><span class="va">mzf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/rpx/man/PXDataset2.html" class="external-link">pxget</a></span><span class="op">(</span><span class="va">px1</span>,</span>
<span>             <span class="st">"TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzML"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01-20141210.mzML from cache.</span></span></code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ms</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mzR/man/openMSfile.html" class="external-link">openMSfile</a></span><span class="op">(</span><span class="va">mzf</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mzR/man/peaks.html" class="external-link">header</a></span><span class="op">(</span><span class="va">ms</span><span class="op">)</span></span>
<span><span class="va">ms1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">hd</span><span class="op">$</span><span class="va">msLevel</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">rtsel</span> <span class="op">&lt;-</span> <span class="va">hd</span><span class="op">$</span><span class="va">retentionTime</span><span class="op">[</span><span class="va">ms1</span><span class="op">]</span> <span class="op">/</span> <span class="fl">60</span> <span class="op">&gt;</span> <span class="fl">30</span> <span class="op">&amp;</span> <span class="va">hd</span><span class="op">$</span><span class="va">retentionTime</span><span class="op">[</span><span class="va">ms1</span><span class="op">]</span> <span class="op">/</span> <span class="fl">60</span> <span class="op">&lt;</span> <span class="fl">35</span></span></code></pre></div>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lout</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="cn">NA</span>, ncol <span class="op">=</span> <span class="fl">10</span>, nrow <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></span>
<span><span class="va">lout</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">ii</span> <span class="kw">in</span> <span class="fl">3</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span></span>
<span>    <span class="va">lout</span><span class="op">[</span><span class="va">ii</span>, <span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">3</span>, <span class="fl">3</span><span class="op">)</span></span>
<span><span class="va">lout</span><span class="op">[</span><span class="fl">5</span>, <span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">4</span><span class="op">:</span><span class="fl">8</span>, each <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">lout</span><span class="op">[</span><span class="fl">6</span>, <span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">4</span><span class="op">:</span><span class="fl">8</span>, each <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">lout</span><span class="op">[</span><span class="fl">7</span>, <span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">9</span><span class="op">:</span><span class="fl">13</span>, each <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">lout</span><span class="op">[</span><span class="fl">8</span>, <span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">9</span><span class="op">:</span><span class="fl">13</span>, each <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">i</span> <span class="op">&lt;-</span> <span class="va">ms1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">rtsel</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">j</span> <span class="op">&lt;-</span> <span class="va">ms1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">rtsel</span><span class="op">)</span><span class="op">]</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="va">ms2</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="va">i</span><span class="op">+</span><span class="fl">1</span><span class="op">)</span><span class="op">:</span><span class="op">(</span><span class="va">j</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/layout.html" class="external-link">layout</a></span><span class="op">(</span><span class="va">lout</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">2</span>,<span class="fl">1</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">chromatogram</a></span><span class="op">(</span><span class="va">ms</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, type <span class="op">=</span> <span class="st">"l"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>v <span class="op">=</span> <span class="va">hd</span><span class="op">[</span><span class="va">i</span>, <span class="st">"retentionTime"</span><span class="op">]</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">2</span>, <span class="fl">1</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">peaks</a></span><span class="op">(</span><span class="va">ms</span>, <span class="va">i</span><span class="op">)</span>, type <span class="op">=</span> <span class="st">"l"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">400</span>, <span class="fl">1000</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></span><span class="op">(</span><span class="st">"topright"</span>, bty <span class="op">=</span> <span class="st">"n"</span>,</span>
<span>       legend <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span></span>
<span>           <span class="st">"Acquisition "</span>, <span class="va">hd</span><span class="op">[</span><span class="va">i</span>, <span class="st">"acquisitionNum"</span><span class="op">]</span>,  <span class="st">"\n"</span>,</span>
<span>           <span class="st">"Retention time "</span>, <span class="fu">formatRt</span><span class="op">(</span><span class="va">hd</span><span class="op">[</span><span class="va">i</span>, <span class="st">"retentionTime"</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>h <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>v <span class="op">=</span> <span class="va">hd</span><span class="op">[</span><span class="va">ms2</span>, <span class="st">"precursorMZ"</span><span class="op">]</span>,</span>
<span>       col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#FF000080"</span>,</span>
<span>           <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"#12121280"</span>, <span class="fl">9</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">0.5</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">peaks</a></span><span class="op">(</span><span class="va">ms</span>, <span class="va">i</span><span class="op">)</span>, type <span class="op">=</span> <span class="st">"l"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">521</span>, <span class="fl">522.5</span><span class="op">)</span>, yaxt <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>h <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>v <span class="op">=</span> <span class="va">hd</span><span class="op">[</span><span class="va">ms2</span>, <span class="st">"precursorMZ"</span><span class="op">]</span>, col <span class="op">=</span> <span class="st">"#FF000080"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">2</span>, <span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">ii</span> <span class="kw">in</span> <span class="va">ms2</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ProtGenerics/man/protgenerics.html" class="external-link">peaks</a></span><span class="op">(</span><span class="va">ms</span>, <span class="va">ii</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">p</span>, xlab <span class="op">=</span> <span class="st">""</span>, ylab <span class="op">=</span> <span class="st">""</span>, type <span class="op">=</span> <span class="st">"h"</span>, cex.axis <span class="op">=</span> <span class="fl">.6</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></span><span class="op">(</span><span class="st">"topright"</span>,</span>
<span>           legend <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Prec M/Z\n"</span>, <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">hd</span><span class="op">[</span><span class="va">ii</span>, <span class="st">"precursorMZ"</span><span class="op">]</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>,</span>
<span>           bty <span class="op">=</span> <span class="st">"n"</span>, cex <span class="op">=</span> <span class="fl">.8</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="figure">
<img src="RProtVis_files/figure-html/msdetailsplot-1.png" alt="Accesing and plotting MS data." width="700"><p class="caption">
Accesing and plotting MS data.
</p>
</div>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">M2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSmap-class.html" class="external-link">MSmap</a></span><span class="op">(</span><span class="va">ms</span>, <span class="va">i</span><span class="op">:</span><span class="va">j</span>, <span class="fl">100</span>, <span class="fl">1000</span>, <span class="fl">1</span>, <span class="va">hd</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSmap-class.html" class="external-link">plot3D</a></span><span class="op">(</span><span class="va">M2</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/maps3D2-1.png" width="700"></p>
<div class="section level4">
<h4 id="ms-barcoding">MS barcoding<a class="anchor" aria-label="anchor" href="#ms-barcoding"></a>
</h4>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/image.html" class="external-link">image</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="va">hd</span><span class="op">$</span><span class="va">msLevel</span>, <span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">hd</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>      xlab<span class="op">=</span><span class="st">"Retention time"</span>,</span>
<span>      xaxt<span class="op">=</span><span class="st">"n"</span>, yaxt<span class="op">=</span><span class="st">"n"</span>, col<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"black"</span>,<span class="st">"steelblue"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">k</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/range.html" class="external-link">range</a></span><span class="op">(</span><span class="va">hd</span><span class="op">$</span><span class="va">retentionTime</span><span class="op">)</span> <span class="op">/</span> <span class="fl">60</span><span class="op">)</span></span>
<span><span class="va">nk</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/axis.html" class="external-link">axis</a></span><span class="op">(</span>side<span class="op">=</span><span class="fl">1</span>, at<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">1</span><span class="op">/</span><span class="va">nk</span><span class="op">)</span>, labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="va">k</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>,<span class="va">k</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,<span class="va">k</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">/</span><span class="va">nk</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/barcode-1.png" width="1152"></p>
</div>
<div class="section level4">
<h4 id="animation">Animation<a class="anchor" aria-label="anchor" href="#animation"></a>
</h4>
<p>The following animation scrolls over 5 minutes of retention time for
a MZ range between 521 and 523.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yihui.org/animation/" class="external-link">"animation"</a></span><span class="op">)</span></span>
<span><span class="va">an1</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">5</span>, <span class="fl">0.2</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="va">rtsel</span> <span class="op">&lt;-</span> <span class="va">hd</span><span class="op">$</span><span class="va">retentionTime</span><span class="op">[</span><span class="va">ms1</span><span class="op">]</span> <span class="op">/</span> <span class="fl">60</span> <span class="op">&gt;</span> <span class="op">(</span><span class="fl">30</span> <span class="op">+</span> <span class="va">i</span><span class="op">)</span> <span class="op">&amp;</span></span>
<span>            <span class="va">hd</span><span class="op">$</span><span class="va">retentionTime</span><span class="op">[</span><span class="va">ms1</span><span class="op">]</span> <span class="op">/</span> <span class="fl">60</span> <span class="op">&lt;</span> <span class="op">(</span><span class="fl">35</span> <span class="op">+</span> <span class="va">i</span><span class="op">)</span></span>
<span>        <span class="va">M</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSmap-class.html" class="external-link">MSmap</a></span><span class="op">(</span><span class="va">ms</span>, <span class="va">ms1</span><span class="op">[</span><span class="va">rtsel</span><span class="op">]</span>, <span class="fl">521</span>, <span class="fl">523</span>, <span class="fl">.005</span>, <span class="va">hd</span><span class="op">)</span></span>
<span>        <span class="va">M</span><span class="op">@</span><span class="va">map</span><span class="op">[</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSmap-class.html" class="external-link">msMap</a></span><span class="op">(</span><span class="va">M</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSmap-class.html" class="external-link">plot3D</a></span><span class="op">(</span><span class="va">M</span>, rgl <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">saveGIF</span><span class="op">(</span><span class="fu">an1</span><span class="op">(</span><span class="op">)</span>, movie.name <span class="op">=</span> <span class="st">"msanim1.gif"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="figures/msanim1.gif" alt=""><p class="caption">MS animation 1</p>
</div>
<p>The code chunk below scrolls of a slice of retention times while
keeping the retention time constant between 30 and 35 minutes.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">an2</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">2.5</span>, <span class="fl">0.1</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>        <span class="va">rtsel</span> <span class="op">&lt;-</span> <span class="va">hd</span><span class="op">$</span><span class="va">retentionTime</span><span class="op">[</span><span class="va">ms1</span><span class="op">]</span> <span class="op">/</span> <span class="fl">60</span> <span class="op">&gt;</span> <span class="fl">30</span> <span class="op">&amp;</span> <span class="va">hd</span><span class="op">$</span><span class="va">retentionTime</span><span class="op">[</span><span class="va">ms1</span><span class="op">]</span> <span class="op">/</span> <span class="fl">60</span> <span class="op">&lt;</span> <span class="fl">35</span></span>
<span>        <span class="va">mz1</span> <span class="op">&lt;-</span> <span class="fl">520</span> <span class="op">+</span> <span class="va">i</span></span>
<span>        <span class="va">mz2</span> <span class="op">&lt;-</span> <span class="fl">522</span> <span class="op">+</span> <span class="va">i</span></span>
<span>        <span class="va">M</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSmap-class.html" class="external-link">MSmap</a></span><span class="op">(</span><span class="va">ms</span>, <span class="va">ms1</span><span class="op">[</span><span class="va">rtsel</span><span class="op">]</span>, <span class="va">mz1</span>, <span class="va">mz2</span>, <span class="fl">.005</span>, <span class="va">hd</span><span class="op">)</span></span>
<span>        <span class="va">M</span><span class="op">@</span><span class="va">map</span><span class="op">[</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSmap-class.html" class="external-link">msMap</a></span><span class="op">(</span><span class="va">M</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/MSmap-class.html" class="external-link">plot3D</a></span><span class="op">(</span><span class="va">M</span>, rgl <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">saveGIF</span><span class="op">(</span><span class="fu">an2</span><span class="op">(</span><span class="op">)</span>, movie.name <span class="op">=</span> <span class="st">"msanim2.gif"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="figures/msanim2.gif" alt=""><p class="caption">MS animation 2</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="the-msnbase-infrastructure">The <em><a href="https://bioconductor.org/packages/3.17/MSnbase" class="external-link">MSnbase</a></em>
infrastructure<a class="anchor" aria-label="anchor" href="#the-msnbase-infrastructure"></a>
</h3>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://lgatto.github.io/MSnbase" class="external-link">"MSnbase"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">itraqdata</span><span class="op">)</span></span>
<span><span class="va">itraqdata2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/pickPeaks-method.html" class="external-link">pickPeaks</a></span><span class="op">(</span><span class="va">itraqdata</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">itraqdata</span><span class="op">[[</span><span class="fl">25</span><span class="op">]</span><span class="op">]</span>, full <span class="op">=</span> <span class="cn">TRUE</span>, reporters <span class="op">=</span> <span class="va">iTRAQ4</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/msnbviz-1.png" width="700"></p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>oma <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">4</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">itraqdata2</span><span class="op">[[</span><span class="fl">25</span><span class="op">]</span><span class="op">]</span>, <span class="va">itraqdata2</span><span class="op">[[</span><span class="fl">28</span><span class="op">]</span><span class="op">]</span>, sequences <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"IMIDLDGTENK"</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/msnbviz-2.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="the-protviz-package">The <em><a href="https://CRAN.R-project.org/package=protViz" class="external-link">protViz</a></em>
package<a class="anchor" aria-label="anchor" href="#the-protviz-package"></a>
</h3>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/cpanse/protViz/" class="external-link">"protViz"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">msms</span><span class="op">)</span></span>
<span></span>
<span><span class="va">fi</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/protViz/man/fragmentIons.html" class="external-link">fragmentIon</a></span><span class="op">(</span><span class="st">"TAFDEAIAELDTLNEESYK"</span><span class="op">)</span></span>
<span><span class="va">fi.cyz</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span>c<span class="op">=</span><span class="va">fi</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">c</span>, y<span class="op">=</span><span class="va">fi</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">y</span>, z<span class="op">=</span><span class="va">fi</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">z</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/protViz/man/peakplot.html" class="external-link">peakplot</a></span><span class="op">(</span><span class="st">"TAFDEAIAELDTLNEESYK"</span>,</span>
<span>              spec <span class="op">=</span> <span class="va">msms</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,</span>
<span>              fi <span class="op">=</span> <span class="va">fi.cyz</span>,</span>
<span>              itol <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>              ion.axes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/protviz-1.png" width="700"></p>
<p>The <code>peakplot</code> function return the annotation of the MSMS
spectrum that is plotted:</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of 7</span></span>
<span><span class="co">##  $ mZ.Da.error : num [1:57] 215.3 144.27 -2.8 -17.06 2.03 ...</span></span>
<span><span class="co">##  $ mZ.ppm.error: num [1:57] 1808046 758830 -8306 -37724 3501 ...</span></span>
<span><span class="co">##  $ idx         : int [1:57] 1 1 1 3 16 24 41 52 67 88 ...</span></span>
<span><span class="co">##  $ label       : chr [1:57] "c1" "c2" "c3" "c4" ...</span></span>
<span><span class="co">##  $ score       : num -1</span></span>
<span><span class="co">##  $ sequence    : chr "TAFDEAIAELDTLNEESYK"</span></span>
<span><span class="co">##  $ fragmentIon :'data.frame':    19 obs. of  3 variables:</span></span>
<span><span class="co">##   ..$ c: num [1:19] 119 190 337 452 581 ...</span></span>
<span><span class="co">##   ..$ y: num [1:19] 147 310 397 526 655 ...</span></span>
<span><span class="co">##   ..$ z: num [1:19] 130 293 380 509 638 ...</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="preprocessing-of-maldi-ms-spectra">Preprocessing of MALDI-MS spectra<a class="anchor" aria-label="anchor" href="#preprocessing-of-maldi-ms-spectra"></a>
</h3>
<p>The following code chunks demonstrate the usage of the mass
spectrometry preprocessing and plotting routines in the
<code>r CRANpkg("MALDIquant")</code> package. <em><a href="https://CRAN.R-project.org/package=MALDIquant" class="external-link">MALDIquant</a></em>
uses the traditional graphics system. Therefore <em><a href="https://CRAN.R-project.org/package=MALDIquant" class="external-link">MALDIquant</a></em>
overloads the traditional functions <code>plot</code>,
<code>lines</code> and <code>points</code> for its own data types. These
data types represents spectrum and peak lists as S4 classes. Please see
the <em><a href="https://CRAN.R-project.org/package=MALDIquant" class="external-link">MALDIquant</a></em>
<a href="http://cran.r-project.org/web/packages/MALDIquant/vignettes/MALDIquant-intro.pdf" class="external-link">vignette</a>
and the corresponding <a href="http://strimmerlab.org/software/maldiquant/" class="external-link">website</a> for more
details.</p>
<p>After loading some example data a simple <code>plot</code> draws the
raw spectrum.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://strimmerlab.github.io/software/maldiquant/" class="external-link">"MALDIquant"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"fiedler2009subset"</span>, package<span class="op">=</span><span class="st">"MALDIquant"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">fiedler2009subset</span><span class="op">[[</span><span class="fl">14</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/mqraw-1.png" width="700"></p>
<p>After some preprocessing, namely variance stabilization and
smoothing, we use <code>lines</code> to draw our baseline estimate in
our processed spectrum.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">transformedSpectra</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/transformIntensity-methods.html" class="external-link">transformIntensity</a></span><span class="op">(</span><span class="va">fiedler2009subset</span>, method <span class="op">=</span> <span class="st">"sqrt"</span><span class="op">)</span></span>
<span><span class="va">smoothedSpectra</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/smoothIntensity-methods.html" class="external-link">smoothIntensity</a></span><span class="op">(</span><span class="va">transformedSpectra</span>, method <span class="op">=</span> <span class="st">"SavitzkyGolay"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">smoothedSpectra</span><span class="op">[[</span><span class="fl">14</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/estimateBaseline-methods.html" class="external-link">estimateBaseline</a></span><span class="op">(</span><span class="va">smoothedSpectra</span><span class="op">[[</span><span class="fl">14</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, lwd <span class="op">=</span> <span class="fl">2</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/mqestimatebaseline-1.png" width="700"></p>
<p>After removing the background removal we could use <code>plot</code>
again to draw our baseline corrected spectrum.</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rbSpectra</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/removeBaseline-methods.html" class="external-link">removeBaseline</a></span><span class="op">(</span><span class="va">smoothedSpectra</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">rbSpectra</span><span class="op">[[</span><span class="fl">14</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/mqremovebaseline-1.png" width="700"></p>
<p><code>detectPeaks</code> returns a <code>MassPeaks</code> object that
offers the same traditional graphics functions. The next code chunk
demonstrates how to mark the detected peaks in a spectrum.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cbSpectra</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/calibrateIntensity-methods.html" class="external-link">calibrateIntensity</a></span><span class="op">(</span><span class="va">rbSpectra</span>, method <span class="op">=</span> <span class="st">"TIC"</span><span class="op">)</span></span>
<span><span class="va">peaks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/detectPeaks-methods.html" class="external-link">detectPeaks</a></span><span class="op">(</span><span class="va">cbSpectra</span>, SNR <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">cbSpectra</span><span class="op">[[</span><span class="fl">14</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/points.html" class="external-link">points</a></span><span class="op">(</span><span class="va">peaks</span><span class="op">[[</span><span class="fl">14</span><span class="op">]</span><span class="op">]</span>, col <span class="op">=</span> <span class="st">"red"</span>, pch <span class="op">=</span> <span class="fl">4</span>, lwd <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/mqpeaks-1.png" width="700"></p>
<p>Additional there is a special function <code>labelPeaks</code> that
allows to draw the <em>M/Z</em> values above the corresponding peaks.
Next we mark the 5 top peaks in the spectrum.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top5</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/AbstractMassObject-class.html" class="external-link">intensity</a></span><span class="op">(</span><span class="va">peaks</span><span class="op">[[</span><span class="fl">14</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/AbstractMassObject-class.html" class="external-link">intensity</a></span><span class="op">(</span><span class="va">peaks</span><span class="op">[[</span><span class="fl">14</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                                         decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/labelPeaks-methods.html" class="external-link">labelPeaks</a></span><span class="op">(</span><span class="va">peaks</span><span class="op">[[</span><span class="fl">14</span><span class="op">]</span><span class="op">]</span>, index <span class="op">=</span> <span class="va">top5</span>, avoidOverlap <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/mqlabelpeaks-1.png" width="700"></p>
<p>Often multiple spectra have to be recalibrated to be comparable.
Therefore <em><a href="https://CRAN.R-project.org/package=MALDIquant" class="external-link">MALDIquant</a></em>
warps the spectra according to so called reference or landmark peaks.
For debugging the <code>determineWarpingFunctions</code> function offers
some warping plots. Here we show only the last 4 plots:</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">warpingFunctions</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/determineWarpingFunctions-functions.html" class="external-link">determineWarpingFunctions</a></span><span class="op">(</span><span class="va">peaks</span>,</span>
<span>                              tolerance <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>                              plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                              plotInteractive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/mqwarp-1.png" width="700"></p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">warpedSpectra</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/warp-functions.html" class="external-link">warpMassSpectra</a></span><span class="op">(</span><span class="va">cbSpectra</span>, <span class="va">warpingFunctions</span><span class="op">)</span></span>
<span><span class="va">warpedPeaks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/warp-functions.html" class="external-link">warpMassPeaks</a></span><span class="op">(</span><span class="va">peaks</span>, <span class="va">warpingFunctions</span><span class="op">)</span></span></code></pre></div>
<p>In the next code chunk we visualise the need and the effect of the
recalibration.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">10</span>, <span class="fl">14</span>, <span class="fl">16</span><span class="op">)</span></span>
<span><span class="va">xlim</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4180</span>, <span class="fl">4240</span><span class="op">)</span></span>
<span><span class="va">ylim</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1.9e-3</span><span class="op">)</span></span>
<span><span class="va">lty</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">4</span>, <span class="fl">2</span>, <span class="fl">6</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">cbSpectra</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, xlim <span class="op">=</span> <span class="va">xlim</span>, ylim <span class="op">=</span> <span class="va">ylim</span>, type <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span>along <span class="op">=</span> <span class="va">sel</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span><span class="va">peaks</span><span class="op">[[</span><span class="va">sel</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">]</span>, lty <span class="op">=</span> <span class="va">lty</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, col <span class="op">=</span> <span class="va">i</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span><span class="va">cbSpectra</span><span class="op">[[</span><span class="va">sel</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">]</span>, lty <span class="op">=</span> <span class="va">lty</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, col <span class="op">=</span> <span class="va">i</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">cbSpectra</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, xlim <span class="op">=</span> <span class="va">xlim</span>, ylim <span class="op">=</span> <span class="va">ylim</span>, type <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span>along <span class="op">=</span> <span class="va">sel</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span><span class="va">warpedPeaks</span><span class="op">[[</span><span class="va">sel</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">]</span>, lty <span class="op">=</span> <span class="va">lty</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, col <span class="op">=</span> <span class="va">i</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span><span class="va">warpedSpectra</span><span class="op">[[</span><span class="va">sel</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">]</span>, lty <span class="op">=</span> <span class="va">lty</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>, col <span class="op">=</span> <span class="va">i</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/mqwarped-1.png" width="700"></p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>The code chunks above generate plots that are very similar to the
figure 7 in the corresponding paper <em>“Visualisation of proteomics
data using R”</em>. Please find the code to exactly reproduce the figure
at: <a href="https://github.com/sgibb/MALDIquantExamples/blob/master/R/createFigure1_color.R" class="external-link uri">https://github.com/sgibb/MALDIquantExamples/blob/master/R/createFigure1_color.R</a></p>
</div>
</div>
<div class="section level2">
<h2 id="genomic-and-protein-sequences">Genomic and protein sequences<a class="anchor" aria-label="anchor" href="#genomic-and-protein-sequences"></a>
</h2>
<p>These visualisations originate from the <code>Pbase</code> <a href="http://bioconductor.org/packages/devel/bioc/vignettes/Pbase/inst/doc/Pbase-data.html" class="external-link"><code>Pbase-data</code></a>
and <a href="http://bioconductor.org/packages/devel/bioc/vignettes/Pbase/inst/doc/mapping.html" class="external-link"><code>mapping</code></a>
vignettes.</p>
</div>
<div class="section level2">
<h2 id="mass-spectrometry-imaging">Mass spectrometry imaging<a class="anchor" aria-label="anchor" href="#mass-spectrometry-imaging"></a>
</h2>
<p>The following code chunk downloads a MALDI imaging dataset from a
mouse kidney shared by <a href="http://figshare.com/articles/MALDI_Imaging_Mass_Spectrometry_of_a_Mouse_Kidney/735961" class="external-link">Adrien
Nyakas and Stefan Schurch</a> and generates a plot with the mean
spectrum and three slices of interesting <em>M/Z</em> regions.</p>
<!-- Not running this code chunk as the access to the data (75M) -->
<!-- on figshare seems to hang or become very slow at times -->
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://strimmerlab.github.io/software/maldiquant/" class="external-link">"MALDIquant"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://strimmerlab.github.io/software/maldiquant/" class="external-link">"MALDIquantForeign"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">spectra</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquantForeign/man/importBrukerFlex-functions.html" class="external-link">importBrukerFlex</a></span><span class="op">(</span><span class="st">"http://files.figshare.com/1106682/MouseKidney_IMS_testdata.zip"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">spectra</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/smoothIntensity-methods.html" class="external-link">smoothIntensity</a></span><span class="op">(</span><span class="va">spectra</span>, <span class="st">"SavitzkyGolay"</span>,  halfWindowSize <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></span>
<span><span class="va">spectra</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/removeBaseline-methods.html" class="external-link">removeBaseline</a></span><span class="op">(</span><span class="va">spectra</span>, method <span class="op">=</span> <span class="st">"TopHat"</span>, halfWindowSize <span class="op">=</span> <span class="fl">16</span><span class="op">)</span></span>
<span><span class="va">spectra</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/calibrateIntensity-methods.html" class="external-link">calibrateIntensity</a></span><span class="op">(</span><span class="va">spectra</span>, method <span class="op">=</span> <span class="st">"TIC"</span><span class="op">)</span></span>
<span><span class="va">avgSpectrum</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/averageMassSpectra-functions.html" class="external-link">averageMassSpectra</a></span><span class="op">(</span><span class="va">spectra</span><span class="op">)</span></span>
<span><span class="va">avgPeaks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/detectPeaks-methods.html" class="external-link">detectPeaks</a></span><span class="op">(</span><span class="va">avgSpectrum</span>, SNR <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span></span>
<span><span class="va">avgPeaks</span> <span class="op">&lt;-</span> <span class="va">avgPeaks</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/AbstractMassObject-class.html" class="external-link">intensity</a></span><span class="op">(</span><span class="va">avgPeaks</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0.0015</span><span class="op">]</span></span>
<span></span>
<span><span class="va">oldPar</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>no.readonly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/layout.html" class="external-link">layout</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span>,<span class="fl">4</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">2</span>, byrow <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/plot.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">avgSpectrum</span>, main <span class="op">=</span> <span class="st">"mean spectrum"</span>,</span>
<span>     xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3000</span>, <span class="fl">6000</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.007</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span><span class="va">avgPeaks</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/labelPeaks-methods.html" class="external-link">labelPeaks</a></span><span class="op">(</span><span class="va">avgPeaks</span>, cex <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">0.5</span>, <span class="fl">1.5</span>, <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/plotMsiSlice-methods.html" class="external-link">plotMsiSlice</a></span><span class="op">(</span><span class="va">spectra</span>,</span>
<span>             center <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/MALDIquant/man/AbstractMassObject-class.html" class="external-link">mass</a></span><span class="op">(</span><span class="va">avgPeaks</span><span class="op">)</span>,</span>
<span>             tolerance <span class="op">=</span> <span class="fl">1</span>,</span>
<span>             plotInteractive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span><span class="va">oldPar</span><span class="op">)</span></span></code></pre></div>
<p><img src="figures/mqmsi-1.png" alt="mqmsi">)]</p>
<div class="section level3">
<h3 id="an-interactive-shiny-app-for-imaging-mass-spectrometry">An interactive <em><a href="https://CRAN.R-project.org/package=shiny" class="external-link">shiny</a></em> app for
Imaging mass spectrometry<a class="anchor" aria-label="anchor" href="#an-interactive-shiny-app-for-imaging-mass-spectrometry"></a>
</h3>
<p>There is also an interactive <a href="https://github.com/sgibb/ims-shiny" class="external-link">MALDIquant IMS shiny app</a>
for demonstration purposes. A screen shot is displayed below. To start
the application:</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://shiny.rstudio.com/" class="external-link">"shiny"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://shiny.rstudio.com/reference/shiny/latest/runUrl.html" class="external-link">runGitHub</a></span><span class="op">(</span><span class="st">"sgibb/ims-shiny"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="figures/ims-shiny.png" alt=""><p class="caption">ims-shiny screeshot</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="spatial-proteomics">Spatial proteomics<a class="anchor" aria-label="anchor" href="#spatial-proteomics"></a>
</h2>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/lgatto/pRoloc" class="external-link">"pRoloc"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/lgatto/pRolocdata" class="external-link">"pRolocdata"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## these params use class weights</span></span>
<span><span class="va">fn</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">dir</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"pRoloc"</span><span class="op">)</span>,</span>
<span>          full.names <span class="op">=</span> <span class="cn">TRUE</span>, pattern <span class="op">=</span> <span class="st">"params2.rda"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">fn</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/getStockcol.html" class="external-link">setStockcol</a></span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/getStockcol.html" class="external-link">setStockcol</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/getStockcol.html" class="external-link">getStockcol</a></span><span class="op">(</span><span class="op">)</span>, <span class="fl">90</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">w</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span><span class="op">[</span>, <span class="st">"pd.markers"</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="op">(</span><span class="va">w</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">/</span><span class="va">w</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">w</span><span class="op">)</span> <span class="op">!=</span> <span class="st">"unknown"</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##  Cytoskeleton            ER         Golgi      Lysosome mitochondrion </span></span>
<span><span class="co">##    0.14285714    0.05000000    0.16666667    0.12500000    0.07142857 </span></span>
<span><span class="co">##       Nucleus    Peroxisome            PM    Proteasome  Ribosome 40S </span></span>
<span><span class="co">##    0.05000000    0.25000000    0.06666667    0.09090909    0.07142857 </span></span>
<span><span class="co">##  Ribosome 60S </span></span>
<span><span class="co">##    0.04000000</span></span></code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tan2009r1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/svmClassification.html" class="external-link">svmClassification</a></span><span class="op">(</span><span class="va">tan2009r1</span>, <span class="va">params2</span>,</span>
<span>                               class.weights <span class="op">=</span> <span class="va">w</span>,</span>
<span>                               fcol <span class="op">=</span> <span class="st">"pd.markers"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "pd.markers"</span></span></code></pre>
<pre><code><span><span class="co">## Registered S3 method overwritten by 'gdata':</span></span>
<span><span class="co">##   method         from  </span></span>
<span><span class="co">##   reorder.factor gplots</span></span></code></pre>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ptsze</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">exp</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span><span class="op">$</span><span class="va">svm.scores</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1</span></span></code></pre></div>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">lout</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">4</span><span class="op">)</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fl">4</span>, nrow <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/layout.html" class="external-link">layout</a></span><span class="op">(</span><span class="va">lout</span><span class="op">)</span></span>
<span><span class="va">cls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/getStockcol.html" class="external-link">getStockcol</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">4</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/plotDist.html" class="external-link">plotDist</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span><span class="op">$</span><span class="va">PLSDA</span> <span class="op">==</span> <span class="st">"mitochondrion"</span><span class="op">)</span>, <span class="op">]</span>,</span>
<span>         markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html" class="external-link">featureNames</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span><span class="op">$</span><span class="va">markers.orig</span> <span class="op">==</span> <span class="st">"mitochondrion"</span><span class="op">)</span><span class="op">]</span>,</span>
<span>         mcol <span class="op">=</span> <span class="va">cls</span><span class="op">[</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></span><span class="op">(</span><span class="st">"topright"</span>, legend <span class="op">=</span> <span class="st">"mitochondrion"</span>, bty <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/plotDist.html" class="external-link">plotDist</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span><span class="op">$</span><span class="va">PLSDA</span> <span class="op">==</span> <span class="st">"ER/Golgi"</span><span class="op">)</span>, <span class="op">]</span>,</span>
<span>         markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html" class="external-link">featureNames</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span><span class="op">$</span><span class="va">markers.orig</span> <span class="op">==</span> <span class="st">"ER"</span><span class="op">)</span><span class="op">]</span>,</span>
<span>         mcol <span class="op">=</span> <span class="va">cls</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></span><span class="op">(</span><span class="st">"topright"</span>, legend <span class="op">=</span> <span class="st">"ER"</span>, bty <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/plotDist.html" class="external-link">plotDist</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span><span class="op">$</span><span class="va">PLSDA</span> <span class="op">==</span> <span class="st">"ER/Golgi"</span><span class="op">)</span>, <span class="op">]</span>,</span>
<span>         markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html" class="external-link">featureNames</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span><span class="op">$</span><span class="va">markers.orig</span> <span class="op">==</span> <span class="st">"Golgi"</span><span class="op">)</span><span class="op">]</span>,</span>
<span>         mcol <span class="op">=</span> <span class="va">cls</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></span><span class="op">(</span><span class="st">"topright"</span>, legend <span class="op">=</span> <span class="st">"Golgi"</span>, bty <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/plotDist.html" class="external-link">plotDist</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span><span class="op">$</span><span class="va">PLSDA</span> <span class="op">==</span> <span class="st">"PM"</span><span class="op">)</span>, <span class="op">]</span>,</span>
<span>         markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureNames.html" class="external-link">featureNames</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">tan2009r1</span><span class="op">)</span><span class="op">$</span><span class="va">markers.orig</span> <span class="op">==</span> <span class="st">"PM"</span><span class="op">)</span><span class="op">]</span>,</span>
<span>         mcol <span class="op">=</span> <span class="va">cls</span><span class="op">[</span><span class="fl">8</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></span><span class="op">(</span><span class="st">"topright"</span>, legend <span class="op">=</span> <span class="st">"PM"</span>, bty <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/plot2D.html" class="external-link">plot2D</a></span><span class="op">(</span><span class="va">tan2009r1</span>, fcol <span class="op">=</span> <span class="st">"svm"</span>, cex <span class="op">=</span> <span class="va">ptsze</span>, method <span class="op">=</span> <span class="st">"kpca"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/addLegend.html" class="external-link">addLegend</a></span><span class="op">(</span><span class="va">tan2009r1</span>, where <span class="op">=</span> <span class="st">"bottomleft"</span>, fcol <span class="op">=</span> <span class="st">"svm"</span>, bty <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span></span></code></pre></div>
<p><img src="RProtVis_files/figure-html/spatplot-1.png" width="1152"></p>
<p>See the <a href="http://bioconductor.org/packages/release/bioc/vignettes/pRoloc/inst/doc/pRoloc-tutorial.pdf" class="external-link"><code>pRoloc-tutorial</code></a>
vignette (pdf) from the <em><a href="https://bioconductor.org/packages/3.17/pRoloc" class="external-link">pRoloc</a></em>
package for details about spatial proteomics data analysis and
visualisation.</p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>, locale <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R Under development (unstable) (2023-01-10 r83596)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.1 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] beanplot_1.3.1         ggplot2_3.4.0          lattice_0.20-45       </span></span>
<span><span class="co">##  [4] e1071_1.7-12           msmsTests_1.37.0       msmsEDA_1.37.0        </span></span>
<span><span class="co">##  [7] pRolocdata_1.37.0      pRoloc_1.39.0          BiocParallel_1.33.9   </span></span>
<span><span class="co">## [10] MLInterfaces_1.79.1    cluster_2.1.4          annotate_1.77.0       </span></span>
<span><span class="co">## [13] XML_3.99-0.13          AnnotationDbi_1.61.0   IRanges_2.33.0        </span></span>
<span><span class="co">## [16] MALDIquantForeign_0.13 MALDIquant_1.22        RColorBrewer_1.1-3    </span></span>
<span><span class="co">## [19] xtable_1.8-4           rpx_2.5.1              knitr_1.41            </span></span>
<span><span class="co">## [22] DT_0.26                protViz_0.7.3          BiocManager_1.30.19   </span></span>
<span><span class="co">## [25] RforProteomics_1.37.1  MSnbase_2.25.1         ProtGenerics_1.31.0   </span></span>
<span><span class="co">## [28] S4Vectors_0.37.3       mzR_2.33.0             Rcpp_1.0.9            </span></span>
<span><span class="co">## [31] Biobase_2.59.0         BiocGenerics_0.45.0    BiocStyle_2.27.0      </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] segmented_1.6-2             fs_1.5.2                   </span></span>
<span><span class="co">##   [3] matrixStats_0.63.0          bitops_1.0-7               </span></span>
<span><span class="co">##   [5] lubridate_1.9.0             httr_1.4.4                 </span></span>
<span><span class="co">##   [7] doParallel_1.0.17           tools_4.3.0                </span></span>
<span><span class="co">##   [9] utf8_1.2.2                  R6_2.5.1                   </span></span>
<span><span class="co">##  [11] lazyeval_0.2.2              withr_2.5.0                </span></span>
<span><span class="co">##  [13] prettyunits_1.1.1           gridExtra_2.3              </span></span>
<span><span class="co">##  [15] preprocessCore_1.61.0       cli_3.6.0                  </span></span>
<span><span class="co">##  [17] textshaping_0.3.6           readBrukerFlexData_1.9.0   </span></span>
<span><span class="co">##  [19] labeling_0.4.2              sass_0.4.4                 </span></span>
<span><span class="co">##  [21] mvtnorm_1.1-3               randomForest_4.7-1.1       </span></span>
<span><span class="co">##  [23] proxy_0.4-27                pkgdown_2.0.7.9000         </span></span>
<span><span class="co">##  [25] systemfonts_1.0.4           R.utils_2.12.2             </span></span>
<span><span class="co">##  [27] parallelly_1.33.0           limma_3.55.2               </span></span>
<span><span class="co">##  [29] impute_1.73.0               RSQLite_2.2.20             </span></span>
<span><span class="co">##  [31] FNN_1.1.3.1                 generics_0.1.3             </span></span>
<span><span class="co">##  [33] crosstalk_1.2.0             gtools_3.9.4               </span></span>
<span><span class="co">##  [35] dplyr_1.0.10                dendextend_1.16.0          </span></span>
<span><span class="co">##  [37] Matrix_1.5-3                fansi_1.0.3                </span></span>
<span><span class="co">##  [39] R.methodsS3_1.8.2           lifecycle_1.0.3            </span></span>
<span><span class="co">##  [41] yaml_2.3.6                  edgeR_3.41.2               </span></span>
<span><span class="co">##  [43] SummarizedExperiment_1.29.1 qvalue_2.31.1              </span></span>
<span><span class="co">##  [45] gplots_3.1.3                biocViews_1.67.2           </span></span>
<span><span class="co">##  [47] recipes_1.0.4               BiocFileCache_2.7.1        </span></span>
<span><span class="co">##  [49] grid_4.3.0                  blob_1.2.3                 </span></span>
<span><span class="co">##  [51] gdata_2.18.0.1              crayon_1.5.2               </span></span>
<span><span class="co">##  [53] KEGGREST_1.39.0             pillar_1.8.1               </span></span>
<span><span class="co">##  [55] GenomicRanges_1.51.4        future.apply_1.10.0        </span></span>
<span><span class="co">##  [57] lpSolve_5.6.17              codetools_0.2-18           </span></span>
<span><span class="co">##  [59] glue_1.6.2                  pcaMethods_1.91.0          </span></span>
<span><span class="co">##  [61] data.table_1.14.6           vctrs_0.5.1                </span></span>
<span><span class="co">##  [63] png_0.1-8                   gtable_0.3.1               </span></span>
<span><span class="co">##  [65] kernlab_0.9-31              assertthat_0.2.1           </span></span>
<span><span class="co">##  [67] cachem_1.0.6                gower_1.0.1                </span></span>
<span><span class="co">##  [69] xfun_0.36                   prodlim_2019.11.13         </span></span>
<span><span class="co">##  [71] coda_0.19-4                 survival_3.5-0             </span></span>
<span><span class="co">##  [73] ncdf4_1.21                  timeDate_4022.108          </span></span>
<span><span class="co">##  [75] iterators_1.0.14            hardhat_1.2.0              </span></span>
<span><span class="co">##  [77] lava_1.7.1                  ellipsis_0.3.2             </span></span>
<span><span class="co">##  [79] ipred_0.9-13                nlme_3.1-161               </span></span>
<span><span class="co">##  [81] bit64_4.0.5                 progress_1.2.2             </span></span>
<span><span class="co">##  [83] filelock_1.0.2              LaplacesDemon_16.1.6       </span></span>
<span><span class="co">##  [85] GenomeInfoDb_1.35.10        rprojroot_2.0.3            </span></span>
<span><span class="co">##  [87] bslib_0.4.2                 affyio_1.69.0              </span></span>
<span><span class="co">##  [89] KernSmooth_2.23-20          rpart_4.1.19               </span></span>
<span><span class="co">##  [91] colorspace_2.0-3            DBI_1.1.3                  </span></span>
<span><span class="co">##  [93] nnet_7.3-18                 tidyselect_1.2.0           </span></span>
<span><span class="co">##  [95] bit_4.0.5                   compiler_4.3.0             </span></span>
<span><span class="co">##  [97] curl_5.0.0                  graph_1.77.1               </span></span>
<span><span class="co">##  [99] xml2_1.3.3                  desc_1.4.2                 </span></span>
<span><span class="co">## [101] DelayedArray_0.25.0         plotly_4.10.1              </span></span>
<span><span class="co">## [103] bookdown_0.31               scales_1.2.1               </span></span>
<span><span class="co">## [105] caTools_1.18.2              hexbin_1.28.2              </span></span>
<span><span class="co">## [107] affy_1.77.0                 RBGL_1.75.0                </span></span>
<span><span class="co">## [109] rappdirs_0.3.3              stringr_1.5.0              </span></span>
<span><span class="co">## [111] digest_0.6.31               mixtools_2.0.0             </span></span>
<span><span class="co">## [113] rmarkdown_2.19              XVector_0.39.0             </span></span>
<span><span class="co">## [115] htmltools_0.5.4             pkgconfig_2.0.3            </span></span>
<span><span class="co">## [117] base64enc_0.1-3             MatrixGenerics_1.11.0      </span></span>
<span><span class="co">## [119] highr_0.10                  dbplyr_2.2.1               </span></span>
<span><span class="co">## [121] fastmap_1.1.0               rlang_1.0.6                </span></span>
<span><span class="co">## [123] htmlwidgets_1.6.1           farver_2.1.1               </span></span>
<span><span class="co">## [125] jquerylib_0.1.4             jsonlite_1.8.4             </span></span>
<span><span class="co">## [127] mclust_6.0.0                mzID_1.37.0                </span></span>
<span><span class="co">## [129] ModelMetrics_1.2.2.2        R.oo_1.25.0                </span></span>
<span><span class="co">## [131] RCurl_1.98-1.9              magrittr_2.0.3             </span></span>
<span><span class="co">## [133] GenomeInfoDbData_1.2.9      munsell_0.5.0              </span></span>
<span><span class="co">## [135] viridis_0.6.2               MsCoreUtils_1.11.2         </span></span>
<span><span class="co">## [137] vsn_3.67.0                  stringi_1.7.12             </span></span>
<span><span class="co">## [139] pROC_1.18.0                 zlibbioc_1.45.0            </span></span>
<span><span class="co">## [141] MASS_7.3-58.1               plyr_1.8.8                 </span></span>
<span><span class="co">## [143] readMzXmlData_2.8.2         parallel_4.3.0             </span></span>
<span><span class="co">## [145] listenv_0.9.0               Biostrings_2.67.0          </span></span>
<span><span class="co">## [147] splines_4.3.0               hms_1.1.2                  </span></span>
<span><span class="co">## [149] locfit_1.5-9.7              RUnit_0.4.32               </span></span>
<span><span class="co">## [151] reshape2_1.4.4              biomaRt_2.55.0             </span></span>
<span><span class="co">## [153] evaluate_0.19               foreach_1.5.2              </span></span>
<span><span class="co">## [155] tidyr_1.2.1                 purrr_1.0.1                </span></span>
<span><span class="co">## [157] future_1.30.0               clue_0.3-63                </span></span>
<span><span class="co">## [159] viridisLite_0.4.1           class_7.3-20.1             </span></span>
<span><span class="co">## [161] ragg_1.2.5                  tibble_3.1.8               </span></span>
<span><span class="co">## [163] memoise_2.0.1               timechange_0.2.0           </span></span>
<span><span class="co">## [165] globals_0.16.2              caret_6.0-93               </span></span>
<span><span class="co">## [167] sampling_2.9</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Laurent Gatto.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
